<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Real-time analysis and forecasting for outbreak analytics with R: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-XZ471458FQ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XZ471458FQ');
    </script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #FFF">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg">
</div>
    <div class="lesson-title-md">
      Real-time analysis and forecasting for outbreak analytics with R
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Real-time analysis and forecasting for outbreak analytics with R
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
          </ul>
</li>
      </ul>
</div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
<input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset>
</form>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Real-time analysis and forecasting for outbreak analytics with R
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="quantify-transmissibility.html">1. Quantifying transmission</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="create-forecast.html">2. Create a short-term forecast</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="severity-static.html">3. Estimate severity</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-quantify-transmissibility"><p>Content from <a href="quantify-transmissibility.html">Quantifying transmission</a></p>
<hr>
<p>Last updated on 2024-03-25 |
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/episodes/quantify-transmissibility.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I estimate key transmission metrics from a time series of
case data?</li>
<li>How can I quantify geographical heterogeneity in these metrics?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to estimate transmission metrics from a time series of
case data using the R package <code>EpiNow2</code>
</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>Learners should familiarise themselves with following concept
dependencies before working through this tutorial:</p>
<p><strong>Statistics</strong> : probability distributions, principle of
Bayesian analysis.</p>
<p><strong>Epidemic theory</strong> : Effective reproduction number.</p>
</div>
</div>
</div>
<div id="reminder-the-effective-reproduction-number-r_t" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="reminder-the-effective-reproduction-number-r_t" class="callout-inner">
<h3 class="callout-title">Reminder: the Effective Reproduction Number,<span class="math inline">\(R_t\)</span><a class="anchor" aria-label="anchor" href="#reminder-the-effective-reproduction-number-r_t"></a>
</h3>
<div class="callout-content">
<p>The <a href="reference.html#basic">basic reproduction number</a>,
<span class="math inline">\(R_0\)</span>, is the average number of cases
caused by one infectious individual in a entirely susceptible
population.</p>
<p>But in an ongoing outbreak, the population does not remain entirely
susceptible as those that recover from infection are typically immune.
Moreover, there can be changes in behaviour or other factors that affect
transmission. When we are interested in monitoring changes in
transmission we are therefore more interested in the value of the
<strong>effective reproduction number</strong>, <span class="math inline">\(R_t\)</span>, the average number of cases caused
by one infectious individual in the population at time <span class="math inline">\(t\)</span>.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Quantifying transmission metrics at the start of an outbreak can give
important information on the strength of transmission (reproduction
number) and the speed of transmission (<a href="reference.html#growth">growth rate</a>, doubling/halving time). To
estimate these key metrics using case data we must account for delays
between the date of infections and date of reported cases. In an
outbreak situation, data are usually available on reported dates only,
therefore we must use estimation methods to account for these delays
when trying to understand changes in transmission over time.</p>
<p>In the next tutorials we will focus on how to implement the functions
in <a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> to estimate transmission metrics of case data.
We will not cover the theoretical background of the models or inference
framework, for details on these concepts see the <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections.html" class="external-link">vignette</a>.
For more details on the distinction between speed and strength of
transmission and implications for control, see <a href="https://royalsocietypublishing.org/doi/full/10.1098/rspb.2020.1556" class="external-link">Dushoff
&amp; Park, 2021</a>.</p>
<div id="bayesian-inference" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="bayesian-inference" class="callout-inner">
<h3 class="callout-title">Bayesian inference<a class="anchor" aria-label="anchor" href="#bayesian-inference"></a>
</h3>
<div class="callout-content">
<p>The R package <code>EpiNow2</code> uses a <a href="reference.html#bayesian">Bayesian inference</a> framework to
estimate reproduction numbers and infection times based on reporting
dates.</p>
<p>In Bayesian inference, we use prior knowledge (prior distributions)
with data (in a likelihood function) to find the posterior
probability.</p>
<p class="text-center" style="background-color: white">
Posterior probability <span class="math inline">\(\propto\)</span>
likelihood <span class="math inline">\(\times\)</span> prior probability
</p>
</div>
</div>
</div>
<p>The first step is to load the <a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> package :</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">EpiNow2</span><span class="op">)</span></span></code></pre>
</div>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>This tutorial illustrates the usage of <code>epinow()</code> to
estimate the time-varying reproduction number and infection times.
Learners should understand the necessary inputs to the model and the
limitations of the model output.</p>
</div>
</div>
</div>
</div>
</section><section id="delay-distributions-and-case-data"><h2 class="section-heading">Delay distributions and case data<a class="anchor" aria-label="anchor" href="#delay-distributions-and-case-data"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="case-data">Case data<a class="anchor" aria-label="anchor" href="#case-data"></a>
</h3>
<p>To illustrate the functions of <code>EpiNow2</code> we will use
outbreak data of the start of the COVID-19 pandemic from the United
Kingdom. The data are available in the R package
<a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a>.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        date          region region_code cases_new cases_total deaths_new
1 2020-01-30   East Midlands   E12000004        NA          NA         NA
2 2020-01-30 East of England   E12000006        NA          NA         NA
3 2020-01-30         England   E92000001         2           2         NA
4 2020-01-30          London   E12000007        NA          NA         NA
5 2020-01-30      North East   E12000001        NA          NA         NA
6 2020-01-30      North West   E12000002        NA          NA         NA
  deaths_total recovered_new recovered_total hosp_new hosp_total tested_new
1           NA            NA              NA       NA         NA         NA
2           NA            NA              NA       NA         NA         NA
3           NA            NA              NA       NA         NA         NA
4           NA            NA              NA       NA         NA         NA
5           NA            NA              NA       NA         NA         NA
6           NA            NA              NA       NA         NA         NA
  tested_total
1           NA
2           NA
3           NA
4           NA
5           NA
6           NA</code></pre>
</div>
<p>To use the data, we must format the data to have two columns:</p>
<ul>
<li>
<code>date</code> : the date (as a date object see
<code>?is.Date()</code>),</li>
<li>
<code>confirm</code> : number of confirmed cases on that date.</li>
</ul>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cases</span> <span class="op">&lt;-</span> <span class="fu">aggregate</span><span class="op">(</span></span>
<span>  <span class="va">cases_new</span> <span class="op">~</span> <span class="va">date</span>,</span>
<span>  data <span class="op">=</span> <span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"cases_new"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  FUN <span class="op">=</span> <span class="va">sum</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">cases</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"confirm"</span><span class="op">)</span></span></code></pre>
</div>
<p>There are case data available for 489 days, but in an outbreak
situation it is likely we would only have access to the beginning of
this data set. Therefore we assume we only have the first 90 days of
this data.</p>
<figure><img src="../fig/quantify-transmissibility-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
<div class="section level3">
<h3 id="delay-distributions">Delay distributions<a class="anchor" aria-label="anchor" href="#delay-distributions"></a>
</h3>
<p>We assume there are delays from the time of infection until the time
a case is reported. We specify these delays as distributions to account
for the uncertainty in individual level differences. The delay can
consist of multiple types of delays/processes. A typical delay from time
of infection to case reporting may consist of :</p>
<p class="text-center" style="background-color: aliceblue">
<strong>time from infection to symptom onset</strong> (the <a href="reference.html#incubation">incubation period</a>) + <strong>time
from symptom onset to case notification</strong> (the reporting time) .
</p>
<p>The delay distribution for each of these processes can either
estimated from data or obtained from the literature. We can express
uncertainty about what the correct parameters of the distributions by
assuming the distributions have <strong>fixed</strong> parameters or
whether they have <strong>variable</strong> parameters. To understand
the difference between <strong>fixed</strong> and
<strong>variable</strong> distributions, let’s consider the incubation
period.</p>
<div id="delays-and-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="delays-and-data" class="callout-inner">
<h3 class="callout-title">Delays and data<a class="anchor" aria-label="anchor" href="#delays-and-data"></a>
</h3>
<div class="callout-content">
<p>The number of delays and type of delay is a flexible input that
depends on the data. The examples below highlight how the delays can be
specified for different data sources:</p>
<center>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Data source</th>
<th>Delay(s)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Time of case report</td>
<td>Incubation period + time from symptom onset to case
notification</td>
</tr>
<tr class="even">
<td>Time of hospitalisation</td>
<td>Incubation period + time from symptom onset to hospitalisation</td>
</tr>
<tr class="odd">
<td>Time of symptom onset</td>
<td>Incubation period</td>
</tr>
</tbody>
</table>
</center>
</div>
</div>
</div>
<div class="section level4">
<h4 id="incubation-period-distribution">Incubation period distribution<a class="anchor" aria-label="anchor" href="#incubation-period-distribution"></a>
</h4>
<p>The distribution of incubation period can usually be obtained from
the literature. The package <code>{epiparameter}</code> contains a
library of epidemiological parameters for different diseases obtained
from the literature.</p>
<p>We specify a (fixed) gamma distribution with mean <span class="math inline">\(\mu = 4\)</span> and standard deviation <span class="math inline">\(\sigma^2= 2\)</span> (shape = <span class="math inline">\(4\)</span>, scale = <span class="math inline">\(1\)</span>) using the function
<code>dist_spec()</code> as follows:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">incubation_period_fixed</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">4</span>, sd <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">incubation_period_fixed</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
  Fixed distribution with PMF [0.019 0.12 0.21 0.21 0.17 0.11 0.069 0.039 0.021 0.011 0.0054 0.0026 0.0012 0.00058 0.00026 0.00012 5.3e-05 2.3e-05 1e-05 4.3e-06]</code></pre>
</div>
<p>The argument <code>max</code> is the maximum value the distribution
can take, in this example 20 days.</p>
<div id="why-a-gamma-distrubution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="why-a-gamma-distrubution" class="callout-inner">
<h3 class="callout-title">Why a gamma distrubution?<a class="anchor" aria-label="anchor" href="#why-a-gamma-distrubution"></a>
</h3>
<div class="callout-content">
<p>The incubation period has to be positive in value. Therefore we must
specific a distribution in <code>dist_spec</code> which is for positive
values only.</p>
<p><code>dist_spec()</code> supports log normal and gamma distributions,
which are distributions for positive values only.</p>
<p>For all types of delay, we will need to use distributions for
positive values only - we don’t want to include delays of negative days
in our analysis!</p>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="including-distribution-uncertainty">Including distribution uncertainty<a class="anchor" aria-label="anchor" href="#including-distribution-uncertainty"></a>
</h4>
<p>To specify a <strong>variable</strong> distribution, we include
uncertainty around the mean <span class="math inline">\(\mu\)</span> and
standard deviation <span class="math inline">\(\sigma^2\)</span> of our
gamma distribution. If our incubation period distribution has a mean
<span class="math inline">\(\mu\)</span> and standard deviation <span class="math inline">\(\sigma^2\)</span>, then we assume the mean (<span class="math inline">\(\mu\)</span>) follows a Normal distribution with
standard deviation <span class="math inline">\(\sigma_{\mu}^2\)</span>:</p>
<p><span class="math display">\[\mbox{Normal}(\mu,\sigma_{\mu}^2)\]</span></p>
<p>and a standard deviation (<span class="math inline">\(\sigma^2\)</span>) follows a Normal distribution
with standard deviation <span class="math inline">\(\sigma_{\sigma^2}^2\)</span>:</p>
<p><span class="math display">\[\mbox{Normal}(\sigma^2,\sigma_{\sigma^2}^2).\]</span></p>
<p>We specify this using <code>dist_spec</code> with the additional
arguments <code>mean_sd</code> (<span class="math inline">\(\sigma_{\mu}^2\)</span>) and <code>sd_sd</code>
(<span class="math inline">\(\sigma_{\sigma^2}^2\)</span>).</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">incubation_period_variable</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">4</span>, sd <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">incubation_period_variable</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
  Uncertain gamma distribution with (untruncated) mean 4 (SD 0.5) and SD 2 (SD 0.5)</code></pre>
</div>
</div>
<div class="section level4">
<h4 id="reporting-delays">Reporting delays<a class="anchor" aria-label="anchor" href="#reporting-delays"></a>
</h4>
<p>After the incubation period, there will be an additional delay of
time from symptom onset to case notification: the reporting delay. We
can specify this as a fixed or variable distribution, or estimate a
distribution from data.</p>
<p>When specifying a distribution, it is useful to visualise the
probability density to see the peak and spread of the distribution, in
this case we will use a log normal distribution. We can use the
functions <code>convert_to_logmean()</code> and
<code>convert_to_logsd()</code> to convert the mean and standard
deviation of a normal distribution to that of a log normal
distribution.</p>
<p>If we want to assume that the mean reporting delay is 2 days (with a
standard deviation of 1 day), the log normal distribution will look
like:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">log_mean</span> <span class="op">&lt;-</span> <span class="fu">convert_to_logmean</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">log_sd</span> <span class="op">&lt;-</span> <span class="fu">convert_to_logsd</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">seq</span><span class="op">(</span>from <span class="op">=</span> <span class="fl">0</span>, to <span class="op">=</span> <span class="fl">10</span>, length <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, density <span class="op">=</span> <span class="fu">dlnorm</span><span class="op">(</span><span class="va">x</span>, meanlog <span class="op">=</span> <span class="va">log_mean</span>, sdlog <span class="op">=</span> <span class="va">log_sd</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span><span class="va">x</span>, <span class="va">density</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_grey</span><span class="op">(</span></span>
<span>    base_size <span class="op">=</span> <span class="fl">15</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/quantify-transmissibility-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Using the mean and standard deviation for the log normal
distribution, we can specify a fixed or variable distribution using
<code>dist_spec()</code> as before:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reporting_delay_variable</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="va">log_mean</span>, sd <span class="op">=</span> <span class="va">log_sd</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">10</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>If data is available on the time between symptom onset and reporting,
we can use the function <code>estimate_delay()</code> to estimate a log
normal distribution from a vector of delays. The code below illustrates
how to use <code>estimate_delay()</code> with synthetic delay data.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_data</span> <span class="op">&lt;-</span> <span class="fu">rlnorm</span><span class="op">(</span><span class="fl">500</span>, <span class="fu">log</span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span> <span class="co"># synthetic delay data</span></span>
<span><span class="va">reporting_delay</span> <span class="op">&lt;-</span> <span class="fu">estimate_delay</span><span class="op">(</span></span>
<span>  <span class="va">delay_data</span>,</span>
<span>  samples <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  bootstraps <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="generation-time">Generation time<a class="anchor" aria-label="anchor" href="#generation-time"></a>
</h4>
<p>We also must specify a distribution for the generation time. Here we
will use a log normal distribution with mean 3.6 and standard deviation
3.1 (<a href="https://doi.org/10.2807/1560-7917.ES.2020.25.17.2000257" class="external-link">Ganyani
et al. 2020</a>).</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">generation_time_variable</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">3.6</span>, sd <span class="op">=</span> <span class="fl">3.1</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</section><section id="finding-estimates"><h2 class="section-heading">Finding estimates<a class="anchor" aria-label="anchor" href="#finding-estimates"></a>
</h2>
<hr class="half-width">
<p>The function <code>epinow()</code> is a wrapper for the function
<code>estimate_infections()</code> used to estimate cases by date of
infection. The generation time distribution and delay distributions must
be passed using the functions <code>generation_time_opts()</code> and
<code>delay_opts()</code> respectively.</p>
<p>There are numerous other inputs that can be passed to
<code>epinow()</code>, see <code>EpiNow2::?epinow()</code> for more
detail. One optional input is to specify a log normal prior for the
effective reproduction number <span class="math inline">\(R_t\)</span>
at the start of the outbreak. We specify a mean and standard deviation
as arguments of <code>prior</code> within <code>rt_opts()</code>:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rt_log_mean</span> <span class="op">&lt;-</span> <span class="fu">convert_to_logmean</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">rt_log_sd</span> <span class="op">&lt;-</span> <span class="fu">convert_to_logsd</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">rt</span> <span class="op">&lt;-</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div id="bayesian-inference-using-stan" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="bayesian-inference-using-stan" class="callout-inner">
<h3 class="callout-title">Bayesian inference using Stan<a class="anchor" aria-label="anchor" href="#bayesian-inference-using-stan"></a>
</h3>
<div class="callout-content">
<p>The Bayesian inference is performed using MCMC methods with the
program <a href="https://mc-stan.org/" class="external-link">Stan</a>. There are a number of
default inputs to the Stan functions including the number of chains and
number of samples per chain (see
<code>?EpiNow2::stan_opts()</code>).</p>
<p>To reduce computation time, we can run chains in parallel. To do
this, we must set the number of cores to be used. By default, 4 MCMC
chains are run (see <code>stan_opts()$chains</code>), so we can set an
equal number of cores to be used in parallel as follows:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">withr</span><span class="fu">::</span><span class="fu">local_options</span><span class="op">(</span><span class="fu">list</span><span class="op">(</span>mc.cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>To find the maximum number of available cores on your machine, use
<code>parallel::detectCores()</code>.</p>
</div>
</div>
</div>
<p><em>Note : in the code below fixed distributions are used instead of
variable. This is to speed up computation time. It is generally
recommended to use variable distributions that account for additional
uncertainty.</em></p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reported_cases</span> <span class="op">&lt;-</span> <span class="va">cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span>, <span class="op">]</span></span>
<span><span class="va">estimates</span> <span class="op">&lt;-</span> <span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">reported_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-03-25 19:10:03] epinow: There were 1 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-03-25 19:10:03] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<div class="section level3">
<h3 id="results">Results<a class="anchor" aria-label="anchor" href="#results"></a>
</h3>
<p>We can extract and visualise estimates of the effective reproduction
number through time:</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimates</span><span class="op">$</span><span class="va">plots</span><span class="op">$</span><span class="va">R</span></span></code></pre>
</div>
<figure><img src="../fig/quantify-transmissibility-rendered-unnamed-chunk-15-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The uncertainty in the estimates increases through time. This is
because estimates are informed by data in the past - within the delay
periods. This difference in uncertainty is categorised into
<strong>Estimate</strong> (green) utilises all data and <strong>Estimate
based on partial data</strong> (orange) estimates that are based on less
data (because infections that happened at the time are more likely to
not have been observed yet) and therefore have increasingly wider
intervals towards the date of the last data point. Finally, the
<strong>Forecast</strong> (purple) is a projection ahead of time.</p>
<p>We can also visualise the growth rate estimate through time:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimates</span><span class="op">$</span><span class="va">plots</span><span class="op">$</span><span class="va">growth_rate</span></span></code></pre>
</div>
<figure><img src="../fig/quantify-transmissibility-rendered-unnamed-chunk-16-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To extract a summary of the key transmission metrics at the
<em>latest date</em> in the data:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">estimates</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 measure                 estimate
                                  &lt;char&gt;                   &lt;char&gt;
1: New confirmed cases by infection date     7126 (4149 -- 12509)
2:        Expected change in daily cases        Likely decreasing
3:            Effective reproduction no.       0.89 (0.58 -- 1.3)
4:                        Rate of growth -0.015 (-0.063 -- 0.037)
5:          Doubling/halving time (days)          -45 (19 -- -11)</code></pre>
</div>
<p>As these estimates are based on partial data, they have a wide
uncertainty interval.</p>
<ul>
<li><p>From the summary of our analysis we see that the expected change
in daily cases is Likely decreasing with the estimated new confirmed
cases 7126 (4149 – 12509).</p></li>
<li><p>The effective reproduction number <span class="math inline">\(R_t\)</span> estimate (on the last date of the
data) is 0.89 (0.58 – 1.3).</p></li>
<li><p>The exponential growth rate of case numbers is -0.015 (-0.063 –
0.037).</p></li>
<li><p>The doubling time (the time taken for case numbers to double) is
-45 (19 – -11).</p></li>
</ul>
<div id="expected-change-in-daily-cases" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="expected-change-in-daily-cases" class="callout-inner">
<h3 class="callout-title">
<code>Expected change in daily cases</code><a class="anchor" aria-label="anchor" href="#expected-change-in-daily-cases"></a>
</h3>
<div class="callout-content">
<p>A factor describing expected change in daily cases based on the
posterior probability that <span class="math inline">\(R_t &lt;
1\)</span>.</p>
<center>
<table class="table">
<thead><tr class="header">
<th>Probability (<span class="math inline">\(p\)</span>)</th>
<th>Expected change</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(p &lt; 0.05\)</span></td>
<td>Increasing</td>
</tr>
<tr class="even">
<td><span class="math inline">\(0.05 \leq p&lt; 0.4\)</span></td>
<td>Likely increasing</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(0.4 \leq p&lt; 0.6\)</span></td>
<td>Stable</td>
</tr>
<tr class="even">
<td><span class="math inline">\(0.6 \leq p &lt; 0.95\)</span></td>
<td>Likely decreasing</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(0.95 \leq p \leq 1\)</span></td>
<td>Decreasing</td>
</tr>
</tbody>
</table>
</center>
</div>
</div>
</div>
</div>
</section><section id="quantify-geographical-heterogeneity"><h2 class="section-heading">Quantify geographical heterogeneity<a class="anchor" aria-label="anchor" href="#quantify-geographical-heterogeneity"></a>
</h2>
<hr class="half-width">
<p>The outbreak data of the start of the COVID-19 pandemic from the
United Kingdom from the R package <a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a> includes the
region in which the cases were recorded. To find regional estimates of
the effective reproduction number and cases, we must format the data to
have three columns:</p>
<ul>
<li>
<code>date</code> : the date,</li>
<li>
<code>region</code> : the region,</li>
<li>
<code>confirm</code> : number of confirmed cases for a region on a
given date.</li>
</ul>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">regional_cases</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"cases_new"</span>, <span class="st">"region"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">regional_cases</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"confirm"</span>, <span class="st">"region"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># extract the first 90 dates for all regions</span></span>
<span><span class="va">dates</span> <span class="op">&lt;-</span> <span class="fu">sort</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">regional_cases</span><span class="op">$</span><span class="va">date</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span><span class="op">]</span></span>
<span><span class="va">regional_cases</span> <span class="op">&lt;-</span> <span class="va">regional_cases</span><span class="op">[</span><span class="fu">which</span><span class="op">(</span><span class="va">regional_cases</span><span class="op">$</span><span class="va">date</span> <span class="op">%in%</span> <span class="va">dates</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">regional_cases</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        date confirm          region
1 2020-01-30      NA   East Midlands
2 2020-01-30      NA East of England
3 2020-01-30       2         England
4 2020-01-30      NA          London
5 2020-01-30      NA      North East
6 2020-01-30      NA      North West</code></pre>
</div>
<p>To find regional estimates, we use the same inputs as
<code>epinow()</code> to the function
<code>regional_epinow()</code>:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimates_regional</span> <span class="op">&lt;-</span> <span class="fu">regional_epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">regional_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>INFO [2024-03-25 19:10:09] Producing following optional outputs: regions, summary, samples, plots, latest
INFO [2024-03-25 19:10:09] Reporting estimates using data up to: 2020-04-28
INFO [2024-03-25 19:10:09] No target directory specified so returning output
INFO [2024-03-25 19:10:09] Producing estimates for: East Midlands, East of England, England, London, North East, North West, Northern Ireland, Scotland, South East, South West, Wales, West Midlands, Yorkshire and The Humber
INFO [2024-03-25 19:10:09] Regions excluded: none
INFO [2024-03-25 19:54:28] Completed regional estimates
INFO [2024-03-25 19:54:28] Regions with estimates: 13
INFO [2024-03-25 19:54:28] Regions with runtime errors: 0
INFO [2024-03-25 19:54:29] Producing summary
INFO [2024-03-25 19:54:29] No summary directory specified so returning summary output
INFO [2024-03-25 19:54:29] No target directory specified so returning timings</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimates_regional</span><span class="op">$</span><span class="va">summary</span><span class="op">$</span><span class="va">summarised_results</span><span class="op">$</span><span class="va">table</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                      Region New confirmed cases by infection date
                      &lt;char&gt;                                &lt;char&gt;
 1:            East Midlands                      339 (213 -- 548)
 2:          East of England                      539 (337 -- 846)
 3:                  England                   3558 (2161 -- 5686)
 4:                   London                      298 (187 -- 460)
 5:               North East                      256 (145 -- 425)
 6:               North West                      563 (325 -- 866)
 7:         Northern Ireland                         42 (22 -- 82)
 8:                 Scotland                      289 (152 -- 551)
 9:               South East                      590 (353 -- 978)
10:               South West                      419 (288 -- 604)
11:                    Wales                        95 (63 -- 139)
12:            West Midlands                      271 (141 -- 494)
13: Yorkshire and The Humber                      483 (287 -- 792)
    Expected change in daily cases Effective reproduction no.
                            &lt;fctr&gt;                     &lt;char&gt;
 1:              Likely increasing          1.2 (0.85 -- 1.6)
 2:              Likely increasing          1.2 (0.84 -- 1.6)
 3:              Likely decreasing         0.92 (0.62 -- 1.3)
 4:              Likely decreasing         0.79 (0.55 -- 1.1)
 5:              Likely decreasing          0.92 (0.6 -- 1.3)
 6:              Likely decreasing         0.87 (0.57 -- 1.2)
 7:              Likely decreasing           0.63 (0.38 -- 1)
 8:              Likely decreasing         0.92 (0.57 -- 1.4)
 9:                         Stable         0.99 (0.67 -- 1.4)
10:                     Increasing           1.4 (1.1 -- 1.8)
11:                     Decreasing        0.57 (0.41 -- 0.77)
12:              Likely decreasing         0.71 (0.42 -- 1.1)
13:                         Stable             1 (0.7 -- 1.4)
               Rate of growth Doubling/halving time (days)
                       &lt;char&gt;                       &lt;char&gt;
 1:    0.023 (-0.02 -- 0.067)               30 (10 -- -34)
 2:   0.022 (-0.022 -- 0.065)               31 (11 -- -32)
 3:  -0.011 (-0.055 -- 0.032)              -62 (22 -- -13)
 4:   -0.029 (-0.068 -- 0.01)              -24 (69 -- -10)
 5:  -0.011 (-0.059 -- 0.038)              -63 (18 -- -12)
 6:  -0.017 (-0.064 -- 0.022)              -40 (31 -- -11)
 7:   -0.053 (-0.1 -- 0.0038)            -13 (180 -- -6.9)
 8:   -0.01 (-0.065 -- 0.047)              -67 (15 -- -11)
 9: -0.0013 (-0.048 -- 0.049)             -540 (14 -- -14)
10:    0.047 (0.011 -- 0.086)               15 (8.1 -- 60)
11: -0.064 (-0.095 -- -0.032)            -11 (-22 -- -7.3)
12:  -0.041 (-0.092 -- 0.012)             -17 (56 -- -7.5)
13:  0.0032 (-0.042 -- 0.051)              220 (14 -- -17)</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimates_regional</span><span class="op">$</span><span class="va">summary</span><span class="op">$</span><span class="va">plots</span><span class="op">$</span><span class="va">R</span></span></code></pre>
</div>
<figure><img src="../fig/quantify-transmissibility-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p><code>EpiNow2</code> can be used to estimate transmission metrics
from case data at the start of an outbreak. The reliability of these
estimates depends on the quality of the data and appropriate choice of
delay distributions. In the next tutorial we will learn how to make
forecasts and investigate some of the additional inference options
available in <code>EpiNow2</code>.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Transmission metrics can be estimated from case data after
accounting for delays</li>
<li>Uncertainty can be accounted for in delay distributions</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-create-forecast"><p>Content from <a href="create-forecast.html">Create a short-term forecast</a></p>
<hr>
<p>Last updated on 2024-03-25 |
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/episodes/create-forecast.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I create short-term forecasts from case data?</li>
<li>How do I account for incomplete reporting in forecasts?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to make forecasts of cases using R package
<code>EpiNow2</code>
</li>
<li>Learn how to include an observation process in estimation</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<ul>
<li>Complete tutorial <a href="quantify-transmissibility.html">Quantifying transmission</a>
</li>
</ul>
<p>Learners should familiarise themselves with following concept
dependencies before working through this tutorial:</p>
<p><strong>Statistics</strong> : probability distributions, principle of
Bayesian analysis.</p>
<p><strong>Epidemic theory</strong> : Effective reproduction number.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Given case data, we can create estimates of the current and future
number of cases by accounting for both delays in reporting and under
reporting. To make statements about the future, we need to make an
assumption of how observations up to today are related to what we expect
to happen in the future. The simplest way of doing so is to assume “no
change”, i.e. the reproduction number remains the same in the future as
last observed. In this tutorial we will create short-term forecasts by
assuming the reproduction number will remain the same as it was
estimated to be on the final date for which data was available.</p>
</section><section id="create-a-short-term-forecast"><h2 class="section-heading">Create a short-term forecast<a class="anchor" aria-label="anchor" href="#create-a-short-term-forecast"></a>
</h2>
<hr class="half-width">
<p>The function <code>epinow()</code> described in the previous tutorial
is a wrapper for the function <code>estimate_infections()</code> used to
estimate cases by date of infection. Using the same code in the previous
tutorial we can extract the short-term forecast using:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reported_cases</span> <span class="op">&lt;-</span> <span class="va">cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span>, <span class="op">]</span></span>
<span><span class="va">estimates</span> <span class="op">&lt;-</span> <span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">reported_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-03-25 20:01:41] epinow: There were 1 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-03-25 20:01:41] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<p>We can visualise the estimates of the effective reproduction number
and the estimated number of cases using <code>plot()</code>. The
estimates are split into three categories:</p>
<ul>
<li><p><strong>Estimate</strong> (green): utilises all data,</p></li>
<li><p><strong>Estimate based on partial data</strong> (orange):
estimates that are based less data are therefore more
uncertain,</p></li>
<li><p><strong>Forecast</strong> (purple): forecasts into the
future.</p></li>
</ul>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">estimates</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="forecasting-with-estimates-of-r_t" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="forecasting-with-estimates-of-r_t" class="callout-inner">
<h3 class="callout-title">Forecasting with estimates of<span class="math inline">\(R_t\)</span><a class="anchor" aria-label="anchor" href="#forecasting-with-estimates-of-r_t"></a>
</h3>
<div class="callout-content">
<p>By default, the short-term forecasts are created using the latest
estimate of the reproduction number <span class="math inline">\(R_t\)</span>. As this estimate is based on partial
data, it has considerable uncertainty.</p>
<p>The reproduction number that is projected into the future can be
changed to a less recent estimate based on more data using
<code>rt_opts()</code>:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rt_opts</span><span class="op">(</span>future <span class="op">=</span> <span class="st">"estimate"</span><span class="op">)</span></span></code></pre>
</div>
<p>The result will be less uncertain forecasts (as they are based on
<span class="math inline">\(R_t\)</span> with a narrower uncertainty
interval) but the forecasts will be based on less recent estimates of
<span class="math inline">\(R_t\)</span> and assume no change since
then.</p>
<p>Additionally, there is the option to project the value of <span class="math inline">\(R_t\)</span> into the future using a generic model
by setting <code>future = "project"</code>. As this option uses a model
to forecast the value of <span class="math inline">\(R_t\)</span>, the
result will be forecasts that are more uncertain than
<code>estimate</code>, for an example <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections_options.html#projecting-the-reproduction-number-with-the-gaussian-process" class="external-link">see
here</a>.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="incomplete-observations">Incomplete observations<a class="anchor" aria-label="anchor" href="#incomplete-observations"></a>
</h3>
<p>In the previous tutorial we accounted for delays in reporting. In
<code>EpiNow2</code> we also can account for incomplete observations as
in reality, 100% of cases are not reported.</p>
<p>We will pass another input into <code>epinow()</code> called
<code>obs</code> to define an observation model. The format of
<code>obs</code> must be the <code>obs_opt()</code> function (see
<code>?EpiNow2::obs_opts</code> for more detail).</p>
<p>Let’s say we believe the COVID-19 outbreak data from the previous
tutorial do not include all reported cases. We believe that we only
observe 40% of cases. To specify this in the observation model, we must
pass a scaling factor with a mean and standard deviation. If we assume
that 40% of cases are in the case data (with standard deviation 1%),
then we specify the <code>scale</code> input to <code>obs_opts()</code>
as follows:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obs_scale</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.4</span>, sd <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<p>To run the inference framework with this observation process, we add
<code>obs = obs_opts(scale = obs_scale)</code> to the input arguments of
<code>epinow()</code>:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obs_scale</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.4</span>, sd <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">reported_cases</span> <span class="op">&lt;-</span> <span class="va">cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span>, <span class="op">]</span></span>
<span><span class="va">estimates</span> <span class="op">&lt;-</span> <span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">reported_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  obs <span class="op">=</span> <span class="fu">obs_opts</span><span class="op">(</span>scale <span class="op">=</span> <span class="va">obs_scale</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-03-25 20:08:12] epinow: There were 4 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-03-25 20:08:12] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">estimates</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 measure                 estimate
                                  &lt;char&gt;                   &lt;char&gt;
1: New confirmed cases by infection date   17836 (10068 -- 30847)
2:        Expected change in daily cases        Likely decreasing
3:            Effective reproduction no.       0.88 (0.56 -- 1.3)
4:                        Rate of growth -0.016 (-0.065 -- 0.037)
5:          Doubling/halving time (days)          -43 (19 -- -11)</code></pre>
</div>
<p>The estimates of transmission measures such as the effective
reproduction number and rate of growth are similar (or the same in
value) compared to when we didn’t account for incomplete observations
(see previous tutorial). However the number of new confirmed cases by
infection date has changed substantially in magnitude to reflect the
assumption that only 40% of cases are in the data set.</p>
<p>We can also change the default distribution from Negative Binomial to
Poisson, remove the default week effect and more. See
<code>?EpiNow2::obs_opts</code> for more detail.</p>
</div>
</section><section id="forecast-secondary-observations"><h2 class="section-heading">Forecast secondary observations<a class="anchor" aria-label="anchor" href="#forecast-secondary-observations"></a>
</h2>
<hr class="half-width">
<p><code>EpiNow2</code> also has the ability to estimate and forecast
secondary observations e.g. deaths, hospitalisations from a primary
observation e.g. cases. Here we will illustrate how to forecast the
number of deaths arising from observed cases of COVID-19 in the early
stages of the UK outbreak.</p>
<p>First, we must format our data to have the following columns:</p>
<ul>
<li>
<code>date</code> : the date (as a date object see
<code>?is.Date()</code>),</li>
<li>
<code>primary</code> : number of primary observations on that date,
in this example <strong>cases</strong>,</li>
<li>
<code>secondary</code> : number of secondary observations date, in
this example <strong>deaths</strong>.</li>
</ul>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reported_cases_deaths</span> <span class="op">&lt;-</span> <span class="fu">aggregate</span><span class="op">(</span></span>
<span>  <span class="fu">cbind</span><span class="op">(</span><span class="va">cases_new</span>, <span class="va">deaths_new</span><span class="op">)</span> <span class="op">~</span> <span class="va">date</span>,</span>
<span>  data <span class="op">=</span></span>
<span>    <span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"cases_new"</span>, <span class="st">"deaths_new"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  FUN <span class="op">=</span> <span class="va">sum</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">reported_cases_deaths</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"primary"</span>, <span class="st">"secondary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Using the first 30 days of data on cases and deaths, we will estimate
the relationship between the primary and secondary observations using
<code>estimate_secondary()</code>, then forecast future deaths using
<code>forecast_secondary()</code>. For detail on the model see the <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_secondary.html" class="external-link">model
documentation</a>.</p>
<p>We must specify the type of observation using <code>type</code> in
<code>secondary_opts()</code>, options include:</p>
<ul>
<li>“incidence” : secondary observations arise from previous primary
observations, i.e. deaths arising from recorded cases.</li>
<li>“prevalence” : secondary observations arise from a combination
current primary observations and past secondary observations,
i.e. hospital bed usage arising from current hospital admissions and
past hospital bed usage.</li>
</ul>
<p>In this example we specify
<code>secondary_opts(type = "incidence")</code>. See
<code>?EpiNow2::secondary_opts</code> for more detail).</p>
<p>The final key input is the delay distribution between the primary and
secondary observations. Here this is the delay between case report and
death, we assume this follows a gamma distribution with mean of 14 days
and standard deviation of 5 days. Using <code>dist_spec()</code> we
specify a fixed gamma distribution.</p>
<p>There are further function inputs to
<code>estimate_secondary()</code> which can be specified, including
adding an observation process, see
<code>?EpiNow2::estimate_secondary</code> for detail on the options.</p>
<p>To find the model fit between cases and deaths :</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimate_cases_to_deaths</span> <span class="op">&lt;-</span> <span class="fu">estimate_secondary</span><span class="op">(</span></span>
<span>  reports <span class="op">=</span> <span class="va">reported_cases_deaths</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span>,</span>
<span>  secondary <span class="op">=</span> <span class="fu">secondary_opts</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"incidence"</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>    mean <span class="op">=</span> <span class="fl">14</span>, sd <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    max <span class="op">=</span> <span class="fl">30</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div id="be-cautious-of-timescale" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="be-cautious-of-timescale" class="callout-inner">
<h3 class="callout-title">Be cautious of timescale<a class="anchor" aria-label="anchor" href="#be-cautious-of-timescale"></a>
</h3>
<div class="callout-content">
<p>In the early stages of an outbreak there can be substantial changes
in testing and reporting. If there are testing changes from one month to
another, then there will be a bias in the model fit. Therefore, you
should be cautious of the time-scale of data used in the model fit and
forecast.</p>
</div>
</div>
</div>
<p>We plot the model fit (shaded ribbons) with the secondary
observations (bar plot) and primary observations (dotted line) as
follows:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">estimate_cases_to_deaths</span>, primary <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To use this model fit to forecast deaths, we pass a data frame
consisting of the primary observation (cases) for dates not used in the
model fit.</p>
<p><em>Note : in this tutorial we are using data where we know the
deaths and cases, so we create a data frame by extracting the cases. But
in practice, this would be a different data set consisting of cases
only.</em></p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cases_to_forecast</span> <span class="op">&lt;-</span> <span class="va">reported_cases_deaths</span><span class="op">[</span><span class="fl">31</span><span class="op">:</span><span class="fl">60</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"primary"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">cases_to_forecast</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<p>To forecast, we use the model fit
<code>estimate_cases_to_deaths</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deaths_forecast</span> <span class="op">&lt;-</span> <span class="fu">forecast_secondary</span><span class="op">(</span></span>
<span>  estimate <span class="op">=</span> <span class="va">estimate_cases_to_deaths</span>,</span>
<span>  primary <span class="op">=</span> <span class="va">cases_to_forecast</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">deaths_forecast</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Removed 30 rows containing missing values (`position_stack()`).</code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot shows the forecast secondary observations (deaths) over the
dates which we have recorded cases for. It is also possible to forecast
deaths using forecast cases, here you would specify <code>primary</code>
as the <code>estimates</code> output from
<code>estimate_infections()</code>.</p>
</section><section id="challenge-ebola-outbreak-analysis"><h2 class="section-heading">Challenge : Ebola outbreak analysis<a class="anchor" aria-label="anchor" href="#challenge-ebola-outbreak-analysis"></a>
</h2>
<hr class="half-width">
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Download the file <a href="../data/ebola_cases.csv">ebola_cases.csv</a>
and read it into R. The simulated data consists of the date of symptom
onset and number of confirmed cases of the early stages of the Ebola
outbreak in Sierra Leone in 2014.</p>
<p>Using the first 3 months (120 days) of data:</p>
<ol style="list-style-type: decimal">
<li>Estimate of cases increasing or decreasing on day 120 of the
outbreak (Hint: Find the effective reproduction number and growth rate
on day 120)</li>
<li>Create a two week forecast of number of cases</li>
</ol>
<p>You can use the following parameter values for the delay
distribution(s) and generation time distribution.</p>
<ul>
<li>Incubation period : Log normal<span class="math inline">\((2.487,0.330)\)</span> (<a href="https://doi.org/10.1016/j.phrp.2011.04.001" class="external-link">Eichner et
al. 2011</a> via <code>{epiparameter}</code>)</li>
<li>Generation time : Gamma<span class="math inline">\((15.3,
10.1)\)</span> (<a href="https://www.nejm.org/doi/full/10.1056/NEJMoa1411100" class="external-link">WHO Ebola
Response Team 2014</a>)</li>
</ul>
<p>You may include some uncertainty around the mean and standard
deviation of these distributions.</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">HINT : data format</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" aria-labelledby="headingHint1" data-bs-parent="#accordionHint1">
<div class="accordion-body">
<p>Ensure the data is in the correct format :</p>
<ul>
<li>
<code>date</code> : the date (as a date object see
<code>?is.Date()</code>),</li>
<li>
<code>confirm</code> : number of confirmed cases on that date.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">SOLUTION</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>To estimate the effective reproduction number and growth rate, we
will use the function <code>epinow()</code>.</p>
<p>As the data consists of date of symptom onset, we only need to
specify a delay distribution for the incubation period and the
generation time.</p>
<p>We specify the distributions with some uncertainty around the mean
and standard deviation of the log normal distribution for the incubation
period and the Gamma distribution for the generation time.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ebola_incubation_period</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">2.487</span>, sd <span class="op">=</span> <span class="fl">0.330</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_generation_time</span> <span class="op">&lt;-</span> <span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">15.3</span>, sd <span class="op">=</span> <span class="fl">10.1</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">30</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>As we want to also create a two week forecast, we specify
<code>horizon = 14</code> to forecast 14 days instead of the default 7
days.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co"># e.g.: if path to file is data/raw-data/ebola_cases.csv then:</span></span>
<span><span class="va">ebola_cases</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read.csv</span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"ebola_cases.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># format date column</span></span>
<span><span class="va">ebola_cases</span><span class="op">$</span><span class="va">date</span> <span class="op">&lt;-</span> <span class="fu">as.Date</span><span class="op">(</span><span class="va">ebola_cases</span><span class="op">$</span><span class="va">date</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_estimates</span> <span class="op">&lt;-</span> <span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">ebola_cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">120</span>, <span class="op">]</span>, <span class="co"># first 3 months of data only</span></span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">ebola_generation_time</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">ebola_incubation_period</span><span class="op">)</span>,</span>
<span>  <span class="co"># horizon needs to be 14 days to create two week forecast (default is 7 days)</span></span>
<span>  horizon <span class="op">=</span> <span class="fl">14</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-03-25 20:10:14] epinow: There were 17 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-03-25 20:10:14] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">ebola_estimates</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 measure                estimate
                                  &lt;char&gt;                  &lt;char&gt;
1: New confirmed cases by infection date         103 (46 -- 258)
2:        Expected change in daily cases              Increasing
3:            Effective reproduction no.          1.7 (1 -- 2.9)
4:                        Rate of growth 0.043 (0.0024 -- 0.092)
5:          Doubling/halving time (days)         16 (7.6 -- 290)</code></pre>
</div>
<p>The effective reproduction number <span class="math inline">\(R_t\)</span> estimate (on the last date of the
data) is 1.7 (1 – 2.9). The exponential growth rate of case numbers is
0.043 (0.0024 – 0.092).</p>
<p>Visualize the estimates:</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">ebola_estimates</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p><code>EpiNow2</code> can be used to create short term forecasts and
to estimate the relationship between different outcomes. There are a
range of model options that can be implemented for different analysis,
including adding an observational process to account for incomplete
reporting. See the <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections_options.html" class="external-link">vignette</a>
for more details on different model options in <code>EpiNow2</code> that
aren’t covered in these tutorials.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>We can create short-term forecasts by making assumptions about the
future behaviour of the reproduction number</li>
<li>Incomplete case reporting can be accounted for in estimates</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-severity-static"><p>Content from <a href="severity-static.html">Estimate severity</a></p>
<hr>
<p>Last updated on 2024-03-25 |
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/episodes/severity-static.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li><p>Why do we estimate the clinical severity of an epidemic?</p></li>
<li><p>How can the Case Fatality Risk (CFR) be estimated early in an
ongoing epidemic?</p></li>
<li><p>How do we estimate other severity measurements from different
data sources?</p></li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li><p>Estimate the CFR from aggregated case data using
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate a delay-adjusted CFR using <code>{epiparameter}</code>
and <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate severity measurements like the Hospitalization Fatality
Risk (HFR) using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>This episode requires you to be familiar with:</p>
<p><strong>Data science</strong> : Basic programming with R.</p>
<p><strong>Epidemic theory</strong> : <a href="reference.html#delaydist">Delay distributions</a>.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Common questions at the early stage of an epidemic include:</p>
<ul>
<li>What is the likely public health impact of the outbreak in terms of
clinical severity?</li>
<li>What are the most severely affected groups?</li>
<li>Does the outbreak have the potential to cause a very severe
pandemic?</li>
</ul>
<p>We can assess the pandemic potential of an epidemic with two critical
measurements: the transmissibility and the clinical severity. (<a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a>, <a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>)</p>
<figure><img src="../fig/cfr-hhs-scenarios-psaf.png" alt="The horizontal axis is the scaled measure of clinical severity, ranging from 1 to 7, where 1 is low, 4 is moderate, and 7 is very severe. The vertical axis is the scaled measure of transmissibility, ranging from 1 to 5, where 1 is low, 3 is moderate, and 5 is highly transmissible. On the graph, HHS pandemic planning scenarios are labeled across four quadrants (A, B, C and D). From left to right, the scenarios are “seasonal range,” “moderate pandemic,” “severe pandemic” and “very severe pandemic.” As clinical severity increases along the horizontal axis, or as transmissibility increases along the vertical axis, the severity of the pandemic planning scenario also increases." class="figure mx-auto d-block"><div class="figcaption">HHS Pandemic Planning Scenarios based on the
Pandemic Severity Assessment Framework. This uses a combined measure of
clinical severity and transmissibility to characterise influenza
pandemic scenarios. HHS: United States Department of Health and Human
Services (<a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>).</div>
</figure><p>One epidemiological approach to estimating the clinical severity is
quantifying the Case Fatality Risk or CFR. CFR is the conditional
probability of death given confirmed diagnosis, calculated as the
cumulative number deaths from an infectious disease over the number of
confirmed diagnosed cases. However, calculating this directly during the
course of an epidemic tends to result in a naive or biased CFR given the
time <a href="reference.html#delaydist">delay</a> from onset to death,
varying substantially as the epidemic progresses and stabilising at the
later stages of the outbreak (<a href="https://academic.oup.com/aje/article/162/5/479/82647?login=false#620743" class="external-link">Ghani
et al., 2005</a>).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_c.png" alt="The periods are relevant: Period 1 -- 15 days where CFR is zero to indicate this is due to no reported deaths; Period from Mar 15 -- Apr 26 where CFR appears to be rising; Period Apr 30 -- May 30 where the CFR estimate stabilises." class="figure mx-auto d-block"><div class="figcaption">Observed biased confirmed case fatality risk
(CFR) estimates as a function of time (thick line) calculated as the
cumulative number of deaths over confirmed cases at time t. The estimate
at the end of an outbreak (~May 30) is the realised CFR by the end of
the epidemic. The horizontal continuous line and dotted lines show the
expected value and the 95% confidence intervals (95% CI) of the
predicted delay-adjusted CFR estimate only by using the observed data
until 27 Mar 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>The periods are relevant: Period 1 – 15 days where CFR is zero to
indicate this is due to no reported deaths; Period from Mar 15 – Apr 26
where CFR appears to be rising; Period Apr 30 – May 30 where the CFR
estimate stabilises.</p>
</div>
</div>
</div>
</div>
<p>More generally, estimating severity can be helpful even outside of a
pandemic planning scenario and in the context of routine public health.
Knowing whether an outbreak has or had a different severity from the
historical record can motivate causal investigations, which could be
intrinsic to the infectious agent (e.g. a new, more severe strain) or
due to underlying factors in the population (e.g. reduced immunity or
morbidity factors) (<a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a>).</p>
<p>In this tutorial we are going to learn how to use the
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package to calculate and adjust a CFR estimation
using <a href="reference.html#delaydist">delay distributions</a> from
<code>{epiparameter}</code> or elsewhere, based on the methods developed
by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>, also, how we can reuse <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> functions
for more severity measurements.</p>
<p>We’ll use the pipe <code>%&gt;%</code> to connect some of their
functions, so let’s also call to the <a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a>
package:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span></code></pre>
</div>
</section><section id="data-sources-for-clinical-severity"><h2 class="section-heading">Data sources for clinical severity<a class="anchor" aria-label="anchor" href="#data-sources-for-clinical-severity"></a>
</h2>
<hr class="half-width">
<p>What are data sources can we use to estimate the clinical severity of
a disease outbreak? <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext" class="external-link">Verity
et al., 2020</a> summaries the spectrum of COVID-19 cases:</p>
<figure><img src="../fig/cfr-spectrum-cases-covid19.jpg" alt="Spectrum of COVID-19 cases. The CFR aims to estimate the proportion of Deaths among confirmed cases in an epidemic. (Verity et al., 2020)" class="figure mx-auto d-block"><div class="figcaption">Spectrum of COVID-19 cases. The CFR aims to
estimate the proportion of Deaths among confirmed cases in an epidemic.
(<a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext#gr1" class="external-link">Verity
et al., 2020</a>)</div>
</figure><ul>
<li>At the top of the pyramid, those meeting the WHO case criteria for
<strong>severe</strong> or critical cases will likely be identified in
the hospital setting, presenting with atypical viral pneumonia. These
cases will have been identified in mainland China and among those
categorised internationally as local transmission.</li>
<li>Many more cases are likely to be <strong>symptomatic</strong> (i.e.,
with fever, cough, or myalgia) but might not require hospitalisation.
These cases will have been identified through links to international
travel to high-risk areas and through contact-tracing of contacts of
confirmed cases. They might also be identified through population
surveillance of, for example, influenza-like illness.</li>
<li>The bottom part of the pyramid represents <strong>mild</strong> (and
possibly <strong>asymptomatic</strong>) cases. These cases might be
identified through contact tracing and subsequently via serological
testing.</li>
</ul></section><section id="naive-cfr"><h2 class="section-heading">Naive CFR<a class="anchor" aria-label="anchor" href="#naive-cfr"></a>
</h2>
<hr class="half-width">
<p>We measure disease severity in terms of case fatality risk (CFR). The
CFR is interpreted as the conditional probability of death given
confirmed diagnosis, calculated as the cumulative number of deaths <span class="math inline">\(D_{t}\)</span> over the cumulative number of
confirmed cases <span class="math inline">\(C_{t}\)</span> at a certain
time <span class="math inline">\(t\)</span>. We can refer to the
<em>naive CFR</em> (also crude or biased CFR, <span class="math inline">\(b_{t}\)</span>):</p>
<p><span class="math display">\[ b_{t} =  \frac{D_{t}}{C_{t}}
\]</span></p>
<p>This calculation is <em>naive</em> because it tends to yield a biased
and mostly underestimated CFR due to the time-delay from onset to death,
only stabilising at the later stages of the outbreak.</p>
<!-- add here the callout on ratio or risk?  -->
<!-- https://github.com/epiverse-trace/cfr/issues/130 -->
<p>To calculate the naive CFR, the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package requires
an input data frame with three columns named:</p>
<ul>
<li><code>date</code></li>
<li><code>cases</code></li>
<li><code>deaths</code></li>
</ul>
<p>Let’s explore the <code>ebola1976</code> dataset, which comes from
the first Ebola outbreak in what was then called Zaire (now the
Democratic Republic of the Congo) in 1976, as analysed by Camacho et
al. (2014).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Ebola 1976 data provided with the package</span></span>
<span><span class="fu">data</span><span class="op">(</span><span class="st">"ebola1976"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># view top data</span></span>
<span><span class="va">ebola1976</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 73 × 3
   date       cases deaths
   &lt;date&gt;     &lt;int&gt;  &lt;int&gt;
 1 1976-08-25     1      0
 2 1976-08-26     0      0
 3 1976-08-27     0      0
 4 1976-08-28     0      0
 5 1976-08-29     0      0
 6 1976-08-30     0      0
 7 1976-08-31     0      0
 8 1976-09-01     1      0
 9 1976-09-02     1      0
10 1976-09-03     1      0
# ℹ 63 more rows</code></pre>
</div>
<div id="we-need-aggregated-incidence-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="we-need-aggregated-incidence-data" class="callout-inner">
<h3 class="callout-title">We need aggregated incidence data<a class="anchor" aria-label="anchor" href="#we-need-aggregated-incidence-data"></a>
</h3>
<div class="callout-content">
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> reads <strong>aggregated</strong> incidence data.
Similar to the <a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> with the difference that for
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> we need one more column named
<code>deaths</code>.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EpiNow2</span><span class="fu">::</span><span class="va">example_confirmed</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 130 × 2
   date       confirm
   &lt;date&gt;       &lt;dbl&gt;
 1 2020-02-22      14
 2 2020-02-23      62
 3 2020-02-24      53
 4 2020-02-25      97
 5 2020-02-26      93
 6 2020-02-27      78
 7 2020-02-28     250
 8 2020-02-29     238
 9 2020-03-01     240
10 2020-03-02     561
# ℹ 120 more rows</code></pre>
</div>
<p>This data input should be <strong>aggregated</strong> by day, which
means one observation <em>per day</em>, containing the <em>daily</em>
number of reported cases and deaths. Observations with zero or missing
values should also be included, similar to a time-series data.</p>
<p>Also, <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> currently works for <em>daily</em> data
only, not weekly data.</p>
<!-- suggest ways to deal with raw input weekly data -->
<!-- https://github.com/epiverse-trace/cfr/issues/117 -->
<div class="section level3">
<h3 id="aggregated-differs-from-linelist">Aggregated differs from Linelist<a class="anchor" aria-label="anchor" href="#aggregated-differs-from-linelist"></a>
</h3>
<p>Lastly, <em>aggregated</em> incidence data differs from
<strong>linelist</strong> data, where each observation contains
subject-level data.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_sierraleone_2014</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 11,903 × 8
      id   age sex   status    date_of_onset date_of_sample district chiefdom   
   &lt;int&gt; &lt;dbl&gt; &lt;fct&gt; &lt;fct&gt;     &lt;date&gt;        &lt;date&gt;         &lt;fct&gt;    &lt;fct&gt;      
 1     1    20 F     confirmed 2014-05-18    2014-05-23     Kailahun Kissi Teng 
 2     2    42 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Teng 
 3     3    45 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Tonge
 4     4    15 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 5     5    19 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 6     6    55 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 7     7    50 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 8     8     8 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
 9     9    54 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
10    10    57 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
# ℹ 11,893 more rows</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>When we use <code>cfr_static()</code> to <code>data</code> only, we
calculate the naive CFR:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the static CFR without adjusting for delays</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1      0.955102    0.9210866     0.9773771</code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Load the <code>ebola_kikwit_1995</code> from the
<a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package and estimate the naive CFR.</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" data-bs-parent="#accordionHint1" aria-labelledby="headingHint1">
<div class="accordion-body">
<p>Inspect the format of the data input.</p>
<ul>
<li>Does it contain daily data?</li>
<li>Does the column names are as required by
<code>cfr_static()</code>?</li>
<li>How would you rename column names from a data frame?</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>We can use <code>dplyr::rename()</code> to adapt the external data to
fit the data input for <code>cfr_static()</code>.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 192
Columns: 4
$ date      &lt;date&gt; 1995-01-06, 1995-01-07, 1995-01-08, 1995-01-09, 1995-01-10,…
$ onset     &lt;int&gt; 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ death     &lt;int&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ reporting &lt;lgl&gt; TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">onset</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">death</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1     0.8082192    0.7583033     0.8517565</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="biases-that-affect-cfr-estimation"><h2 class="section-heading">Biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#biases-that-affect-cfr-estimation"></a>
</h2>
<hr class="half-width">
<div id="two-biases-that-affect-cfr-estimation" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="two-biases-that-affect-cfr-estimation" class="callout-inner">
<h3 class="callout-title">Two biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#two-biases-that-affect-cfr-estimation"></a>
</h3>
<div class="callout-content">
<p><a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a> describe two potential biases that can affect the
estimation of CFR (and its possible solutions):</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Preferential ascertainment of severe cases</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>For diseases with a <em>spectrum</em> of clinical presentation, those
cases that come to the attention of public health authorities and
registered into surveillance databases will typically be people with the
most severe symptoms who seek medical care, are admitted to a hospital,
or die.</p>
<p>Therefore, the CFR will typically be higher among <em>detected
cases</em> than among the entire population of cases, given that the
latter may include individuals with mild, subclinical, and (under some
definitions of “case”) asymptomatic presentations.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Bias due to delayed reporting of death</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>During an <em>ongoing</em> epidemic, there is a delay between the
time someone dies and the time their death is reported. Therefore, at
any moment in time, the list of cases includes people who will die and
whose death has not yet occurred or has occurred but not yet been
reported. Thus, dividing the cumulative number of reported deaths by the
cumulative number of reported cases at any moment will underestimate the
true CFR.</p>
<p>The key determinants of the magnitude of the bias are the epidemic
<em>growth rate</em> and the <em>distribution of delays</em> from
case-reporting to death-reporting; the longer the delays and the faster
the growth rate, the greater the bias.</p>
<p>In this tutorial episode, we are going to focus on solutions to deal
with this specific bias using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>!</p>
</div>
</div>
</div>
</div>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor2" aria-labelledby="headingInstructor2">
<div class="accordion-body">
<p>We can showcase this last bias using the <a href="https://epiverse-trace.github.io/cfr/articles/cfr.html#concept-how-reporting-delays-bias-cfr-estimates" class="external-link">concept
described in this <code>{cfr}</code> vignette</a>.</p>
<!-- create code and then a .gif? -->
</div>
</div>
</div>
</div>
<div id="biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div id="biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009" class="callout-inner">
<h3 class="callout-title">Biases during the Influenza A (H1N1) outbreak
in Mexico, 2009<a class="anchor" aria-label="anchor" href="#biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009"></a>
</h3>
<div class="callout-content">
<p>Improving an <em>early</em> epidemiological assessment of a
delay-adjusted CFR is crucial for determining virulence, shaping the
level and choices of public health intervention, and providing advice to
the general public.</p>
<p>In 2009, during the swine-flu virus, Influenza A (H1N1), Mexico had
an early biased estimation of the CFR. Initial reports from the
government of Mexico suggested a virulent infection, whereas, in other
countries, the same virus was perceived as mild (<a href="https://content.time.com/time/health/article/0,8599,1894534,00.html" class="external-link">TIME,
2009</a>).</p>
<p>In the USA and Canada, no deaths were attributed to the virus in the
first ten days following the World Health Organization’s declaration of
a public health emergency. Even under similar circumstances at the early
stage of the global pandemic, public health officials, policymakers and
the general public want to know the virulence of an emerging infectious
agent.</p>
<p><a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a> reinterpreted the data assessing the biases and getting
a clinical severity lower than the 1918 influenza pandemic but
comparable with that seen in the 1957 pandemic.</p>
</div>
</div>
</div>
<div id="discussion2" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion2"></a>
</h3>
<div class="callout-content">
<p>Based on your experience:</p>
<ul>
<li>Share any previous outbreak in which you participated in its
response.</li>
</ul>
<p>Answer to these questions:</p>
<ul>
<li>How did you assess the clinical severity of the outbreak?</li>
<li>What were the primary sources of bias?</li>
<li>What did you do to take into account the identified bias?</li>
<li>What complementary analysis would you do to solve the bias?</li>
</ul>
</div>
</div>
</div>
</section><section id="delay-adjusted-cfr"><h2 class="section-heading">Delay-adjusted CFR<a class="anchor" aria-label="anchor" href="#delay-adjusted-cfr"></a>
</h2>
<hr class="half-width">
<p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> developed a method that considers the <strong>time
delay</strong> from the onset of symptoms to death.</p>
<p>Real-time outbreaks may have a number of deaths that are insufficient
to determine the time distribution between onset and death. Therefore,
we can estimate the <em>distribution delay</em> from historical
outbreaks or reuse the ones accessible via R packages like
<code>{epiparameter}</code> or <code>{epireview}</code>, which collects
them from literature searches and systematic reviews.</p>
<p>Let’s use <code>{epiparameter}</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get delay distribution</span></span>
<span><span class="va">onset_to_death_ebola</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"Ebola"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset_to_death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot &lt;epidist&gt; object</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, day_range <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To calculate the delay-adjusted CFR, we can use the
<code>cfr_static()</code> function with the <code>data</code> and
<code>delay_density</code> arguments.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the static CFR while adjusting for delays</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.971        0.852             1</code></pre>
</div>
<p>The delay-adjusted CFR indicated that the overall disease severity
<em>at the end of the outbreak</em> or with the <em>latest data
available at the moment</em> is 0.971 with a 95% confidence interval
between 0.852 and 1, slightly higher than the naive one.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>How does {cfr} works?</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>To adjust the CFR, <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> use the case and death incidence data to estimate the
number of cases with known outcomes:</p>
<p><span class="math display">\[
  u_t = \dfrac{\sum_{i = 0}^t
        \sum_{j = 0}^\infty c_{i - j} f_{j}}{\sum_{i = 0} c_i},
\]</span></p>
<p>where:</p>
<ul>
<li>
<span class="math inline">\(c_{t}\)</span> is the daily case
incidence at time <span class="math inline">\(t\)</span>,</li>
<li>
<span class="math inline">\(f_{t}\)</span> is the value of the
Probability Mass Function (PMF) of the <strong>delay
distribution</strong> between onset and death, and</li>
<li>
<span class="math inline">\(u_{t}\)</span> represents the
underestimation factor of the known outcomes.</li>
</ul>
<p><span class="math inline">\(u_{t}\)</span> is used to
<strong>scale</strong> the value of the cumulative number of cases in
the denominator in the calculation of the CFR. This is calculated
internally with the <a href="https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html" class="external-link"><code>estimate_outcomes()</code></a>
function.</p>
<p>The estimator for CFR can be written as:</p>
<p><span class="math display">\[p_{t} = \frac{b_{t}}{u_{t}}\]</span></p>
<p>where <span class="math inline">\(p_{t}\)</span> is the realized
proportion of confirmed cases to die from the infection (or the unbiased
CFR), and <span class="math inline">\(b_{t}\)</span>, the crude and
biased estimate of CFR (also naive CFR).</p>
<p>From this last equation, we observe that the unbiased CFR <span class="math inline">\(p_{t}\)</span> is larger than biased CFR <span class="math inline">\(b_{t}\)</span> because in <span class="math inline">\(u_{t}\)</span> the numerator is smaller than the
denominator (note that <span class="math inline">\(f_{t}\)</span> is the
probability distribution of the <em>delay distribution</em> between
onset and death). Therefore, we refer to <span class="math inline">\(b_{t}\)</span> as the biased estimator of CFR.</p>
<p>When we observe the entire course of an epidemic (from <span class="math inline">\(t \rightarrow \infty\)</span>), <span class="math inline">\(u_{t}\)</span> tends to 1, making <span class="math inline">\(b_{t}\)</span> tends to <span class="math inline">\(p_{t}\)</span> and become an unbiased estimator
(<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>).</p>
</div>
</div>
</div>
</div>
<div id="use-the-epidist-class" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="use-the-epidist-class" class="callout-inner">
<h3 class="callout-title">Use the<code>&lt;epidist&gt;</code>class<a class="anchor" aria-label="anchor" href="#use-the-epidist-class"></a>
</h3>
<div class="callout-content">
<p>When using an <code>&lt;epidist&gt;</code> class object we can use
this expression as a template:</p>
<p><code>function(x) density(&lt;EPIDIST_OBJECT&gt;, x)</code></p>
<p>For distribution functions with parameters not available in
<code>{epiparameter}</code>, we suggest you two alternatives:</p>
<ul>
<li><p>Create an <code>&lt;epidist&gt;</code> class object, to plug into
other R packages of the outbreak analytics pipeline. Read the <a href="https://epiverse-trace.github.io/epiparameter/reference/epidist.html" class="external-link">reference
documentation of <code>epiparameter::epidist()</code></a>, or</p></li>
<li><p>Read <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette for <a href="https://epiverse-trace.github.io/cfr/articles/delay_distributions.html" class="external-link">a
primer on working with delay distributions</a>.</p></li>
</ul>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>When to use discrete distributions?</h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler2" aria-labelledby="headingSpoiler2">
<div class="accordion-body">
<p>For <code>cfr_static()</code> and all the <code>cfr_*()</code> family
of functions, the most appropriate choice to pass are
<strong>discrete</strong> distributions. This is because we will work
with daily case and death data.</p>
<p>We can assume that evaluating the Probability Distribution Function
(PDF) of a <em>continuous</em> distribution is equivalent to the
Probability Mass Function (PMF) of the equivalent <em>discrete</em>
distribution.</p>
<p>However, this assumption may not be appropriate for distributions
with greater picks. For example, when onset-to-death is strongly peaked
with a low variance. In such cases, the average disparity between the
PDF and PMF is expected to be more pronounced compared to distributions
with broader spreads.</p>
<p>In a scenario like the last one, you can opt to discretise the
continuous distribution using <code>epiparameter::discretise()</code> to
an <code>&lt;epidist&gt;</code> object.</p>
</div>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge2"></a>
</h3>
<div class="callout-content">
<p>Load the <code>ebola_kikwit_1995</code> from <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a>
and estimate the delay-adjusted CFR using the appropriate distribution
delay. Then:</p>
<ul>
<li>Compare the naive and the delay-adjusted CFR solutions!</li>
</ul>
</div>
</div>
</div>
<div id="accordionHint2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint2" aria-expanded="false" aria-controls="collapseHint2">
  <h4 class="accordion-header" id="headingHint2">Give me a hint</h4>
</button>
<div id="collapseHint2" class="accordion-collapse collapse" data-bs-parent="#accordionHint2" aria-labelledby="headingHint2">
<div class="accordion-body">
<ul>
<li>Find the appropriate <code>&lt;epidist&gt;</code> object!</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 192
Columns: 4
$ date      &lt;date&gt; 1995-01-06, 1995-01-07, 1995-01-08, 1995-01-09, 1995-01-10,…
$ onset     &lt;int&gt; 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ death     &lt;int&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ reporting &lt;lgl&gt; TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">onset</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">death</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>    delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.819         0.72         0.928</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="an-early-stage-cfr-estimate"><h2 class="section-heading">An early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#an-early-stage-cfr-estimate"></a>
</h2>
<hr class="half-width">
<p>We just found that the difference between the <em>naive</em> and
<em>delay-adjusted</em> CFR estimates using <code>cfr_static()</code>
are <em>just slightly</em> different. So, what is the main
difference?</p>
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> method aims to obtain an unbiased estimator “well
before” observing the entire course of the outbreak. For this
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> uses the underestimation factor <span class="math inline">\(u_{t}\)</span> to estimate the unbiased CFR <span class="math inline">\(p_{t}\)</span> using maximum-likelihood methods,
given the sampling process defined by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>.</p>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>What is the sampling process?</h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler3" aria-labelledby="headingSpoiler3">
<div class="accordion-body">
<figure><img src="../fig/cfr-pone.0006852.g001.png" alt="The population of confirmed cases and sampling process for estimating the unbiased CFR during the course of an outbreak. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">The population of confirmed cases and sampling
process for estimating the unbiased CFR during the course of an
outbreak. (<a href="https://doi.org/10.1371/journal.pone.0006852.g001" class="external-link">Nishiura et
al., 2009</a>)</div>
</figure><p>From <em>aggregated incidence data</em>, at time <span class="math inline">\(t\)</span> we know the cumulative number of
confirmed cases and deaths, <span class="math inline">\(C_{t}\)</span>
and <span class="math inline">\(D_{t}\)</span>, and wish to estimate the
unbiased CFR <span class="math inline">\(\pi\)</span>, by way of the
factor of underestimation <span class="math inline">\(u_{t}\)</span>.</p>
<p>If we knew the factor of underestimation <span class="math inline">\(u_{t}\)</span> we could specify the size of the
population of confirmed cases no longer at risk (<span class="math inline">\(u_{t}C_{t}\)</span>, <strong>shaded</strong>),
although we do not know which surviving individuals belong to this
group. A proportion <span class="math inline">\(\pi\)</span> of those in
the group of cases still at risk (size <span class="math inline">\((1-
u_{t})C_{t}\)</span>, <strong>unshaded</strong>) is expected to die.</p>
<p>Because each case no longer at risk had an independent probability of
dying, <span class="math inline">\(\pi\)</span>, the number of deaths,
<span class="math inline">\(D_{t}\)</span>, is a sample from a binomial
distribution with sample size <span class="math inline">\(u_{t}C_{t}\)</span>, and probability of dying
<span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>.</p>
<p>This is represented by the following likelihood function to obtain
the maximum likelihood estimate of the unbiased CFR <span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>:</p>
<p><span class="math display">\[
  {\sf L}(\pi | C_{t},D_{t},u_{t}) = \log{\dbinom{u_{t}C_{t}}{D_{t}}} +
D_{t} \log{\pi} +
  (u_{t}C_{t} - D_{t})\log{(1 - \pi)},
\]</span></p>
<p>This estimation is performed by the internal function
<code>?cfr:::estimate_severity()</code>.</p>
</div>
</div>
</div>
</div>
<p>The <code>cfr_static()</code> estimate is useful to get an overall
severity estimate of the outbreak (so far). Once the outbreak has
clearly ended, or has progressed such that deaths are reported more, the
estimated CFR is then <em>closest to</em> the ‘true’ unbiased CFR.</p>
<p>Additionally, a <em>delay-adjusted CFR</em> estimated with
<code>cfr_static()</code> can assess the severity of an emerging
infectious disease <strong>earlier</strong> than the biased or naive
CFR, during an epidemic.</p>
<p>This is particularly useful when an <em>epidemic curve of confirmed
cases</em> is the only data available (i.e. when individual data from
onset to death are unavailable, especially during the early stage of the
epidemic). When there are few deaths or none at all, an assumption has
to be made for the <em>delay distribution</em> from onset to death,
e.g. from literature based on previous outbreaks. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> depict this in the figures with data from the SARS
outbreak in Hong Kong, 2003.</p>
<div id="accordionSpoiler4" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler4" aria-expanded="false" aria-controls="collapseSpoiler4">
  <h3 class="accordion-header" id="headingSpoiler4">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>CFR estimation during SARS outbreak, Hong Kong, 2003</h3>
</button>
<div id="collapseSpoiler4" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler4" aria-labelledby="headingSpoiler4">
<div class="accordion-body">
<p>Figures A and B show the cumulative numbers of cases and deaths of
SARS, and Figure C shows the observed (biased) CFR estimates as a
function of time, i.e. the cumulative number of deaths over cases at
time <span class="math inline">\(t\)</span>. Due to the delay from the
onset of symptoms to death, the biased estimate of CFR at time <span class="math inline">\(t\)</span> underestimates the realised CFR at the
end of an outbreak (i.e. 302/1755 = 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_abc.png" alt="Observed (biased) confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Observed (biased) confirmed case fatality risk
of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><p>Nevertheless, even by only using the observed data for the period
March 19 to April 2, <code>cfr_static()</code> can yield an appropriate
prediction (Figure D), e.g. the delay-adjusted CFR at March 27 is 18.1 %
(95% CI: 10.5, 28.1). An overestimation is seen in the very early stages
of the epidemic, but the 95% confidence limits in the later stages
include the realised CFR (i.e. 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_d.png" alt="Early determination of the delay-adjusted confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Early determination of the delay-adjusted
confirmed case fatality risk of severe acute respiratory syndrome (SARS)
in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure>
</div>
</div>
</div>
</div>
<p>We can explore the <strong>early</strong> determination of the
<em>delay-adjusted CFR</em> using the <code>cfr_rolling()</code>
function.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout3"></a>
</h3>
<div class="callout-content">
<p><code>cfr_rolling()</code> is a utility function that automatically
calculates CFR for each day of the outbreak with the data available up
to that day, saving the user time.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily naive CFR</span></span>
<span><span class="va">rolling_cfr_naive</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_naive</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31     0.9510204    0.9160061     0.9744387
69 1976-11-01     0.9510204    0.9160061     0.9744387
70 1976-11-02     0.9510204    0.9160061     0.9744387
71 1976-11-03     0.9510204    0.9160061     0.9744387
72 1976-11-04     0.9510204    0.9160061     0.9744387
73 1976-11-05     0.9551020    0.9210866     0.9773771</code></pre>
</div>
<p>With <code>tail()</code>, we show that the latest CFR estimates are
equal to the <code>cfr_static()</code> outputs</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily delay-adjusted CFR</span></span>
<span><span class="va">rolling_cfr_adjusted</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_adjusted</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31         0.975        0.856             1
69 1976-11-01         0.975        0.856             1
70 1976-11-02         0.971        0.852             1
71 1976-11-03         0.971        0.852             1
72 1976-11-04         0.971        0.852             1
73 1976-11-05         0.971        0.852             1</code></pre>
</div>
<p>Now, let’s visualise both results in a time series. How would the
naive and delay-adjusted CFR estimates perform in real time?</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the latest delay-adjusted CFR</span></span>
<span><span class="va">rolling_cfr_adjusted_end</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">slice_tail</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># bind by rows both output data frames</span></span>
<span><span class="fu">bind_rows</span><span class="op">(</span></span>
<span>  <span class="va">rolling_cfr_naive</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"naive"</span><span class="op">)</span>,</span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"adjusted"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># visualise both adjusted and unadjusted rolling estimates</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_ribbon</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span></span>
<span>      <span class="va">date</span>,</span>
<span>      ymin <span class="op">=</span> <span class="va">severity_low</span>,</span>
<span>      ymax <span class="op">=</span> <span class="va">severity_high</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">method</span></span>
<span>    <span class="op">)</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span><span class="va">date</span>, <span class="va">severity_mean</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_mean</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_low</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_high</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The horizontal line represents the delay-adjusted CFR estimated at
the outbreak’s end. The dotted line means the estimate has a 95%
confidence interval (95% CI).</p>
<div id="interpret-the-early-stage-cfr-estimate" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="interpret-the-early-stage-cfr-estimate" class="callout-inner">
<h3 class="callout-title">Interpret the early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#interpret-the-early-stage-cfr-estimate"></a>
</h3>
<div class="callout-content">
<p>Based on the figure above:</p>
<ul>
<li>How much difference in days is between the date in which the 95% CI
of the estimated <em>delay-adjusted CFR</em> vs <em>naive CFR</em> cross
with the CFR estimated at the end of the outbreak?</li>
</ul>
<p>Discuss:</p>
<ul>
<li>What are the Public health policy implications of having a
<em>delay-adjusted CFR</em> estimate?</li>
</ul>
</div>
</div>
</div>
<div id="accordionHint3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint3" aria-expanded="false" aria-controls="collapseHint3">
  <h4 class="accordion-header" id="headingHint3">Give me a hint</h4>
</button>
<div id="collapseHint3" class="accordion-collapse collapse" data-bs-parent="#accordionHint3" aria-labelledby="headingHint3">
<div class="accordion-body">
<p>We can use either visual inspection or analysis of the output data
frames.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<p>There is almost one month of difference.</p>
<p>Note that the estimate has considerable uncertainty at the beginning
of the time series. After two weeks, the delay-adjusted CFR approaches
the overall CFR estimate at the outbreak’s end.</p>
<p>Is this pattern similar to other outbreaks? We can use the data sets
in this episode’s challenges. We invite you to find it out!</p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler5" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler5" aria-expanded="false" aria-controls="collapseSpoiler5">
  <h3 class="accordion-header" id="headingSpoiler5">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>rolling or time-varying?</h3>
</button>
<div id="collapseSpoiler5" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler5" aria-labelledby="headingSpoiler5">
<div class="accordion-body">
<p>The <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package has a function called
<code>cfr_time_varying()</code> with different functionality that makes
them not comparable.</p>
<div class="section level4">
<h4 id="when-to-use-cfr_rolling">When to use <code>cfr_rolling()</code>?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_rolling"></a>
</h4>
<p><code>cfr_rolling()</code> shows the estimated CFR on each outbreak
day, given that future data on cases and deaths is unavailable at the
time. The final value of <code>cfr_rolling()</code> estimates is
identical to <code>cfr_static()</code> on the same data.</p>
<p>Remember, as shown above, <code>cfr_rolling()</code> is helpful to
get early-stage CFR estimates and check whether an outbreak’s CFR
estimate has stabilised. Thus, <code>cfr_rolling()</code> is not
sensitive to the length or size of the epidemic.</p>
</div>
<div class="section level4">
<h4 id="when-to-use-cfr_time_varying">When to use <code>cfr_time_varying()</code>?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_time_varying"></a>
</h4>
<p>On the other hand, <code>cfr_time_varying()</code> calculates the CFR
over a moving window and helps to understand changes in CFR due to
changes in the epidemic, e.g. due to a new variant or increased immunity
from vaccination.</p>
<p>However, <code>cfr_time_varying()</code> is sensitive to sampling
uncertainty. Thus, it is sensitive to the size of the outbreak. The
higher the number of cases with expected outcomes on a given day, the
more reasonable estimates of the time-varying CFR we will get.</p>
<p>For example, with 100 cases, the fatality risk estimate will, roughly
speaking, have a 95% confidence interval ±10% of the mean estimate
(binomial CI). So if we have &gt;100 cases with expected outcomes <em>on
a given day</em>, we can get reasonable estimates of the time varying
CFR. But if we only have &gt;100 cases <em>over the course of the whole
epidemic</em>, we probably need to rely on <code>cfr_rolling()</code>
that uses the cumulative data.</p>
<p>We invite you to read this <a href="https://epiverse-trace.github.io/cfr/articles/estimate_time_varying_severity.html" class="external-link">vignette
about the <code>cfr_time_varying()</code> function</a>.</p>
</div>
</div>
</div>
</div>
</div>
<div id="checklist1" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Checklist<a class="anchor" aria-label="anchor" href="#checklist1"></a>
</h3>
<div class="callout-content">
<p>With <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we estimate the CFR as the proportion of
deaths among <strong>confirmed</strong> cases.</p>
<p>By only using <strong>confirmed</strong> cases, it is clear that all
cases that do not seek medical treatment or are not notified will be
missed, as well as all asymptomatic cases. This means that the CFR
estimate is higher than the proportion of deaths among the infected.</p>
</div>
</div>
</div>
<div id="severity-heterogeneity" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="severity-heterogeneity" class="callout-inner">
<h3 class="callout-title">Severity heterogeneity<a class="anchor" aria-label="anchor" href="#severity-heterogeneity"></a>
</h3>
<div class="callout-content">
<p>The CFR may differ across populations (e.g. age, space, treatment);
quantifying these heterogeneities can help target resources
appropriately and compare different care regimens (<a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371" class="external-link">Cori
et al., 2017</a>).</p>
<p>Use the <code>cfr::covid_data</code> data frame to estimate a
delay-adjusted CFR stratified by country.</p>
</div>
</div>
</div>
<div id="accordionHint4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint4" aria-expanded="false" aria-controls="collapseHint4">
  <h4 class="accordion-header" id="headingHint4">Give me a hint</h4>
</button>
<div id="collapseHint4" class="accordion-collapse collapse" data-bs-parent="#accordionHint4" aria-labelledby="headingHint4">
<div class="accordion-body">
<p>One way to do a <em>stratified analysis</em> is to apply a model to
nested data. This <a href="https://tidyr.tidyverse.org/articles/nest.html#nested-data-and-models" class="external-link"><code>{tidyr}</code>
vignette</a> shows you how to apply the <code>group_by()</code> +
<code>nest()</code> to nest data, and then <code>mutate()</code> +
<code>map()</code> to apply the model.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6">Show me the solution</h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 20,786
Columns: 4
$ date    &lt;date&gt; 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2…
$ country &lt;chr&gt; "Argentina", "Brazil", "Colombia", "France", "Germany", "India…
$ cases   &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ deaths  &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_onset_death</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    temp <span class="op">=</span></span>
<span>      <span class="fu">map</span><span class="op">(</span></span>
<span>        .x <span class="op">=</span> <span class="va">data</span>,</span>
<span>        .f <span class="op">=</span> <span class="fu">cfr</span><span class="fu">::</span><span class="va">cfr_static</span>,</span>
<span>        delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">delay_onset_death</span>, <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 19 × 5
# Groups:   country [19]
   country        data                 severity_mean severity_low severity_high
   &lt;chr&gt;          &lt;list&gt;                       &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
 1 Argentina      &lt;tibble [1,094 × 3]&gt;         0.013        0.013         0.013
 2 Brazil         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 3 Colombia       &lt;tibble [1,094 × 3]&gt;         0.022        0.022         0.022
 4 France         &lt;tibble [1,094 × 3]&gt;         0.004        0.004         0.004
 5 Germany        &lt;tibble [1,094 × 3]&gt;         0.005        0.005         0.005
 6 India          &lt;tibble [1,094 × 3]&gt;         0.012        0.012         0.012
 7 Indonesia      &lt;tibble [1,094 × 3]&gt;         0.024        0.024         0.024
 8 Iran           &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 9 Italy          &lt;tibble [1,094 × 3]&gt;         0.007        0.007         0.007
10 Mexico         &lt;tibble [1,094 × 3]&gt;         0.046        0.046         0.046
11 Peru           &lt;tibble [1,094 × 3]&gt;         0.05         0.05          0.05 
12 Poland         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
13 Russia         &lt;tibble [1,094 × 3]&gt;         0.018        0.018         0.018
14 South Africa   &lt;tibble [1,094 × 3]&gt;         0.025        0.025         0.025
15 Spain          &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
16 Turkey         &lt;tibble [1,094 × 3]&gt;         0.006        0.006         0.006
17 Ukraine        &lt;tibble [1,094 × 3]&gt;         0.02         0.02          0.02 
18 United Kingdom &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
19 United States  &lt;tibble [1,094 × 3]&gt;         0.011        0.011         0.011</code></pre>
</div>
<p>Great! Now you can use similar code for any other stratified analysis
like age, regions or more!</p>
<p>But, how can we interpret that there is a country variability of
severity from the same diagnosed pathogen?</p>
<p>Local factors like testing capacity, the case definition, and
sampling regime can affect the report of cases and deaths, thus
affecting case ascertainment. Take a look to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>
vignette on <a href="https://epiverse-trace.github.io/cfr/articles/estimate_ascertainment.html" class="external-link">Estimating
the proportion of cases that are ascertained during an outbreak</a>!</p>
</div>
</div>
</div>
</div>
<div id="limitations" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="limitations" class="callout-inner">
<h3 class="callout-title">Limitations<a class="anchor" aria-label="anchor" href="#limitations"></a>
</h3>
<div class="callout-content">
<ul>
<li>The delay-adjusted CFR does not address all sources of error in data
like the underdiagnosis of infected individuals.</li>
</ul>
</div>
</div>
</div>
</section><section id="more-severity-measures"><h2 class="section-heading">More severity measures<a class="anchor" aria-label="anchor" href="#more-severity-measures"></a>
</h2>
<hr class="half-width">
<p>Suppose we need to assess the clinical severity of the epidemic in a
context different from surveillance data, like the severity among cases
that arrive at hospitals or cases you collected from a representative
serological survey.</p>
<p>Using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we can change the inputs for the numerator
(<code>cases</code>) and denominator (<code>deaths</code>) to estimate
more severity measures like the Infection fatality risk (IFR) or the
Hospitalisation Fatality Risk (HFR). We can follow this analogy:</p>
<p>If for a <em>Case</em> fatality risk (CFR), we require:</p>
<ul>
<li>
<em>case</em> and death incidence data, with a</li>
<li>case-to-death delay distribution (or close approximation, such as
symptom onset-to-death).</li>
</ul>
<p>Then, the <em>Infection</em> fatality risk (IFR) requires:</p>
<ul>
<li>
<em>infection</em> and death incidence data, with an</li>
<li>exposure-to-death delay distribution (or close approximation).</li>
</ul>
<p>Similarly, the <em>Hospitalisation</em> Fatality Risk (HFR)
requires:</p>
<ul>
<li>
<em>hospitalisation</em> and death incidence data, and a</li>
<li>hospitalisation-to-death delay distribution.</li>
</ul>
<div id="data-sources-for-more-severity-measures" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="data-sources-for-more-severity-measures" class="callout-inner">
<h3 class="callout-title">Data sources for more severity measures<a class="anchor" aria-label="anchor" href="#data-sources-for-more-severity-measures"></a>
</h3>
<div class="callout-content">
<p><a href="https://www.nature.com/articles/s41467-020-19238-2/figures/1" class="external-link">Yang
et al., 2020</a> summarises different definitions and data sources:</p>
<figure><img src="../fig/cfr-s41467-020-19238-2-fig_a.png" alt="Severity levels of infections with SARS-CoV-2 and parameters of interest. Each level is assumed to be a subset of the level below." class="figure mx-auto d-block"><div class="figcaption">Severity levels of infections with SARS-CoV-2
and parameters of interest. Each level is assumed to be a subset of the
level below.</div>
</figure><ul>
<li>sCFR symptomatic case-fatality risk,</li>
<li>sCHR symptomatic case-hospitalisation risk,</li>
<li>mCFR medically attended case-fatality risk,</li>
<li>mCHR medically attended case-hospitalisation risk,</li>
<li>HFR hospitalisation-fatality risk.</li>
</ul>
<figure><img src="../fig/cfr-s41467-020-19238-2-fig_b.png" alt="Data source of COVID-19 cases in Wuhan: D1) 32,583 laboratory-confirmed COVID-19 cases as of March 84, D2) 17,365 clinically-diagnosed COVID-19 cases during February 9–194, D3)daily number of laboratory-confirmed cases on March 9–April 243, D4) total number of COVID-19 deaths as of April 24 obtained from the Hubei Health Commission3, D5) 325 laboratory-confirmed cases and D6) 1290 deaths were added as of April 16 through a comprehensive and systematic verification by Wuhan Authorities3, and D7) 16,781 laboratory-confirmed cases identified through universal screening10,11. Pse: RT-PCR sensitivity12. Pmed.care: proportion of seeking medical assistance among patients suffering from acute respiratory infections13." class="figure mx-auto d-block"><div class="figcaption">Schematic diagram of the baseline analyses. Red,
blue, and green arrows denote the data flow from laboratory-confirmed
cases of passive surveillance, clinically-diagnosed cases, and
laboratory-confirmed cases of active screenings.</div>
</figure>
</div>
</div>
</div>
<div id="how-to-rearrange-my-input-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-rearrange-my-input-data" class="callout-inner">
<h3 class="callout-title">How to rearrange my input data?<a class="anchor" aria-label="anchor" href="#how-to-rearrange-my-input-data"></a>
</h3>
<div class="callout-content">
<p>Rearranging the input data for data analysis can take most of the
time.</p>
<p>For <em>aggregated case data</em>, we encourage you to use
<a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a>! Refer to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette on
<a href="https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html" class="external-link">Handling
data from <code>{incidence2}</code></a>.</p>
<!-- cite howto entry one lineslist + incidence2 + cfr connection -->
</div>
</div>
</div>
<div id="hospitalisation-fatality-risk-hfr" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="hospitalisation-fatality-risk-hfr" class="callout-inner">
<h3 class="callout-title">Hospitalisation Fatality Risk (HFR)<a class="anchor" aria-label="anchor" href="#hospitalisation-fatality-risk-hfr"></a>
</h3>
<div class="callout-content">
<p>From the <code>{covidregionaldata}</code> package, load COVID cases
from the region England from country UK to estimate the delay-adjusted
HFR using the corresponding distribution delay.</p>
</div>
</div>
</div>
<div id="accordionHint5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint5" aria-expanded="false" aria-controls="collapseHint5">
  <h4 class="accordion-header" id="headingHint5">Give me a hint</h4>
</button>
<div id="collapseHint5" class="accordion-collapse collapse" data-bs-parent="#accordionHint5" aria-labelledby="headingHint5">
<div class="accordion-body">
<p>Use <code>covidregionaldata::get_regional_data()</code> to get COVID
data from the region England from the country UK.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_england</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">covidregionaldata</span><span class="fu">::</span><span class="fu">get_regional_data</span><span class="op">(</span></span>
<span>    country <span class="op">=</span> <span class="st">"uk"</span>,</span>
<span>    regions <span class="op">=</span> <span class="st">"England"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Also, check out the vignette on <a href="https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html" class="external-link">Handling
data from <code>{incidence2}</code></a></p>
</div>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7">Show me the solution</h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" data-bs-parent="#accordionSolution7" aria-labelledby="headingSolution7">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">covidregionaldata</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_england</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">covidregionaldata</span><span class="fu">::</span><span class="fu">get_regional_data</span><span class="op">(</span></span>
<span>    country <span class="op">=</span> <span class="st">"uk"</span>,</span>
<span>    regions <span class="op">=</span> <span class="st">"England"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_england</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 1,415
Columns: 26
$ date                             &lt;date&gt; 2020-01-30, 2020-01-31, 2020-02-01, …
$ region                           &lt;chr&gt; "England", "England", "England", "Eng…
$ region_code                      &lt;chr&gt; "E92000001", "E92000001", "E92000001"…
$ cases_new                        &lt;dbl&gt; 1, 0, 0, 1, 18, 0, 1, 0, 0, 3, 1, 1, …
$ cases_total                      &lt;dbl&gt; 1, 1, 1, 2, 20, 20, 21, 21, 21, 24, 2…
$ deaths_new                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ deaths_total                     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ recovered_new                    &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ recovered_total                  &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ hosp_new                         &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ hosp_total                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ tested_new                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ tested_total                     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ areaType                         &lt;chr&gt; "nation", "nation", "nation", "nation…
$ cumCasesByPublishDate            &lt;dbl&gt; NA, 2, 2, 2, 2, 2, 2, 3, 3, 3, 4, 8, …
$ cumCasesBySpecimenDate           &lt;dbl&gt; 1, 1, 1, 2, 20, 20, 21, 21, 21, 24, 2…
$ newCasesByPublishDate            &lt;dbl&gt; NA, 2, 0, 0, 0, 0, 0, 1, 0, 0, 1, 4, …
$ newCasesBySpecimenDate           &lt;dbl&gt; 1, 0, 0, 1, 18, 0, 1, 0, 0, 3, 1, 1, …
$ cumDeaths28DaysByDeathDate       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ cumDeaths28DaysByPublishDate     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newDeaths28DaysByDeathDate       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newDeaths28DaysByPublishDate     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarFourTestsByPublishDate  &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarOneTestsByPublishDate   &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarThreeTestsByPublishDate &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarTwoTestsByPublishDate   &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_data_incidence</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">covid_england</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="fu">incidence</span><span class="op">(</span></span>
<span>    date_index <span class="op">=</span> <span class="st">"date"</span>,</span>
<span>    counts <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"hosp_total"</span>, <span class="st">"deaths_total"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_incidence</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># incidence:  2,830 x 3
# count vars: hosp_total, deaths_total
   date_index count_variable count
 * &lt;date&gt;     &lt;fct&gt;          &lt;dbl&gt;
 1 2020-01-30 hosp_total        NA
 2 2020-01-30 deaths_total      NA
 3 2020-01-31 hosp_total        NA
 4 2020-01-31 deaths_total      NA
 5 2020-02-01 hosp_total        NA
 6 2020-02-01 deaths_total      NA
 7 2020-02-02 hosp_total        NA
 8 2020-02-02 deaths_total      NA
 9 2020-02-03 hosp_total        NA
10 2020-02-03 deaths_total      NA
# ℹ 2,820 more rows</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_data_prepared</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">prepare_data</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">covid_data_incidence</span>,</span>
<span>    cases_variable <span class="op">=</span> <span class="st">"hosp_total"</span>,</span>
<span>    deaths_variable <span class="op">=</span> <span class="st">"deaths_total"</span>,</span>
<span>    fill_NA <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_prepared</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 1,415
Columns: 3
$ date   &lt;date&gt; 2020-01-30, 2020-01-31, 2020-02-01, 2020-02-02, 2020-02-03, 20…
$ cases  &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ deaths &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_hosp_death</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"hospitalisation to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_prepared</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>    delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">delay_hosp_death</span>, <span class="va">x</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.215        0.215         0.215</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li><p>Use <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> to estimate severity</p></li>
<li><p>Use <code>cfr_static()</code> to estimate the overall CFR with
the latest data available.</p></li>
<li><p>Use <code>cfr_rolling()</code> to show what the estimated CFR
would be on each day of the outbreak.</p></li>
<li><p>Use the <code>delay_density</code> argument to adjust the CFR by
the corresponding delay distribution.</p></li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/epiverse-trace/tutorials/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/epiverse-trace/tutorials/" class="external-link">Source</a></p>
				<p><a href="https://github.com/epiverse-trace/tutorials/blob/main/" class="external-link">Cite</a> | <a href="mailto:andree.valle-campos@lshtm.ac.uk">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.3)</a>, <a href="https://github.com/carpentries/pegboard/tree/e9ca729c7db3ca97f016480679f9f58d115d2721" class="external-link">pegboard (0.7.4)</a>, and <a href="https://github.com/epiverse-trace/varnish/tree/epiversetheme" class="external-link">varnish (1.0.2.9000)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://epiverse-trace.github.io/tutorials/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "forecasts, epidemic models, interventions",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://epiverse-trace.github.io/tutorials/instructor/aio.html",
  "identifier": "https://epiverse-trace.github.io/tutorials/instructor/aio.html",
  "dateCreated": "2024-03-25",
  "dateModified": "2024-03-26",
  "datePublished": "2024-03-26"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

