<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Real-time analysis and forecasting for outbreak analytics with R: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-XZ471458FQ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XZ471458FQ');
    </script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #FFF">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg">
</div>
    <div class="lesson-title-md">
      Real-time analysis and forecasting for outbreak analytics with R
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Real-time analysis and forecasting for outbreak analytics with R
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a class="btn btn-primary" href="../aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Real-time analysis and forecasting for outbreak analytics with R
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="create-forecast.html">1. Create a short-term forecast</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="severity-static.html">2. Estimation of outbreak severity</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="superspreading-estimate.html">3. Account for superspreading</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="superspreading-simulate.html">4. Simulate transmission chains</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-create-forecast"><p>Content from <a href="create-forecast.html">Create a short-term forecast</a></p>
<hr>
<p>Last updated on 2024-05-04 |
        
        <a href="https://github.com/epiverse-trace/tutorials-middle/edit/main/episodes/create-forecast.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I create short-term forecasts from case data?</li>
<li>How do I account for incomplete reporting in forecasts?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to make forecasts of cases using R package
<code>EpiNow2</code>
</li>
<li>Learn how to include an observation process in estimation</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<ul>
<li>Complete tutorial <a href="quantify-transmissibility.html">Quantifying transmission</a>
</li>
</ul>
<p>Learners should familiarise themselves with following concept
dependencies before working through this tutorial:</p>
<p><strong>Statistics</strong> : probability distributions, principle of
Bayesian analysis.</p>
<p><strong>Epidemic theory</strong> : Effective reproduction number.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Given case data of an epidemic, we can create estimates of the
current and future number of cases by accounting for both delays in
reporting and under reporting. To make statements about the future of th
epidemic, we need to make an assumption of how observations up to the
present are related to what we expect to happen in the future. The
simplest way of doing so is to assume “no change”, i.e., the
reproduction number remains the same in the future as last observed. In
this tutorial we will create short-term <a href="reference.html#forecast">forecasts</a> by assuming the
reproduction number will remain the same as its estimate was on the
final date for which data was available.</p>
<p>In this tutorial we are going to learn how to use the
<a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> package to forecast cases accounting for
incomplete observations and forecast secondary observations like
deaths.</p>
<p>We’ll use the pipe <code>%&gt;%</code> operator to connect functions,
so let’s also call to the <a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a> package:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">EpiNow2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span></code></pre>
</div>
</section><section id="create-a-short-term-forecast"><h2 class="section-heading">Create a short-term forecast<a class="anchor" aria-label="anchor" href="#create-a-short-term-forecast"></a>
</h2>
<hr class="half-width">
<p>The function <code>epinow()</code> described in the <a href="quantify-transmissibility.html">quantifying transmission</a>
episode is a wrapper for the functions:</p>
<ul>
<li>
<code>estimate_infections()</code> used to estimate cases by date of
infection.</li>
<li>
<code>forecast_infections()</code> used to simulate infections using
an existing fit (estimate) to observed cases.</li>
</ul>
<p>Let’s use the same code used in <a href="quantify-transmissibility.html">quantifying transmission</a>
episode to get the input data, delays and priors:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read cases dataset</span></span>
<span><span class="va">cases</span> <span class="op">&lt;-</span> <span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">date</span>, <span class="va">cases_new</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>confirm <span class="op">=</span> <span class="fu">sum</span><span class="op">(</span><span class="va">cases_new</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Incubation period</span></span>
<span><span class="va">incubation_period_fixed</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">4</span>, sd <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Log-tranformed mean</span></span>
<span><span class="va">log_mean</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">convert_to_logmean</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Log-transformed std</span></span>
<span><span class="va">log_sd</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">convert_to_logsd</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reporting dalay</span></span>
<span><span class="va">reporting_delay_fixed</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="va">log_mean</span>, sd <span class="op">=</span> <span class="va">log_sd</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">10</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Generation time</span></span>
<span><span class="va">generation_time_fixed</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">3.6</span>, sd <span class="op">=</span> <span class="fl">3.1</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rt mean prior</span></span>
<span><span class="va">rt_log_mean</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">convert_to_logmean</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rt std prior</span></span>
<span><span class="va">rt_log_sd</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">convert_to_logsd</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we can extract the short-term forecast using:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assume we only have the first 90 days of this data</span></span>
<span><span class="va">reported_cases</span> <span class="op">&lt;-</span> <span class="va">cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Estimate and forecast</span></span>
<span><span class="va">estimates</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">reported_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div id="do-not-wait-for-this-to-complete" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="do-not-wait-for-this-to-complete" class="callout-inner">
<h3 class="callout-title">Do not wait for this to complete!<a class="anchor" aria-label="anchor" href="#do-not-wait-for-this-to-complete"></a>
</h3>
<div class="callout-content">
<p>This last chunk may take 10 minutes to run. Keep reading this
tutorial episode while this runs in the background. For more information
on computing time, read the “Bayesian inference using Stan” section
within the <a href="quantify-transmissibility.html">quantifying
transmission</a> episode.</p>
</div>
</div>
</div>
<p>We can visualise the estimates of the effective reproduction number
and the estimated number of cases using <code>plot()</code>. The
estimates are split into three categories:</p>
<ul>
<li><p><strong>Estimate</strong> (green): utilises all data,</p></li>
<li><p><strong>Estimate based on partial data</strong> (orange):
contains a higher degree of uncertainty because such estimates are based
on less data,</p></li>
<li><p><strong>Forecast</strong> (purple): forecasts into the
future.</p></li>
</ul>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">estimates</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="section level3">
<h3 id="forecasting-with-incomplete-observations">Forecasting with incomplete observations<a class="anchor" aria-label="anchor" href="#forecasting-with-incomplete-observations"></a>
</h3>
<p>In the <a href="quantify-transmissibility.html">quantifying
transmission</a> episode we accounted for delays in reporting.In
<code>EpiNow2</code> we also can account for incomplete observations as
in reality, 100% of cases are not reported. We will pass another
argument into <code>epinow()</code> function called <code>obs</code> to
define an observation model. The format of <code>obs</code> is defined
by the <code>obs_opt()</code> function (see
<code>?EpiNow2::obs_opts</code> for more detail).</p>
<p>Let’s say we believe the COVID-19 outbreak data in the
<code>cases</code> object do not include all reported cases. We believe
that we only observe 40% of cases. To specify this in the observation
model, we must pass a scaling factor with a mean and standard deviation.
If we assume that 40% of cases are in the case data (with standard
deviation 1%), then we specify the <code>scale</code> input to
<code>obs_opts()</code> as follows:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obs_scale</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.4</span>, sd <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<p>To run the inference framework with this observation process, we add
<code>obs = obs_opts(scale = obs_scale)</code> to the input arguments of
<code>epinow()</code>:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define observation model</span></span>
<span><span class="va">obs_scale</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.4</span>, sd <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assume we only have the first 90 days of this data</span></span>
<span><span class="va">reported_cases</span> <span class="op">&lt;-</span> <span class="va">cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">90</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Estimate and forecast</span></span>
<span><span class="va">estimates</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">reported_cases</span>,</span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">generation_time_fixed</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">delay_opts</span><span class="op">(</span><span class="va">incubation_period_fixed</span> <span class="op">+</span> <span class="va">reporting_delay_fixed</span><span class="op">)</span>,</span>
<span>  rt <span class="op">=</span> <span class="fu">rt_opts</span><span class="op">(</span>prior <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="va">rt_log_mean</span>, sd <span class="op">=</span> <span class="va">rt_log_sd</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  <span class="co"># Add observation model</span></span>
<span>  obs <span class="op">=</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">obs_opts</span><span class="op">(</span>scale <span class="op">=</span> <span class="va">obs_scale</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-05-04 02:26:33] epinow: There were 2 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-05-04 02:26:33] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">base</span><span class="fu">::</span><span class="fu">summary</span><span class="op">(</span><span class="va">estimates</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 measure                 estimate
                                  &lt;char&gt;                   &lt;char&gt;
1: New confirmed cases by infection date   18016 (10081 -- 30016)
2:        Expected change in daily cases        Likely decreasing
3:            Effective reproduction no.       0.89 (0.57 -- 1.3)
4:                        Rate of growth -0.015 (-0.064 -- 0.034)
5:          Doubling/halving time (days)          -46 (21 -- -11)</code></pre>
</div>
<p>The estimates of transmission measures such as the effective
reproduction number and rate of growth are similar (or the same in
value) compared to when we didn’t account for incomplete observations
(see <a href="quantify-transmissibility.html">quantifying transmission
episode</a> in the “Finding estimates” section). However the number of
new confirmed cases by infection date has changed substantially in
magnitude to reflect the assumption that only 40% of cases are in the
data set.</p>
<p>We can also change the default distribution from Negative Binomial to
Poisson, remove the default week effect and more. See
<code>?EpiNow2::obs_opts</code> for more details.</p>
<div id="what-are-the-implications-of-this-change" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="what-are-the-implications-of-this-change" class="callout-inner">
<h3 class="callout-title">What are the implications of this change?<a class="anchor" aria-label="anchor" href="#what-are-the-implications-of-this-change"></a>
</h3>
<div class="callout-content">
<ul>
<li>Compare different percents of observations %</li>
<li>How are they different in the number of infections estimated?</li>
<li>What are the public health implications of this change?</li>
</ul>
</div>
</div>
</div>
</div>
</section><section id="forecasting-secondary-observations"><h2 class="section-heading">Forecasting secondary observations<a class="anchor" aria-label="anchor" href="#forecasting-secondary-observations"></a>
</h2>
<hr class="half-width">
<p><code>EpiNow2</code> also has the ability to estimate and forecast
secondary observations, e.g., deaths and hospitalisations, from a
primary observation, e.g., cases. Here we will illustrate how to
forecast the number of deaths arising from observed cases of COVID-19 in
the early stages of the UK outbreak.</p>
<p>First, we must format our data to have the following columns:</p>
<ul>
<li>
<code>date</code>: the date (as a date object see
<code>?is.Date()</code>),</li>
<li>
<code>primary</code>: number of primary observations on that date,
in this example <strong>cases</strong>,</li>
<li>
<code>secondary</code>: number of secondary observations date, in
this example <strong>deaths</strong>.</li>
</ul>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reported_cases_deaths</span> <span class="op">&lt;-</span> <span class="fu">incidence2</span><span class="fu">::</span><span class="va">covidregionaldataUK</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">date</span>, <span class="va">cases_new</span>, <span class="va">deaths_new</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span></span>
<span>    primary <span class="op">=</span> <span class="fu">sum</span><span class="op">(</span><span class="va">cases_new</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    secondary <span class="op">=</span> <span class="fu">sum</span><span class="op">(</span><span class="va">deaths_new</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure style="text-align: center"><img src="../fig/create-forecast-rendered-unnamed-chunk-8-1.png" alt="Distribution of secondary cases (deaths). We will drop the first 30 days with no observed deaths. We will use the deaths between day 31 and day 60 to estimate the secondary observations. We will forecast deaths from day 61 to day 90." class="figure mx-auto d-block"><figcaption>
Distribution of secondary cases (deaths). We will drop the first 30 days
with no observed deaths. We will use the deaths between day 31 and day
60 to estimate the secondary observations. We will forecast deaths from
day 61 to day 90.
</figcaption></figure><p>Using the data on cases and deaths between day 31 and day 60, we will
estimate the relationship between the primary and secondary observations
using <code>estimate_secondary()</code>, then forecast future deaths
using <code>forecast_secondary()</code>. For more details on the model
see the <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_secondary.html" class="external-link">model
documentation</a>.</p>
<p>We must specify the type of observation using <code>type</code> in
<code>secondary_opts()</code>, options include:</p>
<ul>
<li>“incidence”: secondary observations arise from previous primary
observations, i.e., deaths arising from recorded cases.</li>
<li>“prevalence”: secondary observations arise from a combination
current primary observations and past secondary observations, i.e.,
hospital bed usage arising from current hospital admissions and past
hospital bed usage.</li>
</ul>
<p>In this example we specify
<code>secondary_opts(type = "incidence")</code>. See
<code>?EpiNow2::secondary_opts</code> for more detail.</p>
<p>The final key input is the delay distribution between the primary and
secondary observations. Here this is the delay between case report and
death, we assume this follows a gamma distribution with mean of 14 days
and standard deviation of 5 days (Alternatively, we can use
<code>{epiparameter}</code> to <a href="https://epiverse-trace.github.io/tutorials-early/delays-reuse.html" class="external-link">access
epidemiological delays</a>). Using <code>dist_spec()</code> we specify a
fixed gamma distribution.</p>
<p>There are further function inputs to
<code>estimate_secondary()</code> which can be specified, including
adding an observation process, see
<code>?EpiNow2::estimate_secondary</code> for detail on these
options.</p>
<p>To find the model fit between cases and deaths:</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate from day 31 to day 60 of this data</span></span>
<span><span class="va">cases_to_estimate</span> <span class="op">&lt;-</span> <span class="va">reported_cases_deaths</span><span class="op">[</span><span class="fl">31</span><span class="op">:</span><span class="fl">60</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Estimate secondary cases</span></span>
<span><span class="va">estimate_cases_to_deaths</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">estimate_secondary</span><span class="op">(</span></span>
<span>  reports <span class="op">=</span> <span class="va">cases_to_estimate</span>,</span>
<span>  secondary <span class="op">=</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">secondary_opts</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"incidence"</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">delay_opts</span><span class="op">(</span><span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>    mean <span class="op">=</span> <span class="fl">14</span>, sd <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    max <span class="op">=</span> <span class="fl">30</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div id="be-cautious-of-time-scale" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="be-cautious-of-time-scale" class="callout-inner">
<h3 class="callout-title">Be cautious of time-scale<a class="anchor" aria-label="anchor" href="#be-cautious-of-time-scale"></a>
</h3>
<div class="callout-content">
<p>In the early stages of an outbreak there can be substantial changes
in testing and reporting. If there are testing changes from one month to
another, then there will be a bias in the model fit. Therefore, you
should be cautious of the time-scale of data used in the model fit and
forecast.</p>
</div>
</div>
</div>
<p>We plot the model fit (shaded ribbons) with the secondary
observations (bar plot) and primary observations (dotted line) as
follows:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">estimate_cases_to_deaths</span>, primary <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-10-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To use this model fit to forecast deaths, we pass a data frame
consisting of the primary observation (cases) for dates not used in the
model fit.</p>
<p><em>Note : in this episode we are using data where we know the deaths
and cases, so we create a data frame by extracting the cases. But in
practice, this would be a different data set consisting of cases
only.</em></p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Forecast from day 61 to day 90</span></span>
<span><span class="va">cases_to_forecast</span> <span class="op">&lt;-</span> <span class="va">reported_cases_deaths</span><span class="op">[</span><span class="fl">61</span><span class="op">:</span><span class="fl">90</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"primary"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">cases_to_forecast</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"date"</span>, <span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<p>To forecast, we use the model fit
<code>estimate_cases_to_deaths</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Forecast secondary cases</span></span>
<span><span class="va">deaths_forecast</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">forecast_secondary</span><span class="op">(</span></span>
<span>  estimate <span class="op">=</span> <span class="va">estimate_cases_to_deaths</span>,</span>
<span>  primary <span class="op">=</span> <span class="va">cases_to_forecast</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">deaths_forecast</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Removed 30 rows containing missing values (`position_stack()`).</code></pre>
</div>
<figure><img src="../fig/create-forecast-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot shows the forecast secondary observations (deaths) over the
dates which we have recorded cases for. It is also possible to forecast
deaths using forecast cases, here you would specify <code>primary</code>
as the <code>estimates</code> output from
<code>estimate_infections()</code>.</p>
<div id="credible-intervals" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="credible-intervals" class="callout-inner">
<h3 class="callout-title">Credible intervals<a class="anchor" aria-label="anchor" href="#credible-intervals"></a>
</h3>
<div class="callout-content">
<p>In all <a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> output figures, shaded regions reflect
90%, 50%, and 20% credible intervals in order from lightest to
darkest.</p>
</div>
</div>
</div>
</section><section id="challenge-ebola-outbreak-analysis"><h2 class="section-heading">Challenge: Ebola outbreak analysis<a class="anchor" aria-label="anchor" href="#challenge-ebola-outbreak-analysis"></a>
</h2>
<hr class="half-width">
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Download the file <a href="../data/ebola_cases.csv">ebola_cases.csv</a>
and read it into R. The simulated data consists of the date of symptom
onset and number of confirmed cases of the early stages of the Ebola
outbreak in Sierra Leone in 2014.</p>
<p>Using the first 3 months (120 days) of data:</p>
<ol style="list-style-type: decimal">
<li>Estimate whether cases are increasing or decreasing on day 120 of
the outbreak</li>
<li>Account for a capacity to observe 80% of cases.</li>
<li>Create a two week forecast of number of cases.</li>
</ol>
<p>You can use the following parameter values for the delay
distribution(s) and generation time distribution.</p>
<ul>
<li>Incubation period: Log normal<span class="math inline">\((2.487,0.330)\)</span> (<a href="https://doi.org/10.1016/j.phrp.2011.04.001" class="external-link">Eichner et
al. 2011</a> via <code>{epiparameter}</code>)</li>
<li>Generation time: Gamma<span class="math inline">\((15.3,
10.1)\)</span> (<a href="https://www.nejm.org/doi/full/10.1056/NEJMoa1411100" class="external-link">WHO Ebola
Response Team 2014</a>)</li>
</ul>
<p>You may include some uncertainty around the mean and standard
deviation of these distributions.</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" aria-labelledby="headingHint1" data-bs-parent="#accordionHint1">
<div class="accordion-body">
<p>We use the effective reproduction number and growth rate to estimate
whether cases are increasing or decreasing.</p>
<p>We can use the <code>horizon</code> argument within the
<code>epinow()</code> function to extend the time period of the
forecast. The default value is of seven days.</p>
<p>Ensure the data is in the correct format :</p>
<ul>
<li>
<code>date</code>: the date (as a date object see
<code>?is.Date()</code>),</li>
<li>
<code>confirm</code>: number of confirmed cases on that date.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">SOLUTION</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>To estimate the effective reproduction number and growth rate, we
will use the function <code>epinow()</code>.</p>
<p>As the data consists of date of symptom onset, we only need to
specify a delay distribution for the incubation period and the
generation time.</p>
<p>We specify the distributions with some uncertainty around the mean
and standard deviation of the log normal distribution for the incubation
period and the Gamma distribution for the generation time.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ebola_incubation_period</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">2.487</span>, sd <span class="op">=</span> <span class="fl">0.330</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">20</span>, distribution <span class="op">=</span> <span class="st">"lognormal"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_generation_time</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">dist_spec</span><span class="op">(</span></span>
<span>  mean <span class="op">=</span> <span class="fl">15.3</span>, sd <span class="op">=</span> <span class="fl">10.1</span>,</span>
<span>  mean_sd <span class="op">=</span> <span class="fl">0.5</span>, sd_sd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max <span class="op">=</span> <span class="fl">30</span>, distribution <span class="op">=</span> <span class="st">"gamma"</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>We read the data input using <code>readr::read_csv()</code>. This
function recognize that the column <code>date</code> is a
<code>&lt;date&gt;</code> class vector.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co"># e.g.: if path to file is data/raw-data/ebola_cases.csv then:</span></span>
<span><span class="va">ebola_cases</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu">read_csv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"ebola_cases.csv"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>We define an observation model to scale the estimated and forecast
number of new infections:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define observation model</span></span>
<span><span class="co"># mean of 80% and standard deviation of 1%</span></span>
<span><span class="va">ebola_obs_scale</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.8</span>, sd <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<p>As we want to also create a two week forecast, we specify
<code>horizon = 14</code> to forecast 14 days instead of the default 7
days.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ebola_estimates</span> <span class="op">&lt;-</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">epinow</span><span class="op">(</span></span>
<span>  reported_cases <span class="op">=</span> <span class="va">ebola_cases</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">120</span>, <span class="op">]</span>, <span class="co"># first 3 months of data only</span></span>
<span>  generation_time <span class="op">=</span> <span class="fu">generation_time_opts</span><span class="op">(</span><span class="va">ebola_generation_time</span><span class="op">)</span>,</span>
<span>  delays <span class="op">=</span> <span class="fu">EpiNow2</span><span class="fu">::</span><span class="fu">delay_opts</span><span class="op">(</span><span class="va">ebola_incubation_period</span><span class="op">)</span>,</span>
<span>  <span class="co"># Add observation model</span></span>
<span>  obs <span class="op">=</span> <span class="fu">obs_opts</span><span class="op">(</span>scale <span class="op">=</span> <span class="va">ebola_obs_scale</span><span class="op">)</span>,</span>
<span>  <span class="co"># horizon needs to be 14 days to create two week forecast (default is 7 days)</span></span>
<span>  horizon <span class="op">=</span> <span class="fl">14</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>WARN [2024-05-04 02:28:45] epinow: There were 13 divergent transitions after warmup. See
https://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
to find out why this is a problem and how to eliminate them. - 
WARN [2024-05-04 02:28:45] epinow: Examine the pairs() plot to diagnose sampling problems
 - </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">ebola_estimates</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                                 measure                estimate
                                  &lt;char&gt;                  &lt;char&gt;
1: New confirmed cases by infection date         126 (57 -- 325)
2:        Expected change in daily cases              Increasing
3:            Effective reproduction no.            1.7 (1 -- 3)
4:                        Rate of growth 0.043 (0.0011 -- 0.092)
5:          Doubling/halving time (days)         16 (7.5 -- 620)</code></pre>
</div>
<p>The effective reproduction number <span class="math inline">\(R_t\)</span> estimate (on the last date of the
data) is 1.7 (1 – 3). The exponential growth rate of case numbers is
0.043 (0.0011 – 0.092).</p>
<p>Visualize the estimates:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span><span class="va">ebola_estimates</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="forecasting-with-estimates-of-r_t" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="forecasting-with-estimates-of-r_t" class="callout-inner">
<h3 class="callout-title">Forecasting with estimates of<span class="math inline">\(R_t\)</span><a class="anchor" aria-label="anchor" href="#forecasting-with-estimates-of-r_t"></a>
</h3>
<div class="callout-content">
<p>By default, the short-term forecasts are created using the latest
estimate of the reproduction number <span class="math inline">\(R_t\)</span>. As this estimate is based on partial
data, it has considerable uncertainty.</p>
<p>The reproduction number that is projected into the future can be
changed to a less recent estimate based on more data using
<code>rt_opts()</code>:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rt_opts</span><span class="op">(</span>future <span class="op">=</span> <span class="st">"estimate"</span><span class="op">)</span></span></code></pre>
</div>
<p>The result will be less uncertain forecasts (as they are based on
<span class="math inline">\(R_t\)</span> with a narrower uncertainty
interval) but the forecasts will be based on less recent estimates of
<span class="math inline">\(R_t\)</span> and assume no change since
then.</p>
<p>Additionally, there is the option to <a href="reference.html#projection">project</a> the value of <span class="math inline">\(R_t\)</span> into the future using a generic model
by setting <code>future = "project"</code>. As this option uses a model
to forecast the value of <span class="math inline">\(R_t\)</span>, the
result will be forecasts that are more uncertain than
<code>estimate</code>, for an example <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections_options.html#projecting-the-reproduction-number-with-the-gaussian-process" class="external-link">see
here</a>.</p>
</div>
</div>
</div>
</section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p><code>EpiNow2</code> can be used to create short term forecasts and
to estimate the relationship between different outcomes. There are a
range of model options that can be implemented for different analysis,
including adding an observational process to account for incomplete
reporting. See the <a href="https://epiforecasts.io/EpiNow2/dev/articles/estimate_infections_options.html" class="external-link">vignette</a>
for more details on different model options in <code>EpiNow2</code> that
aren’t covered in these tutorials.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>We can create short-term forecasts by making assumptions about the
future behaviour of the reproduction number</li>
<li>Incomplete case reporting can be accounted for in estimates</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-severity-static"><p>Content from <a href="severity-static.html">Estimation of outbreak severity</a></p>
<hr>
<p>Last updated on 2024-05-04 |
        
        <a href="https://github.com/epiverse-trace/tutorials-middle/edit/main/episodes/severity-static.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li><p>Why do we estimate the clinical severity of an epidemic?</p></li>
<li><p>How can the Case Fatality Risk (CFR) be estimated early in an
ongoing epidemic?</p></li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li><p>Estimate the CFR from aggregated case data using
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate a delay-adjusted CFR using <code>{epiparameter}</code>
and <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate a delay-adjusted severity for an expanding time series
using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>This episode requires you to be familiar with:</p>
<p><strong>Data science</strong>: Basic programming with R.</p>
<p><strong>Epidemic theory</strong>: <a href="reference.html#delaydist">Delay distributions</a>.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Common questions at the early stage of an epidemic include:</p>
<ul>
<li>What is the likely public health impact of the outbreak in terms of
clinical severity?</li>
<li>What are the most severely affected groups?</li>
<li>Does the outbreak have the potential to cause a very severe
pandemic?</li>
</ul>
<p>We can assess the pandemic potential of an epidemic with two critical
measurements: the transmissibility and the clinical severity (<a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a>, <a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>).</p>
<figure><img src="../fig/cfr-hhs-scenarios-psaf.png" alt="The horizontal axis is the scaled measure of clinical severity, ranging from 1 to 7, where 1 is low, 4 is moderate, and 7 is very severe. The vertical axis is the scaled measure of transmissibility, ranging from 1 to 5, where 1 is low, 3 is moderate, and 5 is highly transmissible. On the graph, HHS pandemic planning scenarios are labeled across four quadrants (A, B, C and D). From left to right, the scenarios are “seasonal range,” “moderate pandemic,” “severe pandemic” and “very severe pandemic.” As clinical severity increases along the horizontal axis, or as transmissibility increases along the vertical axis, the severity of the pandemic planning scenario also increases." class="figure mx-auto d-block"><div class="figcaption">HHS Pandemic Planning Scenarios based on the
Pandemic Severity Assessment Framework. This uses a combined measure of
clinical severity and transmissibility to characterise influenza
pandemic scenarios. <strong>HHS</strong>: United States Department of
Health and Human Services (<a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>).</div>
</figure><p>One epidemiological approach to estimating the clinical severity is
quantifying the Case Fatality Risk (CFR). CFR is the conditional
probability of death given confirmed diagnosis, calculated as the
cumulative number of deaths from an infectious disease over the number
of confirmed diagnosed cases. However, calculating this directly during
the course of an epidemic tends to result in a naive or biased CFR given
the time <a href="reference.html#delaydist">delay</a> from onset to
death, varying substantially as the epidemic progresses and stabilising
at the later stages of the outbreak (<a href="https://academic.oup.com/aje/article/162/5/479/82647?login=false#620743" class="external-link">Ghani
et al., 2005</a>).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_c.png" alt="The periods are relevant: Period 1 -- 15 days where CFR is zero to indicate this is due to no reported deaths; Period from Mar 15 -- Apr 26 where CFR appears to be rising; Period Apr 30 -- May 30 where the CFR estimate stabilises." class="figure mx-auto d-block"><div class="figcaption">Observed biased confirmed case fatality risk
(CFR) estimates as a function of time (thick line) calculated as the
cumulative number of deaths over confirmed cases at time t. The estimate
at the end of an outbreak (~May 30) is the realised CFR by the end of
the epidemic. The horizontal continuous line and dotted lines show the
expected value and the 95% confidence intervals (<span class="math inline">\(95\%\)</span> CI) of the predicted delay-adjusted
CFR estimate only by using the observed data until 27 Mar 2003 (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>The periods are relevant: Period 1 – 15 days where CFR is zero to
indicate this is due to no reported deaths; Period from Mar 15 – Apr 26
where CFR appears to be rising; Period Apr 30 – May 30 where the CFR
estimate stabilises.</p>
</div>
</div>
</div>
</div>
<p>More generally, estimating severity can be helpful even outside of a
pandemic planning scenario and in the context of routine public health.
Knowing whether an outbreak has or had a different severity from the
historical record can motivate causal investigations, which could be
intrinsic to the infectious agent (e.g., a new, more severe strain) or
due to underlying factors in the population (e.g. reduced immunity or
morbidity factors) (<a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a>).</p>
<p>In this tutorial we are going to learn how to use the
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package to calculate and adjust a CFR estimation
using <a href="reference.html#delaydist">delay distributions</a> from
<code>{epiparameter}</code> or elsewhere, based on the methods developed
by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>, also, how we can reuse <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> functions
for more severity measurements.</p>
<p>We’ll use the pipe <code>%&gt;%</code> operator to connect functions,
so let’s also call to the <a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a> package:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span></code></pre>
</div>
</section><section id="data-sources-for-clinical-severity"><h2 class="section-heading">Data sources for clinical severity<a class="anchor" aria-label="anchor" href="#data-sources-for-clinical-severity"></a>
</h2>
<hr class="half-width">
<p>What are data sources can we use to estimate the clinical severity of
a disease outbreak? <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext" class="external-link">Verity
et al., 2020</a> summarises the spectrum of COVID-19 cases:</p>
<figure><img src="../fig/cfr-spectrum-cases-covid19.jpg" alt="Spectrum of COVID-19 cases. The CFR aims to estimate the proportion of Deaths among confirmed cases in an epidemic. (Verity et al., 2020)" class="figure mx-auto d-block"><div class="figcaption">Spectrum of COVID-19 cases. The CFR aims to
estimate the proportion of Deaths among confirmed cases in an epidemic.
(<a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext#gr1" class="external-link">Verity
et al., 2020</a>)</div>
</figure><ul>
<li>At the top of the pyramid, those who met the WHO case criteria for
<strong>severe</strong> or critical cases would likely have been
identified in the hospital setting, presenting with atypical viral
pneumonia. These cases would have been identified in mainland China and
among those categorised internationally as local transmission.</li>
<li>Many more cases are likely to be <strong>symptomatic</strong> (i.e.,
with fever, cough, or myalgia) but might not require hospitalisation.
These cases would have been identified through links to international
travel to high-risk areas and through contact-tracing of contacts of
confirmed cases. They might be identifiable through population
surveillance of, for example, influenza-like illness.</li>
<li>The bottom part of the pyramid represents <strong>mild</strong> (and
possibly <strong>asymptomatic</strong>) cases. These cases might be
identifiable through contact tracing and subsequently via serological
testing.</li>
</ul></section><section id="naive-cfr"><h2 class="section-heading">Naive CFR<a class="anchor" aria-label="anchor" href="#naive-cfr"></a>
</h2>
<hr class="half-width">
<p>We measure disease severity in terms of case fatality risk (CFR). The
CFR is interpreted as the conditional probability of death given
confirmed diagnosis, calculated as the cumulative number of deaths <span class="math inline">\(D_{t}\)</span> over the cumulative number of
confirmed cases <span class="math inline">\(C_{t}\)</span> at a certain
time <span class="math inline">\(t\)</span>. We can refer to the
<em>naive CFR</em> (also crude or biased CFR, <span class="math inline">\(b_{t}\)</span>):</p>
<p><span class="math display">\[ b_{t} =  \frac{D_{t}}{C_{t}}
\]</span></p>
<p>This calculation is <em>naive</em> because it tends to yield a biased
and mostly underestimated CFR due to the time-delay from onset to death,
only stabilising at the later stages of the outbreak.</p>
<!-- add here the callout on ratio or risk?  -->
<!-- https://github.com/epiverse-trace/cfr/issues/130 -->
<p>To calculate the naive CFR, the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package requires
an input data frame with three columns named:</p>
<ul>
<li><code>date</code></li>
<li><code>cases</code></li>
<li><code>deaths</code></li>
</ul>
<p>Let’s explore the <code>ebola1976</code> dataset, included in {cfr},
which comes from the first Ebola outbreak in what was then called Zaire
(now the Democratic Republic of the Congo) in 1976, as analysed by
Camacho et al. (2014).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Ebola 1976 data provided with the {cfr} package</span></span>
<span><span class="fu">data</span><span class="op">(</span><span class="st">"ebola1976"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assume we only have the first 30 days of this data</span></span>
<span><span class="va">ebola1976</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 30 × 3
   date       cases deaths
   &lt;date&gt;     &lt;int&gt;  &lt;int&gt;
 1 1976-08-25     1      0
 2 1976-08-26     0      0
 3 1976-08-27     0      0
 4 1976-08-28     0      0
 5 1976-08-29     0      0
 6 1976-08-30     0      0
 7 1976-08-31     0      0
 8 1976-09-01     1      0
 9 1976-09-02     1      0
10 1976-09-03     1      0
# ℹ 20 more rows</code></pre>
</div>
<div id="we-need-aggregated-incidence-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="we-need-aggregated-incidence-data" class="callout-inner">
<h3 class="callout-title">We need aggregated incidence data<a class="anchor" aria-label="anchor" href="#we-need-aggregated-incidence-data"></a>
</h3>
<div class="callout-content">
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> reads <strong>aggregated</strong> incidence
data.</p>
<!-- Similar to the `{EpiNow2}` with the difference that for `{cfr}` we need one more column named `deaths`. -->
<p>This data input should be <strong>aggregated</strong> by day, which
means one observation <em>per day</em>, containing the <em>daily</em>
number of reported cases and deaths. Observations with zero or missing
values should also be included, similar to time-series data.</p>
<p>Also, <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> currently works for <em>daily</em> data
only, but not for other temporal units of data aggregation, e.g.,
weeks.</p>
<!-- suggest ways to deal with raw input weekly data -->
<!-- https://github.com/epiverse-trace/cfr/issues/117 -->
</div>
</div>
</div>
<p>When we apply <code>cfr_static()</code> to <code>data</code>
directly, we are calculating the naive CFR:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the naive CFR for the first 30 days</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1     0.4740741    0.3875497     0.5617606</code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Download the file <a href="../data/sarscov2_cases_deaths.csv">sarscov2_cases_deaths.csv</a> and
read it into R.</p>
<p>Estimate the naive CFR.</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" aria-labelledby="headingHint1" data-bs-parent="#accordionHint1">
<div class="accordion-body">
<p>Inspect the format of the data input.</p>
<ul>
<li>Does it contain daily data?</li>
<li>Does the column names are as required by
<code>cfr_static()</code>?</li>
<li>How would you rename column names from a data frame?</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>We read the data input using <code>readr::read_csv()</code>. This
function recognize that the column <code>date</code> is a
<code>&lt;date&gt;</code> class vector.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co"># e.g.: if path to file is data/raw-data/ebola_cases.csv then:</span></span>
<span><span class="va">sarscov2_input</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu">read_csv</span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"sarscov2_cases_deaths.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Inspect data</span></span>
<span><span class="va">sarscov2_input</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 93 × 3
   date       cases_jpn deaths_jpn
   &lt;date&gt;         &lt;dbl&gt;      &lt;dbl&gt;
 1 2020-01-20         1          0
 2 2020-01-21         1          0
 3 2020-01-22         0          0
 4 2020-01-23         1          0
 5 2020-01-24         1          0
 6 2020-01-25         3          0
 7 2020-01-26         3          0
 8 2020-01-27         4          0
 9 2020-01-28         6          0
10 2020-01-29         7          0
# ℹ 83 more rows</code></pre>
</div>
<p>We can use <code>dplyr::rename()</code> to adapt the external data to
fit the data input for <code>cfr_static()</code>.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Rename before Estimate naive CFR</span></span>
<span><span class="va">sarscov2_input</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">cases_jpn</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">deaths_jpn</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr_static</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1    0.01895208   0.01828832    0.01963342</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="biases-that-affect-cfr-estimation"><h2 class="section-heading">Biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#biases-that-affect-cfr-estimation"></a>
</h2>
<hr class="half-width">
<div id="two-biases-that-affect-cfr-estimation" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="two-biases-that-affect-cfr-estimation" class="callout-inner">
<h3 class="callout-title">Two biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#two-biases-that-affect-cfr-estimation"></a>
</h3>
<div class="callout-content">
<p><a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a> describe two potential biases that can affect the
estimation of CFR (and their potential solutions):</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">1. Preferential ascertainment of severe cases</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>For diseases with a <em>spectrum</em> of clinical presentations,
those cases that come to the attention of public health authorities and
registered into surveillance databases will typically be people with the
most severe symptoms who seek medical care, are admitted to a hospital,
or die.</p>
<p>Therefore, the CFR will typically be higher among <em>detected
cases</em> than among the entire population of cases, given that the
latter may include individuals with mild, subclinical, and (under some
definitions of “case”) asymptomatic presentations.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">2. Bias due to delayed reporting of death</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>During an <em>ongoing</em> epidemic, there is a delay between the
time someone dies and the time their death is reported. Therefore, at
any given moment in time, the list of cases includes people who will die
and whose death has not yet occurred or has occurred but not yet been
reported. Thus, dividing the cumulative number of reported deaths by the
cumulative number of reported cases at a specific time point during an
outbreak will underestimate the true CFR.</p>
<p>The key determinants of the magnitude of the bias are the epidemic
<em>growth rate</em> and the <em>distribution of delays</em> from
case-reporting to death-reporting; the longer the delays and the faster
the growth rate, the greater the bias.</p>
<p>In this tutorial episode, we are going to focus on solutions to deal
with this specific bias using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>!</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Case study: Influenza A (H1N1), Mexico, 2009</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>Improving an <em>early</em> epidemiological assessment of a
delay-adjusted CFR is crucial for determining virulence, shaping the
level and choices of public health intervention, and providing advice to
the general public.</p>
<p>In 2009, during the swine-flu virus, Influenza A (H1N1), Mexico had
an early biased estimation of the CFR. Initial reports from the
government of Mexico suggested a virulent infection, whereas, in other
countries, the same virus was perceived as mild (<a href="https://content.time.com/time/health/article/0,8599,1894534,00.html" class="external-link">TIME,
2009</a>).</p>
<p>In the USA and Canada, no deaths were attributed to the virus in the
first ten days following the World Health Organization’s declaration of
a public health emergency. Even under similar circumstances at the early
stage of the global pandemic, public health officials, policymakers and
the general public want to know the virulence of an emerging infectious
agent.</p>
<p><a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a> reinterpreted the data assessing the biases and getting
a clinical severity lower than the 1918 influenza pandemic but
comparable with that seen in the 1957 pandemic.</p>
</div>
</div>
</div>
</div>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" aria-labelledby="headingInstructor2" data-bs-parent="#accordionInstructor2">
<div class="accordion-body">
<p>We can showcase this last bias using the <a href="https://epiverse-trace.github.io/cfr/articles/cfr.html#concept-how-reporting-delays-bias-cfr-estimates" class="external-link">concept
described in this <code>{cfr}</code> vignette</a>.</p>
<!-- create code and then a .gif? -->
</div>
</div>
</div>
</div>
</section><section id="delay-adjusted-cfr"><h2 class="section-heading">Delay-adjusted CFR<a class="anchor" aria-label="anchor" href="#delay-adjusted-cfr"></a>
</h2>
<hr class="half-width">
<p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> developed a method that considers the <strong>time
delay</strong> from the onset of symptoms to death.</p>
<p>Real-time outbreaks may have a number of deaths that are insufficient
to determine the time distribution between onset and death. Therefore,
we can estimate the <em>distribution delay</em> from historical
outbreaks or reuse the ones accessible via R packages like
<code>{epiparameter}</code> or <code>{epireview}</code>, which collect
them from published scientific literature. For an step-by-step guide,
read the tutorial episode on how to <a href="https://epiverse-trace.github.io/tutorials-early/delays-reuse.html" class="external-link">access
to epidemiological delays</a>.</p>
<p>Let’s use <code>{epiparameter}</code>:</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get delay distribution</span></span>
<span><span class="va">onset_to_death_ebola</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"Ebola"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset_to_death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot &lt;epidist&gt; object</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, day_range <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To calculate the delay-adjusted CFR, we can use the
<code>cfr_static()</code> function with the <code>data</code> and
<code>delay_density</code> arguments.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the delay-adjusted CFR</span></span>
<span><span class="co"># for the first 30 days</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">]</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.955         0.74             1</code></pre>
</div>
<p>The delay-adjusted CFR indicated that the overall disease severity
<em>at the end of the outbreak</em> or with the <em>latest data
available at the moment</em> is 0.955 with a 95% confidence interval
between 0.74 and 1, slightly higher than the naive one.</p>
<div id="use-the-epidist-class" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="use-the-epidist-class" class="callout-inner">
<h3 class="callout-title">Use the epidist class<a class="anchor" aria-label="anchor" href="#use-the-epidist-class"></a>
</h3>
<div class="callout-content">
<p>When using an <code>&lt;epidist&gt;</code> class object we can use
this expression as a template:</p>
<p><code>function(x) density(&lt;EPIDIST_OBJECT&gt;, x)</code></p>
<p>For distribution functions with parameters not available in
<code>{epiparameter}</code>, we suggest you two alternatives:</p>
<ul>
<li><p>Create an <code>&lt;epidist&gt;</code> class object, to plug into
other R packages of the outbreak analytics pipeline. Read the <a href="https://epiverse-trace.github.io/epiparameter/reference/epidist.html" class="external-link">reference
documentation of <code>epiparameter::epidist()</code></a>, or</p></li>
<li><p>Read <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette for <a href="https://epiverse-trace.github.io/cfr/articles/delay_distributions.html" class="external-link">a
primer on working with delay distributions</a>.</p></li>
</ul>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge2"></a>
</h3>
<div class="callout-content">
<p>Use the same file from the previous challenge (<a href="../data/sarscov2_cases_deaths.csv">sarscov2_cases_deaths.csv</a>).</p>
<p>Estimate the delay-adjusted CFR using the appropriate distribution
delay. Then:</p>
<ul>
<li>Compare the naive and the delay-adjusted CFR solutions!</li>
</ul>
</div>
</div>
</div>
<div id="accordionHint2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint2" aria-expanded="false" aria-controls="collapseHint2">
  <h4 class="accordion-header" id="headingHint2">Give me a hint</h4>
</button>
<div id="collapseHint2" class="accordion-collapse collapse" aria-labelledby="headingHint2" data-bs-parent="#accordionHint2">
<div class="accordion-body">
<ul>
<li>Find the appropriate <code>&lt;epidist&gt;</code> object!</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<p>We use <code>{epiparameter}</code> to access a delay distribution for
the SARS-CoV-2 aggregated incidence data:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sarscov2_delay</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>We read the data input using <code>readr::read_csv()</code>. This
function recognize that the column <code>date</code> is a
<code>&lt;date&gt;</code> class vector.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co"># e.g.: if path to file is data/raw-data/ebola_cases.csv then:</span></span>
<span><span class="va">sarscov2_input</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu">read_csv</span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"sarscov2_cases_deaths.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Inspect data</span></span>
<span><span class="va">sarscov2_input</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 93 × 3
   date       cases_jpn deaths_jpn
   &lt;date&gt;         &lt;dbl&gt;      &lt;dbl&gt;
 1 2020-01-20         1          0
 2 2020-01-21         1          0
 3 2020-01-22         0          0
 4 2020-01-23         1          0
 5 2020-01-24         1          0
 6 2020-01-25         3          0
 7 2020-01-26         3          0
 8 2020-01-27         4          0
 9 2020-01-28         6          0
10 2020-01-29         7          0
# ℹ 83 more rows</code></pre>
</div>
<p>We can use <code>dplyr::rename()</code> to adapt the external data to
fit the data input for <code>cfr_static()</code>.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Rename before Estimate naive CFR</span></span>
<span><span class="va">sarscov2_input</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">cases_jpn</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">deaths_jpn</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr_static</span><span class="op">(</span>delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">sarscov2_delay</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.055        0.054         0.057</code></pre>
</div>
<p>Interpret the comparison between the naive and delay-adjusted CFR
estimates.</p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>When to use discrete distributions?</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p>For <code>cfr_static()</code> and all the <code>cfr_*()</code> family
of functions, the most appropriate choice to pass are
<strong>discrete</strong> distributions. This is because we will work
with daily case and death data.</p>
<p>We can assume that evaluating the Probability Distribution Function
(PDF) of a <em>continuous</em> distribution is equivalent to the
Probability Mass Function (PMF) of the equivalent <em>discrete</em>
distribution.</p>
<p>However, this assumption may not be appropriate for distributions
with larger peaks. For instance, diseases with an onset-to-death
distribution that is strongly peaked with a low variance. In such cases,
the average disparity between the PDF and PMF is expected to be more
pronounced compared to distributions with broader spreads. One way to
deal with this is to discretise the continuous distribution using
<code>epiparameter::discretise()</code> to an
<code>&lt;epidist&gt;</code> object.</p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>How does {cfr} works?</h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<p>To adjust the CFR, <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> use the case and death incidence data to estimate the
number of cases with known outcomes:</p>
<p><span class="math display">\[
  u_t = \dfrac{\sum_{i = 0}^t
        \sum_{j = 0}^\infty c_{i - j} f_{j}}{\sum_{i = 0} c_i},
\]</span></p>
<p>where:</p>
<ul>
<li>
<span class="math inline">\(c_{t}\)</span> is the daily case
incidence at time <span class="math inline">\(t\)</span>,</li>
<li>
<span class="math inline">\(f_{t}\)</span> is the value of the
Probability Mass Function (PMF) of the <strong>delay
distribution</strong> between onset and death, and</li>
<li>
<span class="math inline">\(u_{t}\)</span> represents the
underestimation factor of the known outcomes.</li>
</ul>
<p><span class="math inline">\(u_{t}\)</span> is used to
<strong>scale</strong> the value of the cumulative number of cases in
the denominator in the calculation of the CFR. This is calculated
internally with the <a href="https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html" class="external-link"><code>estimate_outcomes()</code></a>
function.</p>
<p>The estimator for CFR can be written as:</p>
<p><span class="math display">\[p_{t} = \frac{b_{t}}{u_{t}}\]</span></p>
<p>where <span class="math inline">\(p_{t}\)</span> is the realized
proportion of confirmed cases to die from the infection (or the unbiased
CFR), and <span class="math inline">\(b_{t}\)</span>, the crude and
biased estimate of CFR (also naive CFR).</p>
<p>From this last equation, we observe that the unbiased CFR <span class="math inline">\(p_{t}\)</span> is larger than biased CFR <span class="math inline">\(b_{t}\)</span> because in <span class="math inline">\(u_{t}\)</span> the numerator is smaller than the
denominator (note that <span class="math inline">\(f_{t}\)</span> is the
probability distribution of the <em>delay distribution</em> between
onset and death). Therefore, we refer to <span class="math inline">\(b_{t}\)</span> as the biased estimator of CFR.</p>
<p>When we observe the entire course of an epidemic (from <span class="math inline">\(t \rightarrow \infty\)</span>), <span class="math inline">\(u_{t}\)</span> tends to 1, making <span class="math inline">\(b_{t}\)</span> tends to <span class="math inline">\(p_{t}\)</span> and become an unbiased estimator
(<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>).</p>
</div>
</div>
</div>
</div>
</section><section id="an-early-stage-cfr-estimate"><h2 class="section-heading">An early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#an-early-stage-cfr-estimate"></a>
</h2>
<hr class="half-width">
<p>On the challenge above, we discovered that the <em>naive</em> and
<em>delay-adjusted</em> CFR estimates are different.</p>
<p>The <strong>naive</strong> estimate is useful to get an overall
severity estimate of the outbreak (so far). Once the outbreak has ended
or has progressed such that more deaths are reported, the estimated CFR
is then <em>closest to</em> the ‘true’ unbiased CFR.</p>
<p>On the other hand, the <strong>delay-adjusted</strong> estimate can
assess the severity of an emerging infectious disease <em>earlier</em>
than the biased or naive CFR, during an epidemic.</p>
<p>We can explore the <strong>early</strong> determination of the
<em>delay-adjusted CFR</em> using the <code>cfr_rolling()</code>
function.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout3"></a>
</h3>
<div class="callout-content">
<p><code>cfr_rolling()</code> is a utility function that automatically
calculates CFR for each day of the outbreak with the data available up
to that day, saving the user time.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily naive CFR</span></span>
<span><span class="co"># for all the 73 days in the Ebola dataset</span></span>
<span><span class="va">rolling_cfr_naive</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_naive</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31     0.9510204    0.9160061     0.9744387
69 1976-11-01     0.9510204    0.9160061     0.9744387
70 1976-11-02     0.9510204    0.9160061     0.9744387
71 1976-11-03     0.9510204    0.9160061     0.9744387
72 1976-11-04     0.9510204    0.9160061     0.9744387
73 1976-11-05     0.9551020    0.9210866     0.9773771</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily delay-adjusted CFR</span></span>
<span><span class="co"># for all the 73 days in the Ebola dataset</span></span>
<span><span class="va">rolling_cfr_adjusted</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_adjusted</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31         0.975        0.856             1
69 1976-11-01         0.975        0.856             1
70 1976-11-02         0.971        0.852             1
71 1976-11-03         0.971        0.852             1
72 1976-11-04         0.971        0.852             1
73 1976-11-05         0.971        0.852             1</code></pre>
</div>
<p>With <code>utils::tail()</code>, we show that the latest CFR
estimates. The naive and delay-adjusted estimates have overlapping
ranges of 95% confidence intervals.</p>
<p>Now, let’s visualise both results in a time series. How would the
naive and delay-adjusted CFR estimates perform in real time?</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the latest delay-adjusted CFR</span></span>
<span><span class="va">rolling_cfr_adjusted_end</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">slice_tail</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># bind by rows both output data frames</span></span>
<span><span class="fu">bind_rows</span><span class="op">(</span></span>
<span>  <span class="va">rolling_cfr_naive</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"naive"</span><span class="op">)</span>,</span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"adjusted"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># visualise both adjusted and unadjusted rolling estimates</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_ribbon</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span></span>
<span>      <span class="va">date</span>,</span>
<span>      ymin <span class="op">=</span> <span class="va">severity_low</span>,</span>
<span>      ymax <span class="op">=</span> <span class="va">severity_high</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">method</span></span>
<span>    <span class="op">)</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span><span class="va">date</span>, <span class="va">severity_mean</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_mean</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_low</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_high</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The horizontal line represents the delay-adjusted CFR estimated at
the outbreak’s end. The dotted line means the estimate has a 95%
confidence interval (95% CI).</p>
<p><strong>Notice</strong> that this delay-adjusted calculation is
particularly useful when an <em>epidemic curve of confirmed cases</em>
is the only data available (i.e. when individual data from onset to
death are unavailable, especially during the early stage of the
epidemic). When there are few deaths or none at all, an assumption has
to be made for the <em>delay distribution</em> from onset to death,
e.g. from literature based on previous outbreaks. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> depict this in the figures with data from the SARS
outbreak in Hong Kong, 2003.</p>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>Case study: SARS outbreak, Hong Kong, 2003</h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" aria-labelledby="headingSpoiler3" data-bs-parent="#accordionSpoiler3">
<div class="accordion-body">
<p>Figures A and B show the cumulative numbers of cases and deaths of
SARS, and Figure C shows the observed (biased) CFR estimates as a
function of time, i.e. the cumulative number of deaths over cases at
time <span class="math inline">\(t\)</span>. Due to the delay from the
onset of symptoms to death, the biased estimate of CFR at time <span class="math inline">\(t\)</span> underestimates the realised CFR at the
end of an outbreak (i.e. 302/1755 = 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_abc.png" alt="Observed (biased) confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Observed (biased) confirmed case fatality risk
of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><p>Nevertheless, even by only using the observed data for the period
March 19 to April 2, <code>cfr_static()</code> can yield an appropriate
prediction (Figure D), e.g. the delay-adjusted CFR at March 27 is 18.1 %
(95% CI: 10.5, 28.1). An overestimation is seen in the very early stages
of the epidemic, but the 95% confidence limits in the later stages
include the realised CFR (i.e. 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_d.png" alt="Early determination of the delay-adjusted confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Early determination of the delay-adjusted
confirmed case fatality risk of severe acute respiratory syndrome (SARS)
in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure>
</div>
</div>
</div>
</div>
<div id="interpret-the-early-stage-cfr-estimate" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="interpret-the-early-stage-cfr-estimate" class="callout-inner">
<h3 class="callout-title">Interpret the early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#interpret-the-early-stage-cfr-estimate"></a>
</h3>
<div class="callout-content">
<p>Based on the figure above:</p>
<ul>
<li>How much difference in days is between the date in which the 95% CI
of the estimated <em>delay-adjusted CFR</em> vs <em>naive CFR</em> cross
with the CFR estimated at the end of the outbreak?</li>
</ul>
<p>Discuss:</p>
<ul>
<li>What are the Public health policy implications of having a
<em>delay-adjusted CFR</em> estimate?</li>
</ul>
</div>
</div>
</div>
<div id="accordionHint3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint3" aria-expanded="false" aria-controls="collapseHint3">
  <h4 class="accordion-header" id="headingHint3">Give me a hint</h4>
</button>
<div id="collapseHint3" class="accordion-collapse collapse" aria-labelledby="headingHint3" data-bs-parent="#accordionHint3">
<div class="accordion-body">
<p>We can use either visual inspection or analysis of the output data
frames.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6">Show me the solution</h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>There is almost one month of difference.</p>
<p>Note that the estimate has considerable uncertainty at the beginning
of the time series. After two weeks, the delay-adjusted CFR approaches
the overall CFR estimate at the outbreak’s end.</p>
<p>Is this pattern similar to other outbreaks? We can use the data sets
in this episode’s challenges. We invite you to find it out!</p>
</div>
</div>
</div>
</div>
<div id="checklist" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="checklist" class="callout-inner">
<h3 class="callout-title">Checklist<a class="anchor" aria-label="anchor" href="#checklist"></a>
</h3>
<div class="callout-content">
<p>With <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we estimate the CFR as the proportion of
deaths among <strong>confirmed</strong> cases.</p>
<p>By only using <strong>confirmed</strong> cases, it is clear that all
cases that do not seek medical treatment or are not notified will be
missed, as well as all asymptomatic cases. This means that the CFR
estimate is higher than the proportion of deaths among the infected.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7">Why the naive and delay-adjusted differ?</h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> method aims to obtain an unbiased estimator “well
before” observing the entire course of the outbreak. For this,
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> uses the underestimation factor <span class="math inline">\(u_{t}\)</span> to estimate the unbiased CFR <span class="math inline">\(p_{t}\)</span> using maximum-likelihood methods,
given the <em>sampling process</em> defined by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8">What is the sampling process?</h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" aria-labelledby="headingSolution8" data-bs-parent="#accordionSolution8">
<div class="accordion-body">
<figure><img src="../fig/cfr-pone.0006852.g001.png" alt="The population of confirmed cases and sampling process for estimating the unbiased CFR during the course of an outbreak. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">The population of confirmed cases and sampling
process for estimating the unbiased CFR during the course of an
outbreak. (<a href="https://doi.org/10.1371/journal.pone.0006852.g001" class="external-link">Nishiura et
al., 2009</a>)</div>
</figure><p>From <em>aggregated incidence data</em>, at time <span class="math inline">\(t\)</span> we know the cumulative number of
confirmed cases and deaths, <span class="math inline">\(C_{t}\)</span>
and <span class="math inline">\(D_{t}\)</span>, and wish to estimate the
unbiased CFR <span class="math inline">\(\pi\)</span>, by way of the
factor of underestimation <span class="math inline">\(u_{t}\)</span>.</p>
<p>If we knew the factor of underestimation <span class="math inline">\(u_{t}\)</span> we could specify the size of the
population of confirmed cases no longer at risk (<span class="math inline">\(u_{t}C_{t}\)</span>, <strong>shaded</strong>),
although we do not know which surviving individuals belong to this
group. A proportion <span class="math inline">\(\pi\)</span> of those in
the group of cases still at risk (size <span class="math inline">\((1-
u_{t})C_{t}\)</span>, <strong>unshaded</strong>) is expected to die.</p>
<p>Because each case no longer at risk had an independent probability of
dying, <span class="math inline">\(\pi\)</span>, the number of deaths,
<span class="math inline">\(D_{t}\)</span>, is a sample from a binomial
distribution with sample size <span class="math inline">\(u_{t}C_{t}\)</span>, and probability of dying
<span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>.</p>
<p>This is represented by the following likelihood function to obtain
the maximum likelihood estimate of the unbiased CFR <span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>:</p>
<p><span class="math display">\[
  {\sf L}(\pi | C_{t},D_{t},u_{t}) = \log{\dbinom{u_{t}C_{t}}{D_{t}}} +
D_{t} \log{\pi} +
  (u_{t}C_{t} - D_{t})\log{(1 - \pi)},
\]</span></p>
<p>This estimation is performed by the internal function
<code>?cfr:::estimate_severity()</code>.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9">Limitations</h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" aria-labelledby="headingSolution9" data-bs-parent="#accordionSolution9">
<div class="accordion-body">
<ul>
<li>The delay-adjusted CFR does not address all sources of error in data
like the underdiagnosis of infected individuals.</li>
</ul>
</div>
</div>
</div>
</div>
</section><section id="challenges"><h2 class="section-heading">Challenges<a class="anchor" aria-label="anchor" href="#challenges"></a>
</h2>
<hr class="half-width">
<div id="aggregated-data-differ-from-linelists" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="aggregated-data-differ-from-linelists" class="callout-inner">
<h3 class="callout-title">Aggregated data differ from linelists<a class="anchor" aria-label="anchor" href="#aggregated-data-differ-from-linelists"></a>
</h3>
<div class="callout-content">
<p><em>Aggregated</em> incidence data differs from
<strong>linelist</strong> data, where each observation contains
individual-level data.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_sierraleone_2014</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 11,903 × 8
      id   age sex   status    date_of_onset date_of_sample district chiefdom   
   &lt;int&gt; &lt;dbl&gt; &lt;fct&gt; &lt;fct&gt;     &lt;date&gt;        &lt;date&gt;         &lt;fct&gt;    &lt;fct&gt;      
 1     1    20 F     confirmed 2014-05-18    2014-05-23     Kailahun Kissi Teng 
 2     2    42 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Teng 
 3     3    45 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Tonge
 4     4    15 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 5     5    19 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 6     6    55 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 7     7    50 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 8     8     8 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
 9     9    54 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
10    10    57 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
# ℹ 11,893 more rows</code></pre>
</div>
</div>
</div>
</div>
<div id="use-incidence2-to-rearrange-your-data" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="use-incidence2-to-rearrange-your-data" class="callout-inner">
<h3 class="callout-title">Use incidence2 to rearrange your data<a class="anchor" aria-label="anchor" href="#use-incidence2-to-rearrange-your-data"></a>
</h3>
<div class="callout-content">
<p>From the <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package, load the MERS linelist of
cases from the <code>mers_korea_2015</code> object.</p>
<p>Rearrange your this linelist to fit into the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>
package input.</p>
<p>Estimate the delay-adjusted HFR using the corresponding distribution
delay.</p>
</div>
</div>
</div>
<div id="accordionHint4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint4" aria-expanded="false" aria-controls="collapseHint4">
  <h4 class="accordion-header" id="headingHint4">Give me a hint</h4>
</button>
<div id="collapseHint4" class="accordion-collapse collapse" aria-labelledby="headingHint4" data-bs-parent="#accordionHint4">
<div class="accordion-body">
<p><strong>How to rearrange my input data?</strong></p>
<p>Rearranging the input data for data analysis can take most of the
time. To get ready-to-analyse <em>aggregated incidence data</em>, we
encourage you to use <a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a>!</p>
<p>First, in the <a href="https://www.reconverse.org/incidence2/articles/incidence2.html" class="external-link">Get
started</a> vignette from the <a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a> package, explore
how to use the <code>date_index</code> argument when reading a linelist
with dates in multiple column.</p>
<p>Then, refer to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette on <a href="https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html" class="external-link">Handling
data from <code>{incidence2}</code></a> on how to use the
<code>cfr::prepare_data()</code> function from incidence2 objects.</p>
<!-- cite howto entry one lineslist + incidence2 + cfr connection -->
</div>
</div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10">Show me the solution</h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" aria-labelledby="headingSolution10" data-bs-parent="#accordionSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load packages</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">incidence2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access delay distribution</span></span>
<span><span class="va">mers_delay</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"mers"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Read linelist</span></span>
<span><span class="va">mers_korea_2015</span><span class="op">$</span><span class="va">linelist</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"dt_"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 162 × 6
   dt_onset   dt_report  dt_start_exp dt_end_exp dt_diag    dt_death  
   &lt;date&gt;     &lt;date&gt;     &lt;date&gt;       &lt;date&gt;     &lt;date&gt;     &lt;date&gt;    
 1 2015-05-11 2015-05-19 2015-04-18   2015-05-04 2015-05-20 NA        
 2 2015-05-18 2015-05-20 2015-05-15   2015-05-20 2015-05-20 NA        
 3 2015-05-20 2015-05-20 2015-05-16   2015-05-16 2015-05-21 2015-06-04
 4 2015-05-25 2015-05-26 2015-05-16   2015-05-20 2015-05-26 NA        
 5 2015-05-25 2015-05-27 2015-05-17   2015-05-17 2015-05-26 NA        
 6 2015-05-24 2015-05-28 2015-05-15   2015-05-17 2015-05-28 2015-06-01
 7 2015-05-21 2015-05-28 2015-05-16   2015-05-17 2015-05-28 NA        
 8 2015-05-26 2015-05-29 2015-05-15   2015-05-15 2015-05-29 NA        
 9 NA         2015-05-29 2015-05-15   2015-05-17 2015-05-29 NA        
10 2015-05-21 2015-05-29 2015-05-16   2015-05-16 2015-05-29 NA        
# ℹ 152 more rows</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use {incidence2} to count daily incidence</span></span>
<span><span class="va">mers_incidence</span> <span class="op">&lt;-</span> <span class="va">mers_korea_2015</span><span class="op">$</span><span class="va">linelist</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># converto to incidence2 object</span></span>
<span>  <span class="fu">incidence</span><span class="op">(</span>date_index <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"dt_onset"</span>, <span class="st">"dt_death"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># complete dates from first to last</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="fu">complete_dates</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Inspect incidence2 output</span></span>
<span><span class="va">mers_incidence</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># incidence:  72 x 3
# count vars: dt_death, dt_onset
   date_index count_variable count
   &lt;date&gt;     &lt;chr&gt;          &lt;int&gt;
 1 2015-05-11 dt_death           0
 2 2015-05-11 dt_onset           1
 3 2015-05-12 dt_death           0
 4 2015-05-12 dt_onset           0
 5 2015-05-13 dt_death           0
 6 2015-05-13 dt_onset           0
 7 2015-05-14 dt_death           0
 8 2015-05-14 dt_onset           0
 9 2015-05-15 dt_death           0
10 2015-05-15 dt_onset           0
# ℹ 62 more rows</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare data from {incidence2} to {cfr}</span></span>
<span><span class="va">mers_incidence</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">prepare_data</span><span class="op">(</span></span>
<span>    cases_variable <span class="op">=</span> <span class="st">"dt_onset"</span>,</span>
<span>    deaths_variable <span class="op">=</span> <span class="st">"dt_death"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date deaths cases
1  2015-05-11      0     1
2  2015-05-12      0     0
3  2015-05-13      0     0
4  2015-05-14      0     0
5  2015-05-15      0     0
6  2015-05-16      0     0
7  2015-05-17      0     1
8  2015-05-18      0     1
9  2015-05-19      0     0
10 2015-05-20      0     5
11 2015-05-21      0     6
12 2015-05-22      0     2
13 2015-05-23      0     4
14 2015-05-24      0     2
15 2015-05-25      0     3
16 2015-05-26      0     1
17 2015-05-27      0     2
18 2015-05-28      0     1
19 2015-05-29      0     3
20 2015-05-30      0     5
21 2015-05-31      0    10
22 2015-06-01      2    16
23 2015-06-02      0    11
24 2015-06-03      1     7
25 2015-06-04      1    12
26 2015-06-05      1     9
27 2015-06-06      0     7
28 2015-06-07      0     7
29 2015-06-08      2     6
30 2015-06-09      0     1
31 2015-06-10      2     6
32 2015-06-11      1     3
33 2015-06-12      0     0
34 2015-06-13      0     2
35 2015-06-14      0     0
36 2015-06-15      0     1</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate delay-adjusted CFR</span></span>
<span><span class="va">mers_incidence</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">prepare_data</span><span class="op">(</span></span>
<span>    cases_variable <span class="op">=</span> <span class="st">"dt_onset"</span>,</span>
<span>    deaths_variable <span class="op">=</span> <span class="st">"dt_death"</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr_static</span><span class="op">(</span>delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">mers_delay</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.137        0.069          0.24</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="severity-heterogeneity" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="severity-heterogeneity" class="callout-inner">
<h3 class="callout-title">Severity heterogeneity<a class="anchor" aria-label="anchor" href="#severity-heterogeneity"></a>
</h3>
<div class="callout-content">
<p>The CFR may differ across populations (e.g. age, space, treatment);
quantifying these heterogeneities can help target resources
appropriately and compare different care regimens (<a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371" class="external-link">Cori
et al., 2017</a>).</p>
<p>Use the <code>cfr::covid_data</code> data frame to estimate a
delay-adjusted CFR stratified by country.</p>
</div>
</div>
</div>
<div id="accordionHint5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint5" aria-expanded="false" aria-controls="collapseHint5">
  <h4 class="accordion-header" id="headingHint5">Give me a hint</h4>
</button>
<div id="collapseHint5" class="accordion-collapse collapse" aria-labelledby="headingHint5" data-bs-parent="#accordionHint5">
<div class="accordion-body">
<p>One way to do a <em>stratified analysis</em> is to apply a model to
nested data. This <a href="https://tidyr.tidyverse.org/articles/nest.html#nested-data-and-models" class="external-link"><code>{tidyr}</code>
vignette</a> shows you how to apply the <code>group_by()</code> +
<code>nest()</code> to nest data, and then <code>mutate()</code> +
<code>map()</code> to apply the model.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11">Show me the solution</h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" aria-labelledby="headingSolution11" data-bs-parent="#accordionSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 20,786
Columns: 4
$ date    &lt;date&gt; 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2…
$ country &lt;chr&gt; "Argentina", "Brazil", "Colombia", "France", "Germany", "India…
$ cases   &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ deaths  &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_onset_death</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    temp <span class="op">=</span></span>
<span>      <span class="fu">map</span><span class="op">(</span></span>
<span>        .x <span class="op">=</span> <span class="va">data</span>,</span>
<span>        .f <span class="op">=</span> <span class="fu">cfr</span><span class="fu">::</span><span class="va">cfr_static</span>,</span>
<span>        delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">delay_onset_death</span>, <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 19 × 5
# Groups:   country [19]
   country        data                 severity_mean severity_low severity_high
   &lt;chr&gt;          &lt;list&gt;                       &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
 1 Argentina      &lt;tibble [1,094 × 3]&gt;         0.013        0.013         0.013
 2 Brazil         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 3 Colombia       &lt;tibble [1,094 × 3]&gt;         0.022        0.022         0.022
 4 France         &lt;tibble [1,094 × 3]&gt;         0.004        0.004         0.004
 5 Germany        &lt;tibble [1,094 × 3]&gt;         0.005        0.005         0.005
 6 India          &lt;tibble [1,094 × 3]&gt;         0.012        0.012         0.012
 7 Indonesia      &lt;tibble [1,094 × 3]&gt;         0.024        0.024         0.024
 8 Iran           &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 9 Italy          &lt;tibble [1,094 × 3]&gt;         0.007        0.007         0.007
10 Mexico         &lt;tibble [1,094 × 3]&gt;         0.046        0.046         0.046
11 Peru           &lt;tibble [1,094 × 3]&gt;         0.05         0.05          0.05 
12 Poland         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
13 Russia         &lt;tibble [1,094 × 3]&gt;         0.018        0.018         0.018
14 South Africa   &lt;tibble [1,094 × 3]&gt;         0.025        0.025         0.025
15 Spain          &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
16 Turkey         &lt;tibble [1,094 × 3]&gt;         0.006        0.006         0.006
17 Ukraine        &lt;tibble [1,094 × 3]&gt;         0.02         0.02          0.02 
18 United Kingdom &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
19 United States  &lt;tibble [1,094 × 3]&gt;         0.011        0.011         0.011</code></pre>
</div>
<p>Great! Now you can use similar code for any other stratified analysis
like age, regions or more!</p>
<p>But, how can we interpret that there is a country variability of
severity from the same diagnosed pathogen?</p>
<p>Local factors like testing capacity, the case definition, and
sampling regime can affect the report of cases and deaths, thus
affecting case ascertainment. Take a look to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>
vignette on <a href="https://epiverse-trace.github.io/cfr/articles/estimate_ascertainment.html" class="external-link">Estimating
the proportion of cases that are ascertained during an outbreak</a>!</p>
</div>
</div>
</div>
</div>
</section><section id="appendix"><h2 class="section-heading">Appendix<a class="anchor" aria-label="anchor" href="#appendix"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package has a function called
<code>cfr_time_varying()</code> with functionality that differs from
<code>cfr_rolling()</code>.</p>
<div id="when-to-use-cfr_rolling" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="when-to-use-cfr_rolling" class="callout-inner">
<h3 class="callout-title">When to use cfr_rolling()?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_rolling"></a>
</h3>
<div class="callout-content">
<p><strong>cfr_rolling()</strong> shows the estimated CFR on each
outbreak day, given that future data on cases and deaths is unavailable
at the time. The final value of <em>cfr_rolling()</em> estimates is
identical to <em>cfr_static()</em> on the same data.</p>
<p>Remember, as shown above, <em>cfr_rolling()</em> is helpful to get
early-stage CFR estimates and check whether an outbreak’s CFR estimate
has stabilised. Thus, <em>cfr_rolling()</em> is not sensitive to the
length or size of the epidemic.</p>
</div>
</div>
</div>
<div id="when-to-use-cfr_time_varying" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="when-to-use-cfr_time_varying" class="callout-inner">
<h3 class="callout-title">When to use cfr_time_varying()?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_time_varying"></a>
</h3>
<div class="callout-content">
<p>On the other hand, <strong>cfr_time_varying()</strong> calculates the
CFR over a moving window and helps to understand changes in CFR due to
changes in the epidemic, e.g. due to a new variant or increased immunity
from vaccination.</p>
<p>However, <em>cfr_time_varying()</em> is sensitive to sampling
uncertainty. Thus, it is sensitive to the size of the outbreak. The
higher the number of cases with expected outcomes on a given day, the
more reasonable estimates of the time-varying CFR we will get.</p>
<p>For example, with 100 cases, the fatality risk estimate will, roughly
speaking, have a 95% confidence interval ±10% of the mean estimate
(binomial CI). So if we have &gt;100 cases with expected outcomes <em>on
a given day</em>, we can get reasonable estimates of the time varying
CFR. But if we only have &gt;100 cases <em>over the course of the whole
epidemic</em>, we probably need to rely on
<strong>cfr_rolling()</strong> that uses the cumulative data.</p>
<p>We invite you to read this <a href="https://epiverse-trace.github.io/cfr/articles/estimate_time_varying_severity.html" class="external-link">vignette
about the <code>cfr_time_varying()</code> function</a>.</p>
</div>
</div>
</div>
<div id="more-severity-measures" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="more-severity-measures" class="callout-inner">
<h3 class="callout-title">More severity measures<a class="anchor" aria-label="anchor" href="#more-severity-measures"></a>
</h3>
<div class="callout-content">
<p>Suppose we need to assess the clinical severity of the epidemic in a
context different from surveillance data, like the severity among cases
that arrive at hospitals or cases you collected from a representative
serological survey.</p>
<p>Using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we can change the inputs for the numerator
(<code>cases</code>) and denominator (<code>deaths</code>) to estimate
more severity measures like the Infection fatality risk (IFR) or the
Hospitalisation Fatality Risk (HFR). We can follow this analogy:</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12">Infection and Hospitalization fatality risk</h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" aria-labelledby="headingSolution12" data-bs-parent="#accordionSolution12">
<div class="accordion-body">
<p>If for a <em>Case</em> fatality risk (CFR), we require:</p>
<ul>
<li>
<em>case</em> and death incidence data, with a</li>
<li>case-to-death delay distribution (or close approximation, such as
symptom onset-to-death).</li>
</ul>
<p>Then, the <em>Infection</em> fatality risk (IFR) requires:</p>
<ul>
<li>
<em>infection</em> and death incidence data, with an</li>
<li>exposure-to-death delay distribution (or close approximation).</li>
</ul>
<p>Similarly, the <em>Hospitalisation</em> Fatality Risk (HFR)
requires:</p>
<ul>
<li>
<em>hospitalisation</em> and death incidence data, and a</li>
<li>hospitalisation-to-death delay distribution.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="accordionSolution13" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution13" aria-expanded="false" aria-controls="collapseSolution13">
  <h4 class="accordion-header" id="headingSolution13">Data sources for more severity measures</h4>
</button>
<div id="collapseSolution13" class="accordion-collapse collapse" aria-labelledby="headingSolution13" data-bs-parent="#accordionSolution13">
<div class="accordion-body">
<p><a href="https://www.nature.com/articles/s41467-020-19238-2/figures/1" class="external-link">Yang
et al., 2020</a> summarises different definitions and data sources:</p>
<figure><img src="../fig/cfr-s41467-020-19238-2-fig_a.png" alt="Severity levels of infections with SARS-CoV-2 and parameters of interest. Each level is assumed to be a subset of the level below." class="figure mx-auto d-block"><div class="figcaption">Severity levels of infections with SARS-CoV-2
and parameters of interest. Each level is assumed to be a subset of the
level below.</div>
</figure><ul>
<li>sCFR symptomatic case-fatality risk,</li>
<li>sCHR symptomatic case-hospitalisation risk,</li>
<li>mCFR medically attended case-fatality risk,</li>
<li>mCHR medically attended case-hospitalisation risk,</li>
<li>HFR hospitalisation-fatality risk.</li>
</ul>
<figure><img src="../fig/cfr-s41467-020-19238-2-fig_b.png" alt="Data source of COVID-19 cases in Wuhan: D1) 32,583 laboratory-confirmed COVID-19 cases as of March 84, D2) 17,365 clinically-diagnosed COVID-19 cases during February 9–194, D3)daily number of laboratory-confirmed cases on March 9–April 243, D4) total number of COVID-19 deaths as of April 24 obtained from the Hubei Health Commission3, D5) 325 laboratory-confirmed cases and D6) 1290 deaths were added as of April 16 through a comprehensive and systematic verification by Wuhan Authorities3, and D7) 16,781 laboratory-confirmed cases identified through universal screening10,11. Pse: RT-PCR sensitivity12. Pmed.care: proportion of seeking medical assistance among patients suffering from acute respiratory infections13." class="figure mx-auto d-block"><div class="figcaption">Schematic diagram of the baseline analyses. Red,
blue, and green arrows denote the data flow from laboratory-confirmed
cases of passive surveillance, clinically-diagnosed cases, and
laboratory-confirmed cases of active screenings.</div>
</figure>
</div>
</div>
</div>
</div>
<div id="discussion5" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion5"></a>
</h3>
<div class="callout-content">
<p>Based on your experience:</p>
<ul>
<li>Share any previous outbreak in which you participated in its
response.</li>
</ul>
<p>Answer to these questions:</p>
<ul>
<li>How did you assess the clinical severity of the outbreak?</li>
<li>What were the primary sources of bias?</li>
<li>What did you do to take into account the identified bias?</li>
<li>What complementary analysis would you do to solve the bias?</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li><p>Use <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> to estimate severity</p></li>
<li><p>Use <code>cfr_static()</code> to estimate the overall CFR with
the latest data available.</p></li>
<li><p>Use <code>cfr_rolling()</code> to show what the estimated CFR
would be on each day of the outbreak.</p></li>
<li><p>Use the <code>delay_density</code> argument to adjust the CFR by
the corresponding delay distribution.</p></li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-superspreading-estimate"><p>Content from <a href="superspreading-estimate.html">Account for superspreading</a></p>
<hr>
<p>Last updated on 2024-05-01 |
        
        <a href="https://github.com/epiverse-trace/tutorials-middle/edit/main/episodes/superspreading-estimate.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 32 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can we estimate individual-level variation in transmission
(i.e. superspreading potential) from contact tracing data?</li>
<li>What are the implications for variation in transmission for
decision-making?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Estimate the distribution of onward transmission from infected
individuals (i.e. offspring distribution) from outbreak data using
<a href="https://www.repidemicsconsortium.org/epicontacts/" class="external-link">epicontacts</a>.</li>
<li>Estimate the extent of individual-level variation (i.e. the
dispersion parameter) of the offspring distribution using
<a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a>.</li>
<li>Estimate the proportion of transmission that is linked to
‘superspreading events’ using <code>{superspreading}</code>.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>Learners should familiarise themselves with following concept
dependencies before working through this tutorial:</p>
<p><strong>Statistics</strong>: common probability distributions,
particularly Poisson and negative binomial.</p>
<p><strong>Epidemic theory</strong>: The reproduction number, R.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<!-- we know --><p>From smallpox to severe acute respiratory syndrome coronavirus 2
(SARS-CoV-2), some infected individuals spread infection to more people
than others. Disease transmission is the result of a combination of
biological and social factors, and these factors average out to some
extent at the population level during a large epidemic. Hence
researchers often use population averages to assess the potential for
disease to spread. However, in the earlier or later phases of an
outbreak, individual differences in infectiousness can be more
important. In particular, they increase the chance of superspreading
events (SSEs), which can ignite explosive epidemics and also influence
the chances of controlling transmission (<a href="https://wellcomeopenresearch.org/articles/5-83" class="external-link">Lloyd-Smith et
al., 2005</a>).</p>
<figure><img src="../fig/see-intro-superspreading.png" alt="Chains of SARS-CoV-2 transmission in Hong Kong initiated by local or imported cases. (a), Transmission network of a cluster of cases traced back to a collection of four bars across Hong Kong (n = 106). (b), Transmission network associated with a wedding without clear infector–infectee pairs but linked back to a preceding social gathering and local source (n = 22). (c), Transmission network associated with a temple cluster of undetermined source (n = 19). (d), All other clusters of SARS-CoV-2 infections where the source and transmission chain could be determined (Adam et al., 2020)." class="figure mx-auto d-block"><div class="figcaption">
<strong>Chains of SARS-CoV-2 transmission in
Hong Kong initiated by local or imported cases.</strong>
(<strong>a</strong>), Transmission network of a cluster of cases traced
back to a collection of four bars across Hong Kong (n = 106).
(<strong>b</strong>), Transmission network associated with a wedding
without clear infector–infectee pairs but linked back to a preceding
social gathering and local source (n = 22). (<strong>c</strong>),
Transmission network associated with a temple cluster of undetermined
source (n = 19). (<strong>d</strong>), All other clusters of SARS-CoV-2
infections where the source and transmission chain could be determined
(<a href="https://www.nature.com/articles/s41591-020-1092-0" class="external-link">Adam et
al., 2020</a>).</div>
</figure><!-- we dont know --><p>The <a href="reference.html#basic">basic reproduction number</a>,
<span class="math inline">\(R_{0}\)</span>, measures the average number
of cases caused by one infectious individual in a entirely susceptible
population. Estimates of <span class="math inline">\(R_{0}\)</span> are
useful for understanding the average dynamics of an epidemic at the
population-level, but can obscure considerable individual variation in
infectiousness. This was highlighted during the global emergence of
SARS-CoV-2 by numerous ‘superspreading events’ in which certain
infectious individuals generated unusually large numbers of secondary
cases (<a href="https://wellcomeopenresearch.org/articles/5-83" class="external-link">LeClerc
et al, 2020</a>).</p>
<figure><img src="../fig/see-intro-secondary-cases-fig-b.png" alt="R = 0.58 and k = 0.43." class="figure mx-auto d-block"><div class="figcaption">
<strong>Observed offspring distribution of
SARS-CoV-2 transmission in Hong Kong.</strong> N = 91 SARS-CoV-2
infectors, N = 153 terminal infectees and N = 46 sporadic local cases.
Histogram bars indicate the proportion of onward transmission per amount
of secondary cases. Line corresponds to a fitted negative binomial
distribution (<a href="https://www.nature.com/articles/s41591-020-1092-0" class="external-link">Adam et al.,
2020</a>).</div>
</figure><!-- we want --><p>In this tutorial, we are going to quantify individual variation in
transmission, and hence estimate the potential for superspreading
events. Then we are going to use these estimates to explore the
implications of superspreading for contact tracing interventions.</p>
<p>We are going to use data from the <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package,
manage the linelist and contacts data using <a href="https://www.repidemicsconsortium.org/epicontacts/" class="external-link">epicontacts</a>,
and estimate distribution parameters with <a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a>.
Lastly, we are going to use <code>{superspreading}</code> to explore the
implications of variation in transmission for decision-making.</p>
<p>We’ll use the pipe <code>%&gt;%</code> to connect some of the
functions from these packages, so let’s also call the
<a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a> package.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epicontacts</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">fitdistrplus</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">superspreading</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span></code></pre>
</div>
<div id="the-double-colon" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="the-double-colon" class="callout-inner">
<h3 class="callout-title">The double-colon<a class="anchor" aria-label="anchor" href="#the-double-colon"></a>
</h3>
<div class="callout-content">
<p>The double-colon <code>::</code> in R let you call a specific
function from a package without loading the entire package into the
current environment.</p>
<p>For example, <code>dplyr::filter(data, condition)</code> uses
<code>filter()</code> from the <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a> package.</p>
<p>This help us remember package functions and avoid namespace
conflicts.</p>
</div>
</div>
</div>
</section><section id="the-individual-reprodution-number"><h2 class="section-heading">The individual reprodution number<a class="anchor" aria-label="anchor" href="#the-individual-reprodution-number"></a>
</h2>
<hr class="half-width">
<p>The individual reproduction number is defined as the number of
secondary cases caused by a particular infected individual.</p>
<p>Early in an outbreak we can use contact data to reconstruct
transmission chains (i.e. who infected whom) and calculate the number of
secondary cases generated by each individual. This reconstruction of
linked transmission events from contact data can provide an
understanding about how different individuals have contributed to
transmission during an epidemic (<a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371" class="external-link">Cori
et al., 2017</a>).</p>
<p>Let’s practice this using the <code>mers_korea_2015</code> linelist
and contact data from the <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package and integrate
them with the <a href="https://www.repidemicsconsortium.org/epicontacts/" class="external-link">epicontacts</a> package to calculate the
distribution of secondary cases during the 2015 MERS-CoV outbreak in
South Korea (<a href="https://community.appliedepi.org/t/estimating-the-degree-of-super-spreading-from-transmission-chain-data/103/2" class="external-link">Campbell,
2022</a>):</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## first, make an epicontacts object</span></span>
<span><span class="va">epi_contacts</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epicontacts</span><span class="fu">::</span><span class="fu">make_epicontacts</span><span class="op">(</span></span>
<span>    linelist <span class="op">=</span> <span class="fu">outbreaks</span><span class="fu">::</span><span class="va">mers_korea_2015</span><span class="op">$</span><span class="va">linelist</span>,</span>
<span>    contacts <span class="op">=</span> <span class="fu">outbreaks</span><span class="fu">::</span><span class="va">mers_korea_2015</span><span class="op">$</span><span class="va">contacts</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># visualise contact network</span></span>
<span><span class="fu">epicontacts</span><span class="fu">::</span><span class="fu">vis_epicontacts</span><span class="op">(</span><span class="va">epi_contacts</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-estimate-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>Is contact data tidy?</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>Contact data from a transmission chain can provide information on
which infected individuals came into contact with others. We expect to
have the infector (<code>from</code>) and the infectee (<code>to</code>)
plus additional columns of variables related to their contact, such as
location (<code>exposure</code>) and date of contact.</p>
<p>Following <a href="https://tidyr.tidyverse.org/articles/tidy-data.html#tidy-data" class="external-link">tidy
data</a> principles, the observation unit in our contact dataset is the
<strong>infector-infectee</strong> pair. Although one infector can
infect multiple infectees, from contact tracing investigations we may
record contacts linked to more than one infector (e.g. within a
household). But we should expect to have unique infector-infectee pairs,
because typically each infected person will have acquired the infection
from one other.</p>
<p>To ensure these unique pairs, we can check on replicates for
infectees:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># no infector-infectee pairs are replicated</span></span>
<span><span class="va">epi_contacts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pluck</span><span class="op">(</span><span class="st">"contacts"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">to</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 5 × 4
# Groups:   to [2]
  from  to     exposure       diff_dt_onset
  &lt;chr&gt; &lt;chr&gt;  &lt;fct&gt;                  &lt;int&gt;
1 SK_16 SK_107 Emergency room            17
2 SK_87 SK_107 Emergency room             2
3 SK_14 SK_39  Hospital room             16
4 SK_11 SK_39  Hospital room             13
5 SK_12 SK_39  Hospital room             12</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>When each infector-infectee row is unique, the number of entries per
infector corresponds to the number of secondary cases generated by that
individual.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># count secondary cases per infector</span></span>
<span><span class="va">infector_secondary</span> <span class="op">&lt;-</span> <span class="va">epi_contacts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pluck</span><span class="op">(</span><span class="st">"contacts"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">from</span>, name <span class="op">=</span> <span class="st">"secondary_cases"</span><span class="op">)</span></span></code></pre>
</div>
<p>But this output only contains number of secondary cases for reported
infectors, not for each of the individuals in the whole
<code>epicontacts</code> object.</p>
<p>To get this, first, we can use <code>epicontacts::get_id()</code> to
get the full list of unique identifiers (“id”) from the
<code>epicontacts</code> class object. Second, join it with the count
secondary cases per infector stored in the
<code>infector_secondary</code> object. Third, replace the missing
values with <code>0</code> to express no report of secondary cases from
them.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_secondary</span> <span class="op">&lt;-</span> <span class="va">epi_contacts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># extract ids in contact *and* linelist using "which" argument</span></span>
<span>  <span class="fu">epicontacts</span><span class="fu">::</span><span class="fu">get_id</span><span class="op">(</span>which <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># transform vector to dataframe to use left_join()</span></span>
<span>  <span class="fu">enframe</span><span class="op">(</span>name <span class="op">=</span> <span class="cn">NULL</span>, value <span class="op">=</span> <span class="st">"from"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># join count secondary cases per infectee</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">infector_secondary</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># infectee with missing secondary cases are replaced with zero</span></span>
<span>  <span class="fu">replace_na</span><span class="op">(</span></span>
<span>    replace <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>secondary_cases <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>From a histogram of the <code>all_secondary</code> object, we can
identify the <strong>individual-level variation</strong> in the number
of secondary cases. Three cases were related to more than 20 secondary
cases, while the complementary cases with less than five or zero
secondary cases.</p>
<!-- Visualizing the number of secondary cases on a histogram will help us to relate this with the statistical distribution to fit: -->
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot the distribution</span></span>
<span><span class="va">all_secondary</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">secondary_cases</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Number of secondary cases"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Frequency"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-estimate-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The number of secondary cases can be used to <em>empirically</em>
estimate the <strong>offspring distribution</strong>, which is the
number of secondary <em>infections</em> caused by each case. One
candidate statistical distribution used to model the offspring
distribution is the <strong>negative binomial</strong> distribution with
two parameters:</p>
<ul>
<li><p><strong>Mean</strong>, which represents the <span class="math inline">\(R_{0}\)</span>, the average number of (secondary)
cases produced by a single individual in an entirely susceptible
population, and</p></li>
<li><p><strong>Dispersion</strong>, expressed as <span class="math inline">\(k\)</span>, which represents the individual-level
variation in transmission by single individuals.</p></li>
</ul>
<figure><img src="../fig/superspreading-estimate-rendered-unnamed-chunk-10-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the histogram and density plot, we can identify that the
offspring distribution is highly skewed or
<strong>overdispersed</strong>. In this framework, the superspreading
events (SSEs) are not arbitrary or exceptional, but simply realizations
from the right-hand tail of the offspring distribution, which we can
quantify and analyse (<a href="https://www.nature.com/articles/nature04153" class="external-link">Lloyd-Smith et al.,
2005</a>).</p>
<div id="terminology-recap" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="terminology-recap" class="callout-inner">
<h3 class="callout-title">Terminology recap<a class="anchor" aria-label="anchor" href="#terminology-recap"></a>
</h3>
<div class="callout-content">
<ul>
<li>From linelist and contact data, we calculate the <strong>number of
secondary cases</strong> caused by the observed infected
individuals.</li>
<li>Whereas <span class="math inline">\(R_{0}\)</span> captures the
average transmission in the population, we can define the
<strong>individual reproduction number</strong> as a random variable
representing the <em>expected</em> number of secondary cases caused by a
infected individual.</li>
<li>From the stochastic effects in transmission, the number of secondary
<em>infections</em> caused by each case is described by an
<strong>offspring distribution</strong>.</li>
<li>An <em>empirical</em> offspring distribution can be modeled by the
<strong>negative-binomial</strong> distribution with mean <span class="math inline">\(R_{0}\)</span> and dispersion parameter <span class="math inline">\(k\)</span>.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>Poisson, overdispersion, and Negative Binomial</h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler2" aria-labelledby="headingSpoiler2">
<div class="accordion-body">
<!-- distribution stories -->
<p>For occurrences of associated discrete events we can use
<strong>Poisson</strong> or negative binomial discrete
distributions.</p>
<p>In a Poisson distribution, mean is equal to variance. But when
variance is higher than the mean, this is called
<strong>overdispersion</strong>. In biological applications,
overdispersion occurs and so a negative binomial may be worth
considering as an alternative to Poisson distribution.</p>
<p><strong>Negative binomial</strong> distribution is specially useful
for discrete data over an unbounded positive range whose sample variance
exceeds the sample mean. In such terms, the observations are
overdispersed with respect to a Poisson distribution, for which the mean
is equal to the variance.</p>
<p>In epidemiology, <a href="https://en.wikipedia.org/wiki/Negative_binomial_distribution" class="external-link">negative
binomial</a> have being used to model disease transmission for
infectious diseases where the likely number of onward infections may
vary considerably from individual to individual and from setting to
setting, capturing all variation in infectious histories of individuals,
including properties of the biological (i.e. degree of viral shedding)
and environmental circumstances (e.g. type and location of contact).</p>
</div>
</div>
</div>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Calculate the distribution of secondary cases for Ebola using the
<code>ebola_sim_clean</code> object from <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a>
package.</p>
<p>Is the offspring distribution of Ebola skewed or overdispersed?</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" data-bs-parent="#accordionHint1" aria-labelledby="headingHint1">
<div class="accordion-body">
<p><strong>Note:</strong> This dataset has 5829 cases. Running
<code>epicontacts::vis_epicontacts()</code> may overload your
session!</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## first, make an epicontacts object</span></span>
<span><span class="va">ebola_contacts</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epicontacts</span><span class="fu">::</span><span class="fu">make_epicontacts</span><span class="op">(</span></span>
<span>    linelist <span class="op">=</span> <span class="va">ebola_sim_clean</span><span class="op">$</span><span class="va">linelist</span>,</span>
<span>    contacts <span class="op">=</span> <span class="va">ebola_sim_clean</span><span class="op">$</span><span class="va">contacts</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># count secondary cases</span></span>
<span></span>
<span><span class="va">ebola_infector_secondary</span> <span class="op">&lt;-</span> <span class="va">ebola_contacts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pluck</span><span class="op">(</span><span class="st">"contacts"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">from</span>, name <span class="op">=</span> <span class="st">"secondary_cases"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_secondary</span> <span class="op">&lt;-</span> <span class="va">ebola_contacts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># extract ids in contact *and* linelist using "which" argument</span></span>
<span>  <span class="fu">epicontacts</span><span class="fu">::</span><span class="fu">get_id</span><span class="op">(</span>which <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># transform vector to dataframe to use left_join()</span></span>
<span>  <span class="fu">enframe</span><span class="op">(</span>name <span class="op">=</span> <span class="cn">NULL</span>, value <span class="op">=</span> <span class="st">"from"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># join count secondary cases per infectee</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">ebola_infector_secondary</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># infectee with missing secondary cases are replaced with zero</span></span>
<span>  <span class="fu">replace_na</span><span class="op">(</span></span>
<span>    replace <span class="op">=</span> <span class="fu">list</span><span class="op">(</span>secondary_cases <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co">## plot the distribution</span></span>
<span><span class="va">ebola_secondary</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">secondary_cases</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Number of secondary cases"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Frequency"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-estimate-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From a visual inspection, the distribution of secondary cases for the
Ebola data set in <code>ebola_sim_clean</code> shows an skewed
distribution with secondary cases equal or lower than 6. We need to
complement this observation with a statistical analysis to evaluate for
overdispersion.</p>
</div>
</div>
</div>
</div>
</section><section id="estimate-the-dispersion-parameter"><h2 class="section-heading">Estimate the dispersion parameter<a class="anchor" aria-label="anchor" href="#estimate-the-dispersion-parameter"></a>
</h2>
<hr class="half-width">
<p>To empirically estimate the dispersion parameter <span class="math inline">\(k\)</span>, we could fit a negative binomial
distribution to the number of secondary cases.</p>
<p>We can fit distributions to data using the
<a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a> package, which provides maximum likelihood
estimates.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">fitdistrplus</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## fit distribution</span></span>
<span><span class="va">offspring_fit</span> <span class="op">&lt;-</span> <span class="va">all_secondary</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">secondary_cases</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">fitdist</span><span class="op">(</span>distr <span class="op">=</span> <span class="st">"nbinom"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">offspring_fit</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Fitting of the distribution ' nbinom ' by maximum likelihood 
Parameters:
       estimate  Std. Error
size 0.02039807 0.007278299
mu   0.60452947 0.337893199</code></pre>
</div>
<div id="name-of-parameters" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="name-of-parameters" class="callout-inner">
<h3 class="callout-title">Name of parameters<a class="anchor" aria-label="anchor" href="#name-of-parameters"></a>
</h3>
<div class="callout-content">
<p>From the <a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a> output:</p>
<ul>
<li>The <code>size</code> object refers to the estimated dispersion
parameter <span class="math inline">\(k\)</span>, and</li>
<li>The <code>mu</code> object refers to the estimated mean, which
represents the <span class="math inline">\(R_{0}\)</span>,</li>
</ul>
</div>
</div>
</div>
<p>From the number secondary cases distribution we estimated a
dispersion parameter <span class="math inline">\(k\)</span> of 0.02,
with a 95% Confidence Interval from 0.006 to 0.035. As the value of
<span class="math inline">\(k\)</span> is significantly lower than one,
we can conclude that there is considerable potential for superspreading
events.</p>
<p>We can overlap the estimated density values of the fitted negative
binomial distribution and the histogram of the number of secondary
cases:</p>
<figure><img src="../fig/superspreading-estimate-rendered-unnamed-chunk-14-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="individual-level-variation-in-transmission" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="individual-level-variation-in-transmission" class="callout-inner">
<h3 class="callout-title">Individual-level variation in
transmission<a class="anchor" aria-label="anchor" href="#individual-level-variation-in-transmission"></a>
</h3>
<div class="callout-content">
<p>The individual-level variation in transmission is defined by the
relationship between the mean (<span class="math inline">\(R_{0}\)</span>), dispersion (<span class="math inline">\(k\)</span>), and the variance of a negative
binomial distribution.</p>
<p>The negative binomial model has <span class="math inline">\(variance
= R_{0}(1+\frac{R_{0}}{k})\)</span>, so smaller values of <span class="math inline">\(k\)</span> indicate greater variance and,
consequently, greater <strong>individual-level variation</strong> in
transmission.</p>
<p><span class="math display">\[\uparrow variance =
R_{0}(1+\frac{R_{0}}{\downarrow k})\]</span></p>
<p>When <span class="math inline">\(k\)</span> approaches infinity
(<span class="math inline">\(k \rightarrow \infty\)</span>) the variance
equals the mean (because <span class="math inline">\(\frac{R_{0}}{\infty}=0\)</span>). This makes the
conventional Poisson model an special case of the negative binomial
model.</p>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge2"></a>
</h3>
<div class="callout-content">
<p>Use the distribution of secondary cases from the
<code>ebola_sim_clean</code> object from <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a>
package.</p>
<p>Fit a negative binomial distribution to estimate the mean and
dispersion parameter of the offspring distribution.</p>
<p>Does the estimated dispersion parameter of Ebola provide evidence of
an individual-level variation in transmission?</p>
</div>
</div>
</div>
<div id="accordionHint2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint2" aria-expanded="false" aria-controls="collapseHint2">
  <h4 class="accordion-header" id="headingHint2">Give me a hint</h4>
</button>
<div id="collapseHint2" class="accordion-collapse collapse" data-bs-parent="#accordionHint2" aria-labelledby="headingHint2">
<div class="accordion-body">
<p>Review how we fitted a negative binomial distribution using the
<code>fitdistrplus::fitdist()</code> function.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ebola_offspring</span> <span class="op">&lt;-</span> <span class="va">ebola_secondary</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">secondary_cases</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">fitdist</span><span class="op">(</span>distr <span class="op">=</span> <span class="st">"nbinom"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_offspring</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Fitting of the distribution ' nbinom ' by maximum likelihood 
Parameters:
     estimate  Std. Error
size 2.353899 0.250124606
mu   0.539300 0.009699219</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## extract the "size" parameter</span></span>
<span><span class="va">ebola_mid</span> <span class="op">&lt;-</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">estimate</span><span class="op">[[</span><span class="st">"size"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## calculate the 95% confidence intervals using the standard error estimate and</span></span>
<span><span class="co">## the 0.025 and 0.975 quantiles of the normal distribution.</span></span>
<span></span>
<span><span class="va">ebola_lower</span> <span class="op">&lt;-</span> <span class="va">ebola_mid</span> <span class="op">+</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">sd</span><span class="op">[[</span><span class="st">"size"</span><span class="op">]</span><span class="op">]</span> <span class="op">*</span> <span class="fu">qnorm</span><span class="op">(</span><span class="fl">0.025</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ebola_upper</span> <span class="op">&lt;-</span> <span class="va">ebola_mid</span> <span class="op">+</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">sd</span><span class="op">[[</span><span class="st">"size"</span><span class="op">]</span><span class="op">]</span> <span class="op">*</span> <span class="fu">qnorm</span><span class="op">(</span><span class="fl">0.975</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ebola_mid</span></span>
<span><span class="co"># ebola_lower</span></span>
<span><span class="co"># ebola_upper</span></span></code></pre>
</div>
<p>From the number secondary cases distribution we estimated a
dispersion parameter <span class="math inline">\(k\)</span> of 2.354,
with a 95% Confidence Interval from 1.864 to 2.844.</p>
<p>For dispersion parameter estimates higher than one we get low
distribution variance, hence, low individual-level variation in
transmission.</p>
<p>But does this mean that the secondary case distribution does not have
superspreading events (SSEs)? You will later find one additional
challenge: How do you define an SSE threshold for Ebola?</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Select the best model</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>We can use the maximum likelihood estimates from
<a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a> to compare different models and assess fit
performance using estimators like the AIC and BIC. Read further in the
vignette on <a href="https://epiverse-trace.github.io/superspreading/articles/estimate_individual_level_transmission.html" class="external-link">Estimate
individual-level transmission</a> and use the
<code>{superspreading}</code> helper function <code>ic_tbl()</code> for
this!</p>
</div>
</div>
</div>
</div>
<div id="the-dispersion-parameter-across-diseases" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="the-dispersion-parameter-across-diseases" class="callout-inner">
<h3 class="callout-title">The dispersion parameter across diseases<a class="anchor" aria-label="anchor" href="#the-dispersion-parameter-across-diseases"></a>
</h3>
<div class="callout-content">
<p>Research into sexually transmitted and vector-borne diseases has
previously suggested a ‘20/80’ rule, with 20% of individuals
contributing at least 80% of the transmission potential (<a href="https://www.pnas.org/doi/10.1073/pnas.94.1.338" class="external-link">Woolhouse et
al</a>).</p>
<p>On its own, the dispersion parameter <span class="math inline">\(k\)</span> is hard to interpret intuitively, and
hence converting into proportional summary can enable easier comparison.
When we consider a wider range of pathogens, we can see there is no hard
and fast rule for the percentage that generates 80% of transmission, but
variation does emerge as a common feature of infectious diseases</p>
<ul>
<li><p>When the 20% most infectious cases contribute to the 80% of
transmission (or more), there is a high individual-level variation in
transmission, with a highly overdispersed offspring distribution (<span class="math inline">\(k&lt;0.1\)</span>), e.g., SARS-1.</p></li>
<li><p>When the 20% most infectious cases contribute to the ~50% of
transmission, there is a low individual-level variation in transmission,
with a moderately dispersed offspring distribution (<span class="math inline">\(k &gt; 0.1\)</span>), e.g. Pneumonic
Plague.</p></li>
</ul>
<figure><img src="../fig/SEE-individual-reproductive-number-fig-c-d.png" alt="Evidence for variation in individual reproductive number. (Left, c) Proportion of transmission expected from the most infectious 20% of cases, for 10 outbreak or surveillance data sets (triangles). Dashed lines show proportions expected under the 20/80 rule (top) and homogeneity (bottom). (Right, d), Reported superspreading events (SSEs; diamonds) relative to estimated reproductive number R (squares) for twelve directly transmitted infections. Crosses show the 99th-percentile proposed as threshold for SSEs. (More figure details in Lloyd-Smith et al., 2005)" class="figure mx-auto d-block"><div class="figcaption">
<strong>Evidence for variation in individual
reproductive number</strong>. (Left, c) Proportion of transmission
expected from the most infectious 20% of cases, for 10 outbreak or
surveillance data sets (triangles). Dashed lines show proportions
expected under the 20/80 rule (top) and homogeneity (bottom). (Right,
d), Reported superspreading events (SSEs; diamonds) relative to
estimated reproductive number R (squares) for twelve directly
transmitted infections. Crosses show the 99th-percentile proposed as
threshold for SSEs. (More figure details in <a href="https://www.nature.com/articles/nature04153" class="external-link">Lloyd-Smith et al.,
2005</a>)</div>
</figure>
</div>
</div>
</div>
</section><section id="controlling-superspreading-with-contact-tracing"><h2 class="section-heading">Controlling superspreading with contact tracing<a class="anchor" aria-label="anchor" href="#controlling-superspreading-with-contact-tracing"></a>
</h2>
<hr class="half-width">
<p>During an outbreak, it is common to try and reduce transmission by
identifying people who have come into contact with an infected person,
then quarantine them in case they subsequently turn out to be infected.
Such contact tracing can be deployed in multiple ways. ‘Forward’ contact
tracing targets downstream contacts who may have been infected by a
newly identifed infection (i.e. the ‘index case’). ‘Backward’ tracing
instead tracks the upstream primary case who infected the index case (or
a setting or event at which the index case was infected), for example by
retracing history of contact to the likely point of exposure. This makes
it possible to identify others who were also potentially infected by
this earlier primary case.</p>
<p>In the presence of individual-level variation in transmission, i.e.,
with an overdispersed offspring distribution, if this primary case is
identified, a larger fraction of the transmission chain can be detected
by forward tracing each of the contacts of this primary case (<a href="https://wellcomeopenresearch.org/articles/5-239/v3" class="external-link">Endo et al.,
2020</a>).</p>
<figure><img src="../fig/contact-tracing-strategies.png" alt="Schematic representation of contact tracing strategies. Black arrows indicate the directions of transmission, blue and Orange arrows, a successful or failed contact tracing, respectivelly. When there is evidence of individual-level variation in transmission, often resulting in superspreading, backward contact tracing from the index case (blue circle) increase the probability to find the primary case (green circle) or clusters with a larger fraction of cases, potentially increasing the number of quarentined cases (yellow circles). Claire Blackmore, 2021" class="figure mx-auto d-block"><div class="figcaption">Schematic representation of contact tracing
strategies. Black arrows indicate the directions of transmission, blue
and Orange arrows, a successful or failed contact tracing,
respectivelly. When there is evidence of individual-level variation in
transmission, often resulting in superspreading, backward contact
tracing from the index case (blue circle) increase the probability to
find the primary case (green circle) or clusters with a larger fraction
of cases, potentially increasing the number of quarentined cases (yellow
circles). <a href="https://www.paho.org/sites/default/files/backward_contact_tracing_v3_0.pdf" class="external-link">Claire
Blackmore, 2021</a>
</div>
</figure><p>When there is evidence of individual-level variation
(i.e. overdispersion), often resulting in so-called superspreading
events, a large proportion of infections may be linked to a small
proportion of original clusters. As a result, finding and targeting
originating clusters in combination with reducing onwards infection may
substantially enhance the effectiveness of tracing methods (<a href="https://wellcomeopenresearch.org/articles/5-239/v3" class="external-link">Endo et al.,
2020</a>).</p>
<p>Empirical evidence focused on evaluating the efficiency of backward
tracing lead to 42% more cases identified than forward tracing
supporting its implementation when rigorous suppression of transmission
is justified (<a href="https://www.nature.com/articles/s41467-022-32531-6" class="external-link">Raymenants et
al., 2022</a>)</p>
</section><section id="probability-of-cases-in-a-given-cluster"><h2 class="section-heading">Probability of cases in a given cluster<a class="anchor" aria-label="anchor" href="#probability-of-cases-in-a-given-cluster"></a>
</h2>
<hr class="half-width">
<p>Using <code>{superspreading}</code>, we can estimate the probability
of having a cluster of secondary infections caused by a primary case
identified by backward tracing of size <span class="math inline">\(X\)</span> or larger (<a href="https://wellcomeopenresearch.org/articles/5-239/v3" class="external-link">Endo et al.,
2020</a>).</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set seed for random number generator</span></span>
<span><span class="fu">set.seed</span><span class="op">(</span><span class="fl">33</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># estimate the probability of</span></span>
<span><span class="co"># having a cluster size of 5, 10, or 25</span></span>
<span><span class="co"># secondary cases from a primary case,</span></span>
<span><span class="co"># given known reproduction number and</span></span>
<span><span class="co"># dispersion parameter.</span></span>
<span><span class="fu">proportion_cluster_size</span><span class="op">(</span></span>
<span>  R <span class="op">=</span> <span class="va">offspring_fit</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"mu"</span><span class="op">]</span>,</span>
<span>  k <span class="op">=</span> <span class="va">offspring_fit</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"size"</span><span class="op">]</span>,</span>
<span>  cluster_size <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">5</span>, <span class="fl">10</span>, <span class="fl">25</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          R          k prop_5 prop_10 prop_25
1 0.6045295 0.02039807  87.9%   74.6%   46.1%</code></pre>
</div>
<p>Even though we have an <span class="math inline">\(R&lt;1\)</span>, a
highly overdispersed offspring distribution (<span class="math inline">\(k=0.02\)</span>) means that if we detect a new
case, there is a 46.1% probability they originated from a cluster of 25
infections or more. Hence, by following a backwards strategy, contact
tracing efforts will increase the probability of successfully contain
and quarantining this large number of earlier infected individuals,
rather than simply focusing on the new case, who is likely to have
infected nobody (because <span class="math inline">\(k\)</span> is very
small).</p>
<p>We can also use this number to prevent gathering of certain sized to
reduce the epidemic by preventing potential superspreading events.
Interventions can target to reduce the reproduction number in order to
reduce the probability of having clusters of secondary cases.</p>
<div id="backward-contact-tracing-for-ebola" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="backward-contact-tracing-for-ebola" class="callout-inner">
<h3 class="callout-title">Backward contact tracing for Ebola<a class="anchor" aria-label="anchor" href="#backward-contact-tracing-for-ebola"></a>
</h3>
<div class="callout-content">
<p>Use the Ebola estimated parameters for <code>ebola_sim_clean</code>
object from <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package.</p>
<p>Calculate the probability of having a cluster of secondary infections
caused by a primary case identified by backward tracing of size 5, 10,
15 or larger.</p>
<p>Would implementing a backward strategy at this stage of the Ebola
outbreak increase the probability of containing and quarantining more
onward cases?</p>
</div>
</div>
</div>
<div id="accordionHint3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint3" aria-expanded="false" aria-controls="collapseHint3">
  <h4 class="accordion-header" id="headingHint3">Give me a hint</h4>
</button>
<div id="collapseHint3" class="accordion-collapse collapse" data-bs-parent="#accordionHint3" aria-labelledby="headingHint3">
<div class="accordion-body">
<p>Review how we estimated the probability of having clusters of a fixed
size, given an offspring distribution mean and dispersion parameters,
using the <code>superspreading::proportion_cluster_size()</code>
function.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the probability of</span></span>
<span><span class="co"># having a cluster size of 5, 10, or 25</span></span>
<span><span class="co"># secondary cases from a primary case,</span></span>
<span><span class="co"># given known reproduction number and</span></span>
<span><span class="co"># dispersion parameter.</span></span>
<span><span class="fu">proportion_cluster_size</span><span class="op">(</span></span>
<span>  R <span class="op">=</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"mu"</span><span class="op">]</span>,</span>
<span>  k <span class="op">=</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"size"</span><span class="op">]</span>,</span>
<span>  cluster_size <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">5</span>, <span class="fl">10</span>, <span class="fl">25</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>       R        k prop_5 prop_10 prop_25
1 0.5393 2.353899   1.8%      0%      0%</code></pre>
</div>
<p>The probability of having clusters of five people is 1.8%. At this
stage, given this offspring distribution parameters, a backward strategy
may not increase the probability of contain and quarantine more onward
cases.</p>
</div>
</div>
</div>
</div>
</section><section id="challenges"><h2 class="section-heading">Challenges<a class="anchor" aria-label="anchor" href="#challenges"></a>
</h2>
<hr class="half-width">
<div id="does-ebola-have-any-superspreading-event" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="does-ebola-have-any-superspreading-event" class="callout-inner">
<h3 class="callout-title">Does Ebola have any superspreading event?<a class="anchor" aria-label="anchor" href="#does-ebola-have-any-superspreading-event"></a>
</h3>
<div class="callout-content">
<p>‘Superspreading events’ can mean different things to different
people, so <a href="https://www.nature.com/articles/nature04153" class="external-link">Lloyd-Smith et al.,
2005</a> proposed a general protocol for defining a superspreading event
(SSE). If the number of secondary infections caused by each case, <span class="math inline">\(Z\)</span>, follows a negative binomial
distribution (<span class="math inline">\(R, k\)</span>):</p>
<ul>
<li>We define an SSE as any infected individual who infects more than
<span class="math inline">\(Z(n)\)</span> others, where <span class="math inline">\(Z(n)\)</span> is the nth percentile of the <span class="math inline">\(Poisson(R)\)</span> distribution.</li>
<li>A 99th-percentile SSE is then any case causing more infections than
would occur in 99% of infectious histories in a homogeneous
population</li>
</ul>
<p>Using the corresponding distribution function, estimate the SSE
threshold to define a SSE for the Ebola offspring distribution estimates
for the <code>ebola_sim_clean</code> object from
<a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package.</p>
</div>
</div>
</div>
<div id="accordionHint4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint4" aria-expanded="false" aria-controls="collapseHint4">
  <h4 class="accordion-header" id="headingHint4">Give me a hint</h4>
</button>
<div id="collapseHint4" class="accordion-collapse collapse" data-bs-parent="#accordionHint4" aria-labelledby="headingHint4">
<div class="accordion-body">
<p>In a Poisson distribution, the <strong>lambda</strong> or
<strong>rate</strong> parameter are equal to the estimated
<strong>mean</strong> from a negative binomial distribution. You can
explore this in <a href="https://ben18785.shinyapps.io/distribution-zoo/" class="external-link">The distribution
zoo</a> shiny app.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<p>To get the quantile value for the 99th-percentile we need to use the
<a href="https://sakai.unc.edu/access/content/group/3d1eb92e-7848-4f55-90c3-7c72a54e7e43/public/docs/lectures/lecture13.htm#probfunc" class="external-link">density
function</a> for the Poisson distribution <code>dpois()</code>.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get mean</span></span>
<span><span class="va">ebola_mu_mid</span> <span class="op">&lt;-</span> <span class="va">ebola_offspring</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"mu"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># get 99th-percentile from poisson distribution</span></span>
<span><span class="co"># with mean equal to mu</span></span>
<span><span class="fu">qpois</span><span class="op">(</span></span>
<span>  p <span class="op">=</span> <span class="fl">0.99</span>,</span>
<span>  lambda <span class="op">=</span> <span class="va">ebola_mu_mid</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 3</code></pre>
</div>
<p>Compare this values with the ones reported by <a href="https://www.nature.com/articles/nature04153" class="external-link">Lloyd-Smith et al.,
2005</a>. See figure below:</p>
<figure><img src="../fig/SEE-individual-reproductive-number-fig-d.png" alt="Reported superspreading events (SSEs; diamonds) relative to estimated reproduction number R (squares) for twelve directly transmitted infections. Lines show 5–95 percentile range of the number of secondary cases following a Poisson distribution with lambda equal to the reproduction number (Z∼Poisson(R)), and crosses show the 99th-percentile proposed as threshold for SSEs. Stars represent SSEs caused by more than one source case. ‘Other’ diseases are: 1, Streptococcus group A; 2, Lassa fever; 3, Mycoplasma pneumonia; 4, pneumonic plague; 5, tuberculosis. R is not shown for ‘other’ diseases, and is off-scale for monkeypox." class="figure mx-auto d-block"><div class="figcaption">Reported superspreading events (SSEs; diamonds)
relative to estimated reproduction number R (squares) for twelve
directly transmitted infections. Lines show 5–95 percentile range of the
number of secondary cases following a Poisson distribution with lambda
equal to the reproduction number (<span class="math inline">\(Z∼Poisson(R)\)</span>), and crosses show the
99th-percentile proposed as threshold for SSEs. Stars represent SSEs
caused by more than one source case. ‘Other’ diseases are: 1,
Streptococcus group A; 2, Lassa fever; 3, Mycoplasma pneumonia; 4,
pneumonic plague; 5, tuberculosis. R is not shown for ‘other’ diseases,
and is off-scale for monkeypox.</div>
</figure>
</div>
</div>
</div>
</div>
<div id="expected-proportion-of-transmission" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="expected-proportion-of-transmission" class="callout-inner">
<h3 class="callout-title">Expected proportion of transmission<a class="anchor" aria-label="anchor" href="#expected-proportion-of-transmission"></a>
</h3>
<div class="callout-content">
<p>What is the proportion of cases responsible for 80% of
transmission?</p>
<p>Use <code>{superspreading}</code> and compare the estimates for
<strong>MERS</strong> using the offspring distributions parameters from
this tutorial episode, with <strong>SARS-1</strong> and
<strong>Ebola</strong> offspring distributions parameter accessible via
the <code>{epiparameter}</code> R package.</p>
</div>
</div>
</div>
<div id="accordionHint5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint5" aria-expanded="false" aria-controls="collapseHint5">
  <h4 class="accordion-header" id="headingHint5">Give me a hint</h4>
</button>
<div id="collapseHint5" class="accordion-collapse collapse" data-bs-parent="#accordionHint5" aria-labelledby="headingHint5">
<div class="accordion-body">
<p>To use <code>superspreading::proportion_transmission()</code> we
recommend to read the <a href="https://epiverse-trace.github.io/superspreading/reference/proportion_transmission.html" class="external-link">Estimate
what proportion of cases cause a certain proportion of transmission</a>
reference manual.</p>
<p>Currently, <code>{epiparameter}</code> has offspring distributions
for SARS, Smallpox, Mpox, Pneumonic Plague, Hantavirus Pulmonary
Syndrome, Ebola Virus Disease. Let’s access the offspring distribution
<code>mean</code> and <code>dispersion</code> parameters for SARS-1:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load parameters</span></span>
<span><span class="va">sars</span> <span class="op">&lt;-</span> <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"SARS"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"offspring distribution"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sars_params</span> <span class="op">&lt;-</span> <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">get_parameters</span><span class="op">(</span><span class="va">sars</span><span class="op">)</span></span>
<span><span class="va">sars_params</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      mean dispersion 
      1.63       0.16 </code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6">Show me the solution</h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#' estimate for ebola --------------</span></span>
<span></span>
<span><span class="va">ebola_epiparameter</span> <span class="op">&lt;-</span> <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"Ebola"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"offspring distribution"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ebola_params</span> <span class="op">&lt;-</span> <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">get_parameters</span><span class="op">(</span><span class="va">ebola_epiparameter</span><span class="op">)</span></span>
<span><span class="va">ebola_params</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      mean dispersion 
       1.5        5.1 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate</span></span>
<span><span class="co"># proportion of cases that</span></span>
<span><span class="co"># generate 80% of transmission</span></span>
<span><span class="fu">proportion_transmission</span><span class="op">(</span></span>
<span>  R <span class="op">=</span> <span class="va">ebola_params</span><span class="op">[[</span><span class="st">"mean"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  k <span class="op">=</span> <span class="va">ebola_params</span><span class="op">[[</span><span class="st">"dispersion"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  percent_transmission <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    R   k prop_80
1 1.5 5.1   43.2%</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#' estimate for sars --------------</span></span>
<span></span>
<span><span class="co"># estimate</span></span>
<span><span class="co"># proportion of cases that</span></span>
<span><span class="co"># generate 80% of transmission</span></span>
<span><span class="fu">proportion_transmission</span><span class="op">(</span></span>
<span>  R <span class="op">=</span> <span class="va">sars_params</span><span class="op">[[</span><span class="st">"mean"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  k <span class="op">=</span> <span class="va">sars_params</span><span class="op">[[</span><span class="st">"dispersion"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  percent_transmission <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>     R    k prop_80
1 1.63 0.16     13%</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#' estimate for mers --------------</span></span>
<span></span>
<span><span class="co"># estimate</span></span>
<span><span class="co"># proportion of cases that</span></span>
<span><span class="co"># generate 80% of transmission</span></span>
<span><span class="fu">proportion_transmission</span><span class="op">(</span></span>
<span>  R <span class="op">=</span> <span class="va">offspring_fit</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"mu"</span><span class="op">]</span>,</span>
<span>  k <span class="op">=</span> <span class="va">offspring_fit</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"size"</span><span class="op">]</span>,</span>
<span>  percent_transmission <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          R          k prop_80
1 0.6045295 0.02039807    2.1%</code></pre>
</div>
<p>MERS has the lowest percent of cases (2.1%) responsible of the 80% of
the transmission, representative of highly overdispersed offspring
distributions.</p>
<p>Ebola has the highest percent of cases (43%) responsible of the 80%
of the transmission. This is representative of offspring distributions
with high dispersion parameters.</p>
</div>
</div>
</div>
</div>
<div id="inverse-dispersion" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="inverse-dispersion" class="callout-inner">
<h3 class="callout-title">inverse-dispersion?<a class="anchor" aria-label="anchor" href="#inverse-dispersion"></a>
</h3>
<div class="callout-content">
<p>The dispersion parameter <span class="math inline">\(k\)</span> can
be expressed differently across the literature.</p>
<ul>
<li>In the Wikipedia page for the negative binomial, this parameter is
defined in its reciprocal form (refer to the <a href="https://en.wikipedia.org/wiki/Negative_binomial_distribution" class="external-link">variance
equation</a>).</li>
<li>In <a href="https://ben18785.shinyapps.io/distribution-zoo/" class="external-link">the
distribution zoo</a> shiny app, the dispersion parameter <span class="math inline">\(k\)</span> is named “Inverse-dispersion” but it is
equal to parameter estimated in this episode. We invite you to explore
this!</li>
</ul>
</div>
</div>
</div>
<div id="heterogeneity" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="heterogeneity" class="callout-inner">
<h3 class="callout-title">heterogeneity?<a class="anchor" aria-label="anchor" href="#heterogeneity"></a>
</h3>
<div class="callout-content">
<p>The individual-level variation in transmission is also referred as
the heterogeneity in the transmission or degree of heterogeneity in <a href="https://wellcomeopenresearch.org/articles/5-83" class="external-link">Lloyd-Smith et
al., 2005</a>, heterogeneous infectiousness in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2330-z" class="external-link">Campbell
et al., 2018</a> when introducing the <code>{outbreaker2}</code>
package. Similarly, a contact network can store heterogeneous
epidemiological contacts as in the documentation of the
<a href="https://www.repidemicsconsortium.org/epicontacts/" class="external-link">epicontacts</a> package (<a href="https://www.repidemicsconsortium.org/epicontacts/articles/epicontacts.html" class="external-link">Nagraj
et al., 2018</a>).</p>
</div>
</div>
</div>
<div id="read-these-blog-posts" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div id="read-these-blog-posts" class="callout-inner">
<h3 class="callout-title">Read these blog posts<a class="anchor" aria-label="anchor" href="#read-these-blog-posts"></a>
</h3>
<div class="callout-content">
<p>The <a href="https://plus.maths.org/content/monkeypox" class="external-link">Tracing
monkeypox</a> article from the <a href="https://maths.org/juniper/" class="external-link">JUNIPER</a> consortium showcases the
usefulness of network models on contact tracing.</p>
<p>The <a href="https://kucharski.substack.com/p/going-viral" class="external-link">Going
viral</a> post from Adam Kucharski shares insides from YouTube virality,
disease outbreaks, and marketing campaigns on the conditions that spark
contagion online.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Use <a href="https://www.repidemicsconsortium.org/epicontacts/" class="external-link">epicontacts</a> to calculate the number of secondary
cases cause by a particular individual from linelist and contact
data.</li>
<li>Use <a href="https://lbbe.univ-lyon1.fr/fr/fitdistrplus" class="external-link">fitdistrplus</a> to empirically estimate the
offspring distribution from the number of secondary cases
distribution.</li>
<li>Use <code>{superspreading}</code> to estimate the probability of
having clusters of a given size from primary cases and inform contact
tracing efforts.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-superspreading-simulate"><p>Content from <a href="superspreading-simulate.html">Simulate transmission chains</a></p>
<hr>
<p>Last updated on 2024-05-04 |
        
        <a href="https://github.com/epiverse-trace/tutorials-middle/edit/main/episodes/superspreading-simulate.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 32 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can we simulate transmission chains based on infection
characteristics?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Create a short term projection using a branching process with
<code>{epichains}</code>.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>Learners should familiarise themselves with the following concept
dependencies before working through this tutorial:</p>
<p><strong>Statistics</strong>: Common probability distributions,
including Poisson and negative binomial.</p>
<p><strong>Epidemic theory</strong>: The reproduction number, <span class="math inline">\(R\)</span>.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<!-- what we know --><p>Individual variation in transmission can affect both the potential
for an epidemic to establish in a population and the ease of control (<a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371" class="external-link">Cori
et al., 2017</a>).</p>
<ul>
<li><p>Greater variation reduces the overall probably of a single new
case causing a large local outbreak, because most cases infect few
others and individuals that generate a large number of secondary cases
are relatively rare.</p></li>
<li><p>However, if a ‘superspreading event’ does occur and the outbreak
gets established, this variation can make an outbreak harder to control
using <em>mass interventions</em> (i.e. blanket interventions that
implicitly assume everyone contributes equally to transmission), because
some cases contribute disproportionality: a single uncontrolled case may
generate a large number of secondary cases.</p></li>
<li><p>Conversely, variation in transmission may provide opportunities
for <em>targeted interventions</em> if the individuals who contribute
more to transmission (due to biological or behavioural factors), or the
settings in which ‘superspreading events’ occur, share
socio-demographic, environmental or geographical characteristics that
can be defined.</p></li>
</ul>
<!-- what we dont know --><p>How can we quantify the potential of a new infection to cause a large
outbreak based on its reproduction number <span class="math inline">\(R\)</span> and the dispersion <span class="math inline">\(k\)</span> of its offspring distribution?</p>
<figure style="text-align: center"><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-1-1.png" alt="Observed number of cumulative cases from the Middle East respiratory syndrome (MERS) outbreak in South Korea, 2015, alongside with simulated transmission chains assuming an offspring distribution with $R=0.6$ and $k=0.02$." class="figure mx-auto d-block"><figcaption>
Observed number of cumulative cases from the Middle East respiratory
syndrome (MERS) outbreak in South Korea, 2015, alongside with simulated
transmission chains assuming an offspring distribution with <span class="math inline">\(R=0.6\)</span> and <span class="math inline">\(k=0.02\)</span>.
</figcaption></figure><!-- what we are going to do --><p>In this episode, we will use the <code>{epichains}</code> package to
simulate transmission chains and estimate the potential for large
outbreaks following the introduction of a new case. We are going to use
it with functions from <code>{epiparameter}</code>, <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a>
and <a href="https://purrr.tidyverse.org/" class="external-link">purrr</a>, so also loading the <a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a>
package:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">epichains</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span></code></pre>
</div>
<div id="the-double-colon" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="the-double-colon" class="callout-inner">
<h3 class="callout-title">The double-colon<a class="anchor" aria-label="anchor" href="#the-double-colon"></a>
</h3>
<div class="callout-content">
<p>The double-colon <code>::</code> in R let you call a specific
function from a package without loading the entire package into the
current environment.</p>
<p>For example, <code>dplyr::filter(data, condition)</code> uses
<code>filter()</code> from the <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a> package.</p>
<p>This help us remember package functions and avoid namespace
conflicts.</p>
</div>
</div>
</div>
</section><section id="simulation-of-uncontrolled-outbreaks"><h2 class="section-heading">Simulation of uncontrolled outbreaks<a class="anchor" aria-label="anchor" href="#simulation-of-uncontrolled-outbreaks"></a>
</h2>
<hr class="half-width">
<p>Infectious disease epidemics spread through populations when a chain
of infected individuals transmit the infection to others. <a href="https://epiverse-trace.github.io/epichains/articles/theoretical_background.html" class="external-link">Branching
processes</a> can be used to model this transmission. A branching
process is a stochastic process (i.e. a random process that can be
described by a known probability distribution), where each infectious
individual gives rise to a random number of individuals in the next
generation of infection, starting with the index case in generation 1.
The distribution of the number of secondary cases each individual
generates is called the offspring distribution (<a href="https://epiverse-trace.github.io/epichains/articles/epichains.html" class="external-link">Azam
&amp; Funk, 2024</a>).</p>
<p><code>{epichains}</code> provides methods to analyse and simulate the
<em>size</em> and <em>length</em> of branching processes with an given
offspring distribution. <code>{epichains}</code> implements a rapid and
simple model to simulate transmission chains to assess epidemic risk,
project cases into the future, and evaluate interventions that change
<span class="math inline">\(R\)</span>.</p>
<div id="chain-size-and-length" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="chain-size-and-length" class="callout-inner">
<h3 class="callout-title">chain size and length<a class="anchor" aria-label="anchor" href="#chain-size-and-length"></a>
</h3>
<div class="callout-content">
<ul>
<li><p>The <strong>size</strong> of the transmission chain is defined as
the total number of individuals infected across all generations of
infection, and</p></li>
<li><p>the <strong>length</strong> of the transmission chain is the
number of generations from the first case to the last case in the
outbreak before the chain ended.</p></li>
</ul>
<p>The <em>size</em> calculation includes the first case, and the
<em>length</em> calculation contains the first generation when the first
case starts the chain (See figure below).</p>
<figure><img src="https://raw.githubusercontent.com/epiverse-trace/epichains/main/vignettes/img/transmission_chain_example.png" alt="An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicating who infected whom. The chain grows through generations Gen 1, Gen 2, and Gen 3, producing cases C2, C3, C4, C5, and C6. The chain ends at generation Gen 3 with cases C4, C5, and C6. The size of C1’s chain is 6, including C1 (that is, the sum of all blue circles), and the length is 3, which includes Gen 1 (maximum number of generations reached by C1’s chain) (Azam &amp; Funk, 2024)." class="figure mx-auto d-block"><div class="figcaption">
<strong>An example of a transmission chain
starting with a single case C1.</strong> Cases are represented by blue
circles and arrows indicating who infected whom. The chain grows through
generations Gen 1, Gen 2, and Gen 3, producing cases C2, C3, C4, C5, and
C6. The chain ends at generation Gen 3 with cases C4, C5, and C6. The
size of C1’s chain is 6, including C1 (that is, the sum of all blue
circles), and the length is 3, which includes Gen 1 (maximum number of
generations reached by C1’s chain) (<a href="https://epiverse-trace.github.io/epichains/articles/epichains.html" class="external-link">Azam
&amp; Funk, 2024</a>).</div>
</figure>
</div>
</div>
</div>
<p>To use <code>{epichains}</code>, we need to know (or assume) two key
epidemiological values: the offspring distribution and the generation
time.</p>
</section><section id="get-the-offspring-distribution"><h2 class="section-heading">Get the offspring distribution<a class="anchor" aria-label="anchor" href="#get-the-offspring-distribution"></a>
</h2>
<hr class="half-width">
<p>Here we assume the MERS offspring distribution follows a negative
binomial distribution, with mean (reproduction number <span class="math inline">\(R\)</span>) and dispersion <span class="math inline">\(k\)</span> values estimated from the linelist and
contact data of <code>mers_korea_2015</code> in the
<a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> R package in the previous episode.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mers_offspring</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span>mean <span class="op">=</span> <span class="fl">0.60</span>, dispersion <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span></span></code></pre>
</div>
<div id="offspring-distribution-for-epichains" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="offspring-distribution-for-epichains" class="callout-inner">
<h3 class="callout-title">offspring distribution for epichains<a class="anchor" aria-label="anchor" href="#offspring-distribution-for-epichains"></a>
</h3>
<div class="callout-content">
<p>We input an offspring distribution to <code>{epichains}</code> by
referring to the R function that generates random values from the
distribution we want. For a negative binomial distribution, we use
<code>rnbinom</code> with its corresponding <code>mu</code> and
<code>size</code> arguments:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>  offspring_dist <span class="ot">=</span> rnbinom,</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  mu <span class="ot">=</span> mers_offspring[<span class="st">"mean"</span>],</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  size <span class="ot">=</span> mers_offspring[<span class="st">"dispersion"</span>],</span></code></pre>
</div>
<p>The reference manual in <code>?rnbinom</code> tells us our required
specific arguments.</p>
</div>
</div>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>Poisson and other distributions</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p><code>{epichains}</code> can accept any R function that generates
random numbers, so the specified arguments will change depending on the
R function used. For more details on the range of possible options, see
the function reference manual.</p>
<p>For example, let’s say we want to use a Poisson distribution for the
offspring distribution. First, read the argument required in the
<code>?rpois</code> reference manual. Second, specify the
<code>lambda</code> argument parameter, also known as rate or mean in
the literature. In <code>{epichains}</code>, this can look like
this:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>  offspring_dist <span class="ot">=</span> rpois,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  lambda <span class="ot">=</span> mers_offspring[<span class="st">"mean"</span>],</span></code></pre>
</div>
<p>In this example, we can specify
<code>lambda = mers_offspring["mean"]</code> because the mean number of
secondary cases generated (i.e. <span class="math inline">\(R\)</span>)
should be the same regardless of the distribution we assume. What
changes is the variance of the distribution, and hence the level of
individual-level variation in transmission. When the dispersion
parameter <span class="math inline">\(k\)</span> approaches infinity
(<span class="math inline">\(k \rightarrow \infty\)</span>) in a
negative binomial distribution, the variance equals the mean. This makes
the conventional Poisson distribution a special case of the negative
binomial distribution.</p>
</div>
</div>
</div>
</div>
</section><section id="get-generation-time"><h2 class="section-heading">Get generation time<a class="anchor" aria-label="anchor" href="#get-generation-time"></a>
</h2>
<hr class="half-width">
<p>The <a href="reference.html#serialinterval">serial interval</a>
distribution is often used to approximate the generation time
distribution. This approximation is commonly used because it is easier
to observe and measure the onset of symptoms in each case than the
precise time of infection.</p>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>generation time vs serial interval</h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler2" aria-labelledby="headingSpoiler2">
<div class="accordion-body">
<figure><img src="../fig/serial-interval-observed.jpeg" alt="A schematic of the relationship of different time periods of transmission between an infector and an infectee in a transmission pair. Exposure window is defined as the time interval having viral exposure, and transmission window is defined as the time interval for onward transmission with respect to the infection time (Chung Lau et al., 2021)." class="figure mx-auto d-block"><div class="figcaption">A schematic of the relationship of different
time periods of transmission between an infector and an infectee in a
transmission pair. Exposure window is defined as the time interval
having viral exposure, and transmission window is defined as the time
interval for onward transmission with respect to the infection time (<a href="https://academic.oup.com/jid/article/224/10/1664/6356465" class="external-link">Chung
Lau et al., 2021</a>).</div>
</figure><p>However, using the <em>serial interval</em> as an approximation of
the <em>generation time</em> is primarily valid for diseases in which
infectiousness starts after symptom onset (<a href="https://academic.oup.com/jid/article/224/10/1664/6356465" class="external-link">Chung
Lau et al., 2021</a>). In cases where infectiousness starts before
symptom onset, the serial intervals can have negative values, which is
the case for diseases with pre-symptomatic transmission (<a href="https://www.ijidonline.com/article/S1201-9712(20)30119-3/fulltext#gr2" class="external-link">Nishiura
et al., 2020</a>).</p>
</div>
</div>
</div>
</div>
<p>Let’s use the <code>{epiparameter}</code> package to access and use
the available serial interval for MERS disease:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">serial_interval</span> <span class="op">&lt;-</span> <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"mers"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"serial"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">serial_interval</span>, day_range <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">25</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The serial interval for MERS has a mean of 12.6 days and a standard
deviation of 2.8 days.</p>
<div id="generation-time-for-epichains" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="generation-time-for-epichains" class="callout-inner">
<h3 class="callout-title">generation time for epichains<a class="anchor" aria-label="anchor" href="#generation-time-for-epichains"></a>
</h3>
<div class="callout-content">
<p>In <code>{epichains}</code>, we need to specify the generation time
as a function that generates random numbers. Using
<code>{epiparameter}</code> has the advantage of using the distribution
function <code>epiparameter::generate()</code> for this input. This will
look like this:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">generate</span><span class="op">(</span>x <span class="op">=</span> <span class="va">serial_interval</span>, times <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span></code></pre>
</div>
<p>This interface is similar to the one <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> uses to link
with <code>{epiparameter}</code>. Read the <a href="https://epiverse-trace.github.io/cfr/articles/delay_distributions.html" class="external-link">work
with delay distributions</a> vignette for further context.</p>
</div>
</div>
</div>
</section><section id="simulate-a-single-chain"><h2 class="section-heading">Simulate a single chain<a class="anchor" aria-label="anchor" href="#simulate-a-single-chain"></a>
</h2>
<hr class="half-width">
<p>Now we are prepared to use the <code>simulate_chains()</code>
function from <code>{epichains}</code> to create <strong>one</strong>
transmission chain:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">simulate_chains</span><span class="op">(</span></span>
<span>  <span class="co"># simulation controls</span></span>
<span>  index_cases <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  statistic <span class="op">=</span> <span class="st">"size"</span>,</span>
<span>  <span class="co"># offspring</span></span>
<span>  offspring_dist <span class="op">=</span> <span class="va">rnbinom</span>,</span>
<span>  mu <span class="op">=</span> <span class="va">mers_offspring</span><span class="op">[</span><span class="st">"mean"</span><span class="op">]</span>,</span>
<span>  size <span class="op">=</span> <span class="va">mers_offspring</span><span class="op">[</span><span class="st">"dispersion"</span><span class="op">]</span>,</span>
<span>  <span class="co"># generation</span></span>
<span>  generation_time <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">generate</span><span class="op">(</span>x <span class="op">=</span> <span class="va">serial_interval</span>, times <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p><code>simulate_chains()</code> requires three sets of arguments as a
minimum:</p>
<ul>
<li>simulation controls,</li>
<li>offspring distribution, and</li>
<li>generation time.</li>
</ul>
<p>In the lines above, we described how to specify the offspring
distribution and generation time. The <strong>simulation
controls</strong> include at least two arguments:</p>
<ul>
<li>
<code>index_case</code>, which defines the number of index cases to
simulate transmission chains for and</li>
<li>
<code>statistic</code>, which defines a chain statistic to track
(either <code>"size"</code> or <code>"length"</code>) as the stopping
criteria for each chain being simulated.</li>
</ul>
<div id="stopping-criteria" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="stopping-criteria" class="callout-inner">
<h3 class="callout-title">Stopping criteria<a class="anchor" aria-label="anchor" href="#stopping-criteria"></a>
</h3>
<div class="callout-content">
<p>This is an customisable feature of <code>{epichains}</code>. By
default, branching process simulations end when they have gone extinct.
For long-lasting transmission chains, in <code>simulate_chains()</code>
you can add the <code>stat_max</code> argument.</p>
<p>For example, if we set an stopping criteria for
<code>statistic = "size"</code> of <code>stat_max = 500</code>, no more
offspring will be produced after a chain of size 500.</p>
</div>
</div>
</div>
<p>The <code>simulate_chains()</code> output creates a
<code>&lt;epichains&gt;</code> class object, which we can then analyse
further in R.</p>
</section><section id="simulate-multiple-chains"><h2 class="section-heading">Simulate multiple chains<a class="anchor" aria-label="anchor" href="#simulate-multiple-chains"></a>
</h2>
<hr class="half-width">
<p>We can use <code>simulate_chains()</code> to create multiple chains
and increase the probability of simulating uncontrolled outbreak
projections given an overdispersed offspring distribution.</p>
<p>We need to specify three additional elements:</p>
<ul>
<li>
<code>set.seed(&lt;integer&gt;)</code>, which is a random number
generator function with a specified seed value, the
<code>&lt;integer&gt;</code> number, to ensure consistent results across
different runs of the code.</li>
<li>
<code>number_chains</code>, which defines the number of simulations
to run.</li>
<li>
<code>initial_cases</code> defines the number of initial cases to
input to the <code>index_cases</code> argument explained in the lines
above.</li>
</ul>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set seed for random number generator</span></span>
<span><span class="fu">set.seed</span><span class="op">(</span><span class="fl">33</span><span class="op">)</span></span>
<span><span class="co"># Number of simulation runs</span></span>
<span><span class="va">number_chains</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span>
<span><span class="co"># Number of initial cases</span></span>
<span><span class="va">initial_cases</span> <span class="op">&lt;-</span> <span class="fl">1</span></span></code></pre>
</div>
<p><code>number_chains</code> and <code>initial_cases</code> are
conveniently stored in objects to facilitate downstream reuse in the
workflow.</p>
<div id="iteration-using-purrr" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="iteration-using-purrr" class="callout-inner">
<h3 class="callout-title">Iteration using purrr<a class="anchor" aria-label="anchor" href="#iteration-using-purrr"></a>
</h3>
<div class="callout-content">
<p><a href="https://r4ds.hadley.nz/iteration.html" class="external-link">Iteration</a> aims to
perform the same action on different objects repeatedly.</p>
<p>Learn how to use the core <a href="https://purrr.tidyverse.org/" class="external-link">purrr</a> functions like
<code>map()</code> from the YouTube tutorial on <a href="https://www.youtube.com/watch?v=nd-Y8b22YaQ" class="external-link">How to purrr</a> by
Equitable Equations.</p>
<p>Or, if you previously used the <code>*apply</code> family of
functions, visit the package vignette on <a href="https://purrr.tidyverse.org/articles/base.html" class="external-link">purrr base R</a>,
which shares key differences, direct translations, and examples.</p>
</div>
</div>
</div>
<p>To get multiple chains, we must apply the
<code>simulate_chains()</code> function to each chain defined by a
sequence of numbers from 1 to 1000.</p>
<div id="purrr-and-epichains" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="purrr-and-epichains" class="callout-inner">
<h3 class="callout-title">purrr and epichains<a class="anchor" aria-label="anchor" href="#purrr-and-epichains"></a>
</h3>
<div class="callout-content">
<p>First, let’s sketch how we use <code>purrr::map()</code> with
<code>epichains::simulate_chains()</code>. The <code>map()</code>
function requires two arguments:</p>
<ul>
<li>
<code>.x</code>, with a vector of numbers, and</li>
<li>
<code>.f</code>, a function to iterate to each vector value.</li>
</ul>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map</span><span class="op">(</span></span>
<span>  <span class="co"># vector of numbers (chain IDs)</span></span>
<span>  .x <span class="op">=</span> <span class="fu">seq_len</span><span class="op">(</span><span class="va">number_chains</span><span class="op">)</span>,</span>
<span>  <span class="co"># function to iterate to each chain ID number</span></span>
<span>  .f <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">sim</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">simulate_chains</span><span class="op">(</span><span class="va">...</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="co"># creates a column with the chain ID number</span></span>
<span>      <span class="fu">mutate</span><span class="op">(</span>chain_id <span class="op">=</span> <span class="va">sim</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># combine list outputs (for each chain ID) into a single data frame</span></span>
<span>  <span class="fu">list_rbind</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>sim</code> element is placed to register the iteration
number (<strong>chain ID</strong>) as a new column in the
<code>&lt;epichains&gt;</code> output. The
<code>purrr::list_rbind()</code> function aims to combine all the list
outputs from <code>map()</code>.</p>
<p><strong>Why a dot (<code>.</code>) as a prefix?</strong> In the <a href="https://design.tidyverse.org/dots-prefix.html" class="external-link">tidy design
principles</a> book we have a chapter on the dot prefix!</p>
</div>
</div>
</div>
<p>Now, we are prepared to use <code>map()</code> to repeatedly simulate
from <code>simulate_chains()</code> and store in a vector from 1 to
1000:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simulated_chains_map</span> <span class="op">&lt;-</span></span>
<span>  <span class="co"># iterate one function across multiple numbers (chain IDs)</span></span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="co"># vector of numbers (chain IDs)</span></span>
<span>    .x <span class="op">=</span> <span class="fu">seq_len</span><span class="op">(</span><span class="va">number_chains</span><span class="op">)</span>,</span>
<span>    <span class="co"># function to iterate to each chain ID number</span></span>
<span>    .f <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">sim</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu">simulate_chains</span><span class="op">(</span></span>
<span>        <span class="co"># simulation controls</span></span>
<span>        index_cases <span class="op">=</span> <span class="va">initial_cases</span>,</span>
<span>        statistic <span class="op">=</span> <span class="st">"size"</span>,</span>
<span>        <span class="co"># offspring</span></span>
<span>        offspring_dist <span class="op">=</span> <span class="va">rnbinom</span>,</span>
<span>        mu <span class="op">=</span> <span class="va">mers_offspring</span><span class="op">[</span><span class="st">"mean"</span><span class="op">]</span>,</span>
<span>        size <span class="op">=</span> <span class="va">mers_offspring</span><span class="op">[</span><span class="st">"dispersion"</span><span class="op">]</span>,</span>
<span>        <span class="co"># generation</span></span>
<span>        generation_time <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">generate</span><span class="op">(</span>x <span class="op">=</span> <span class="va">serial_interval</span>, times <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="co"># creates a column with the chain ID number</span></span>
<span>        <span class="fu">mutate</span><span class="op">(</span>chain_id <span class="op">=</span> <span class="va">sim</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># combine list outputs (for each chain ID) into a single data frame</span></span>
<span>  <span class="fu">list_rbind</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div id="read-the-epichains-output" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="read-the-epichains-output" class="callout-inner">
<h3 class="callout-title">Read the epichains output<a class="anchor" aria-label="anchor" href="#read-the-epichains-output"></a>
</h3>
<div class="callout-content">
<p>To explore the output format of the <code>&lt;epichains&gt;</code>
class object of name <code>simulated_chains_map</code>, let’s look at
the simulated <code>chain_id</code> number 806.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">The epichains object</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Let’s use <code>dplyr::filter()</code> for this:</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chain_to_observe</span> <span class="op">&lt;-</span> <span class="fl">806</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### get epichain summary ----------------------------------------------------</span></span>
<span></span>
<span><span class="va">simulated_chains_map</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">chain_id</span> <span class="op">==</span> <span class="va">chain_to_observe</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`&lt;epichains&gt;` object

&lt; tree head (from first known infector_id) &gt;

  infectee_id sim_id infector_id generation     time chain_id
2           1      2           1          2 16.38623      806
3           1      3           1          2 11.79430      806
4           1      4           1          2 10.77252      806
5           1      5           1          2 11.39945      806
6           1      6           1          2 10.23130      806
7           1      7           2          3 26.01046      806


Number of infectors (known): 3
Number of generations: 3
Use `as.data.frame(&lt;object_name&gt;)` to view the full output in the console.</code></pre>
</div>
<p>Key elements from this output are in the footer, the piece of text
that appears at the bottom:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Number of infectors (known): 3
Number of generations: 3</code></pre>
</div>
<p>The simulated <code>chain_id</code> number 806 has three known
infectors and three generations. These numbers are more visible when
reading the <code>&lt;epichains&gt;</code> objects as a data frame.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">The epichains data frame</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### infector-infectee data frame --------------------------------------------</span></span>
<span></span>
<span><span class="va">simulated_chains_map</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">chain_id</span> <span class="op">==</span> <span class="va">chain_to_observe</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 9 × 6
  infectee_id sim_id infector_id generation  time chain_id
        &lt;int&gt;  &lt;dbl&gt;       &lt;dbl&gt;      &lt;int&gt; &lt;dbl&gt;    &lt;int&gt;
1           1      1          NA          1   0        806
2           1      2           1          2  16.4      806
3           1      3           1          2  11.8      806
4           1      4           1          2  10.8      806
5           1      5           1          2  11.4      806
6           1      6           1          2  10.2      806
7           1      7           2          3  26.0      806
8           1      8           2          3  29.8      806
9           1      9           2          3  26.6      806</code></pre>
</div>
<p>Chain 806 tells us a <strong>story</strong>: “In the first
transmission generation at <code>time = 0</code>, one index case
infected the first case with <code>sim_id = 1</code>. Then, in the
second transmission generation (between <code>time</code> 10 to 16),
<code>sim_id = 1</code> infected five cases. Later, in the third
transmission generation (between <code>time</code> 26 to 30),
<code>sim_id = 2</code> infected three new cases.”</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">An infectee data frame</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>The output data frame collects <strong>infectees</strong> as the
observation unit:</p>
<ul>
<li>Each infectee has a <code>sim_id</code>.</li>
<li>Each <em>infectee</em> that behaved as an <em>infector</em> is
registered in the <code>infector_id</code> column using
<code>sim_id</code> of that infectee.</li>
<li>Each infectee got infected in a specific <code>generation</code> and
(continuous) <code>time</code>.</li>
<li>The simulation number is registered under the <code>chain_id</code>
column.</li>
</ul>
<p><strong>Note:</strong> The <code>Number of infectors (known)</code>
includes the <code>NA</code> observation under the
<code>infector_id</code> column. This refers to the infector specified
as index case (in the <code>index_cases</code> argument), which started
the transmission chain to the infectee of <code>sim_id = 1</code>, at
<code>generation = 1</code>, and <code>time = 0</code>.</p>
</div>
</div>
</div>
</div>
</section><section id="visualize-multiple-chains"><h2 class="section-heading">Visualize multiple chains<a class="anchor" aria-label="anchor" href="#visualize-multiple-chains"></a>
</h2>
<hr class="half-width">
<p>To visualize the simulated chains, we need some pre-processing:</p>
<ol style="list-style-type: decimal">
<li>Let’s use <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a> to get round time numbers to resemble
surveillance days.</li>
<li>Count the daily cases in each simulation (by
<code>chain_id</code>).</li>
<li>Calculate the cumulative number of cases within a simulation.</li>
</ol>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># daily aggregate of cases</span></span>
<span><span class="va">simulated_chains_day</span> <span class="op">&lt;-</span> <span class="va">simulated_chains_map</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># use data.frame output from &lt;epichains&gt; object</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># transform chain ID column to factor (categorical variable)</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>chain_id <span class="op">=</span> <span class="fu">as_factor</span><span class="op">(</span><span class="va">chain_id</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># get the round number (day) of infection times</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>day <span class="op">=</span> <span class="fu">ceiling</span><span class="op">(</span><span class="va">time</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># count the daily number of cases in each simulation (chain ID)</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">chain_id</span>, <span class="va">day</span>, name <span class="op">=</span> <span class="st">"cases"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># calculate the cumulative number of cases for each simulation (chain ID)</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">chain_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cases_cumsum <span class="op">=</span> <span class="fu">cumsum</span><span class="op">(</span><span class="va">cases</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Before the plot, let’s create a summary table with the total time
duration and size of each chain. We can use the <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a>
“combo” of <code>group_by()</code>, <code>summarise()</code> and
<code>ungroup()</code>:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summarise the chain duration and size</span></span>
<span><span class="va">sim_chains_max</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">simulated_chains_day</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">chain_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span></span>
<span>    <span class="co"># duration</span></span>
<span>    day_max <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">day</span><span class="op">)</span>,</span>
<span>    <span class="co"># size</span></span>
<span>    cases_total <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">cases_cumsum</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim_chains_max</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1,000 × 3
   chain_id day_max cases_total
   &lt;fct&gt;      &lt;dbl&gt;       &lt;int&gt;
 1 1              0           1
 2 2              0           1
 3 3              0           1
 4 4              0           1
 5 5              0           1
 6 6              0           1
 7 7              0           1
 8 8              0           1
 9 9              0           1
10 10             0           1
# ℹ 990 more rows</code></pre>
</div>
<p>Now, we are prepared for using the <a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a>
package:</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize transmission chains by cumulative cases</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># create grouped chain trajectories</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">simulated_chains_day</span>,</span>
<span>    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">day</span>,</span>
<span>      y <span class="op">=</span> <span class="va">cases_cumsum</span>,</span>
<span>      group <span class="op">=</span> <span class="va">chain_id</span></span>
<span>    <span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>    show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># create points to visualize the chain endpoint</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">sim_chains_max</span>,</span>
<span>    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">day_max</span>,</span>
<span>      y <span class="op">=</span> <span class="va">cases_total</span>,</span>
<span>      group <span class="op">=</span> <span class="va">chain_id</span>,</span>
<span>      color <span class="op">=</span> <span class="va">chain_id</span></span>
<span>    <span class="op">)</span>,</span>
<span>    show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># define a 100-case threshold</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Day"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Cumulative cases"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-15-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Although most introductions of 1 index case do not generate secondary
cases (N = 940) or most outbreaks rapidly become extinct (median
duration of 17.5 and median size of 9), only 2 epidemic trajectories
among 1000 simulations (0.2%) can reach to more than 100 infected cases.
This finding is particularly remarkable because the reproduction number
<span class="math inline">\(R\)</span> is less than 1 (offspring
distribution mean of 0.6), but, given an offspring distribution
dispersion parameter of 0.02, it shows the potential for explosive
outbreaks of MERS disease.</p>
<p>We can count how many chains reached the 100-case threshold using
<a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a> functions:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># number of chains that reached the 100-case threshold</span></span>
<span><span class="va">sim_chains_max</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">day_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">cases_total</span> <span class="op">&gt;</span> <span class="fl">100</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 2 × 3
  chain_id day_max cases_total
  &lt;fct&gt;      &lt;dbl&gt;       &lt;int&gt;
1 666           88         110
2 23            45         113</code></pre>
</div>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>Observed cases vs simulated chains</h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler3" aria-labelledby="headingSpoiler3">
<div class="accordion-body">
<p>Let’s overlap the cumulative number of observed cases using the
linelist object from the <code>mers_korea_2015</code> dataset of the
<a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> R package. To prepare the dataset so we can
plot daily total cases over time, we use <a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a> to
convert the linelist to an <code>&lt;incidence2&gt;</code> object,
complete the missing dates of the time series with
<code>complete_dates()</code></p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mers_cumcases</span> <span class="op">&lt;-</span> <span class="va">mers_korea_2015</span><span class="op">$</span><span class="va">linelist</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># incidence2 workflow</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="fu">incidence</span><span class="op">(</span>date_index <span class="op">=</span> <span class="st">"dt_onset"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="fu">complete_dates</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># wrangling using {dplyr}</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>count_cumsum <span class="op">=</span> <span class="fu">cumsum</span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rownames_to_column</span><span class="op">(</span>var <span class="op">=</span> <span class="st">"day"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>day <span class="op">=</span> <span class="fu">as.numeric</span><span class="op">(</span><span class="va">day</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Use <code>plot()</code> to make an incidence plot:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the incidence2 object</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">mers_cumcases</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
<p>When plotting the observed number of cumulative cases from the Middle
East respiratory syndrome (MERS) outbreak in South Korea in 2015
alongside the previously simulated chains, we see that the observed
cases followed a trajectory that is consistent with the simulated
explosive outbreak dynamics (which makes sense, given the simulation
uses parameters based on this specific outbreak).</p>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-20-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>When we increase the dispersion parameter from <span class="math inline">\(k = 0.01\)</span> to <span class="math inline">\(k
= \infty\)</span> - and hence reduce individual-level variation in
transmission - and assume a fixed reproduction number <span class="math inline">\(R = 1.5\)</span>, the proportion of simulated
outbreaks that reached the 100-case threshold increases. This is because
the simulated outbreaks now have more of a consistent, clockwise
dynamic, rather than the high level of variability seen previously.</p>
<figure><img src="../fig/see-nature04153_Fig2-c.jpg" alt="Growth of simulated outbreaks with R = 1.5 and one initial case, conditional on non-extinction. Boxes show the median and interquartile range (IQR) of the first disease generation with 100 cases; whiskers show the most extreme values within 1.5 × IQR of the boxes, and crosses show outliers. Percentages show the proportion of 10,000 simulated outbreaks that reached the 100-case threshold (Lloyd-Smith et al., 2005)." class="figure mx-auto d-block"><div class="figcaption">
<strong>Growth of simulated outbreaks with R =
1.5 and one initial case, conditional on non-extinction.</strong> Boxes
show the median and interquartile range (IQR) of the first disease
generation with 100 cases; whiskers show the most extreme values within
1.5 × IQR of the boxes, and crosses show outliers. Percentages show the
proportion of 10,000 simulated outbreaks that reached the 100-case
threshold (<a href="https://www.nature.com/articles/nature04153" class="external-link">Lloyd-Smith et al.,
2005</a>).</div>
</figure><div id="early-spread-projections" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div id="early-spread-projections" class="callout-inner">
<h3 class="callout-title">Early spread projections<a class="anchor" aria-label="anchor" href="#early-spread-projections"></a>
</h3>
<div class="callout-content">
<p>In the epidemic’s initial phase, you can use <code>{epichains}</code>
to apply a branching process model to project the number of future
cases. Even though the model accounts for randomness in transmission and
variation in the number of secondary cases, there may be additional
local features we have not considered. Analysis of early forecasts made
for COVID in different countries using this model structure found that
predictions were often overconfident (<a href="https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2020.25.18.2000543#validationofthemodel-1" class="external-link">Pearson
et al., 2020</a>). This is likely because the real-time model did not
include all the changes in the offspring distribution that were
happening at the local level as a result of behaviour change and control
measures. You can read more about the importance of local context in
COVID-19 models in <a href="https://www.nature.com/articles/s43588-020-00014-7" class="external-link">Eggo et
al. (2020)</a>.</p>
<p>We invite you to read the vignette on <a href="https://epiverse-trace.github.io/epichains/articles/projecting_incidence.html" class="external-link">Projecting
infectious disease incidence: a COVID-19 example</a>! for more on making
predictions using <code>{epichains}</code>.</p>
</div>
</div>
</div>
</section><section id="challenges"><h2 class="section-heading">Challenges<a class="anchor" aria-label="anchor" href="#challenges"></a>
</h2>
<hr class="half-width">
<div id="monkeypox-large-outbreak-potential" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="monkeypox-large-outbreak-potential" class="callout-inner">
<h3 class="callout-title">Monkeypox large outbreak potential<a class="anchor" aria-label="anchor" href="#monkeypox-large-outbreak-potential"></a>
</h3>
<div class="callout-content">
<p>Evaluate the potential for a new Monkey pox (Mpox) case to generate
an explosive large outbreak.</p>
<ul>
<li>Simulate 1000 transmission chains with 1 initial case each.</li>
<li>Use the appropriate package to access delay data from previous
outbreaks.</li>
<li>How many simulated trajectories reach more than 100 infected
cases?</li>
</ul>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" data-bs-parent="#accordionHint1" aria-labelledby="headingHint1">
<div class="accordion-body">
<p>With <code>{epiparameter}</code>, you can access and use offspring
and delay distributions from previous Ebola outbreaks.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">epidist_db</span><span class="op">(</span>epi_dist <span class="op">=</span> <span class="st">"offspring"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">list_distributions</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">disease</span>, <span class="va">epi_distribution</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                        disease       epi_distribution n
1           Ebola Virus Disease offspring distribution 1
2 Hantavirus Pulmonary Syndrome offspring distribution 1
3                          Mpox offspring distribution 1
4              Pneumonic Plague offspring distribution 1
5                          SARS offspring distribution 2
6                      Smallpox offspring distribution 4</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">epidist_db</span><span class="op">(</span>epi_dist <span class="op">=</span> <span class="st">"serial interval"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">list_distributions</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">disease</span>, <span class="va">epi_distribution</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                disease epi_distribution n
1              COVID-19  serial interval 4
2   Ebola Virus Disease  serial interval 4
3             Influenza  serial interval 1
4                  MERS  serial interval 2
5 Marburg Virus Disease  serial interval 2
6                  Mpox  serial interval 5</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load packages -----------------------------------------------------------</span></span>
<span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epichains</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># delays ------------------------------------------------------------------</span></span>
<span></span>
<span><span class="va">mpox_offspring_epidist</span> <span class="op">&lt;-</span> <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"mpox"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"offspring"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">mpox_offspring</span> <span class="op">&lt;-</span> <span class="fu">get_parameters</span><span class="op">(</span><span class="va">mpox_offspring_epidist</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mpox_serialint</span> <span class="op">&lt;-</span> <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"mpox"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"serial interval"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># iterate -----------------------------------------------------------------</span></span>
<span></span>
<span><span class="co"># Set seed for random number generator</span></span>
<span><span class="fu">set.seed</span><span class="op">(</span><span class="fl">33</span><span class="op">)</span></span>
<span><span class="co"># Number of simulation runs</span></span>
<span><span class="va">number_chains</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span>
<span><span class="co"># Number of initial cases</span></span>
<span><span class="va">initial_cases</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">simulated_chains_mpox</span> <span class="op">&lt;-</span></span>
<span>  <span class="co"># iterate one function across multiple numbers (chain IDs)</span></span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="co"># vector of numbers (chain IDs)</span></span>
<span>    .x <span class="op">=</span> <span class="fu">seq_len</span><span class="op">(</span><span class="va">number_chains</span><span class="op">)</span>,</span>
<span>    <span class="co"># function to iterate to each chain ID number</span></span>
<span>    .f <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">sim</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu">simulate_chains</span><span class="op">(</span></span>
<span>        <span class="co"># simulation controls</span></span>
<span>        index_cases <span class="op">=</span> <span class="va">initial_cases</span>,</span>
<span>        statistic <span class="op">=</span> <span class="st">"size"</span>,</span>
<span>        <span class="co"># offspring</span></span>
<span>        offspring_dist <span class="op">=</span> <span class="va">rnbinom</span>,</span>
<span>        mu <span class="op">=</span> <span class="va">mpox_offspring</span><span class="op">[</span><span class="st">"mean"</span><span class="op">]</span>,</span>
<span>        size <span class="op">=</span> <span class="va">mpox_offspring</span><span class="op">[</span><span class="st">"dispersion"</span><span class="op">]</span>,</span>
<span>        <span class="co"># generation</span></span>
<span>        generation_time <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">generate</span><span class="op">(</span>x <span class="op">=</span> <span class="va">mpox_serialint</span>, times <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="co"># creates a column with the chain ID number</span></span>
<span>        <span class="fu">mutate</span><span class="op">(</span>chain_id <span class="op">=</span> <span class="va">sim</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># combine list outputs (for each chain ID) into a single data frame</span></span>
<span>  <span class="fu">list_rbind</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize ---------------------------------------------------------------</span></span>
<span></span>
<span><span class="co"># daily aggregate of cases</span></span>
<span><span class="va">simulated_chains_mpox_day</span> <span class="op">&lt;-</span> <span class="va">simulated_chains_mpox</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># use data.frame output from &lt;epichains&gt; object</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># transform chain ID column to factor (categorical variable)</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>chain_id <span class="op">=</span> <span class="fu">as_factor</span><span class="op">(</span><span class="va">chain_id</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># get the round number (day) of infection times</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>day <span class="op">=</span> <span class="fu">ceiling</span><span class="op">(</span><span class="va">time</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># count the daily number of cases in each simulation (chain ID)</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">chain_id</span>, <span class="va">day</span>, name <span class="op">=</span> <span class="st">"cases"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># calculate the cumulative number of cases for each simulation (chain ID)</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">chain_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cases_cumsum <span class="op">=</span> <span class="fu">cumsum</span><span class="op">(</span><span class="va">cases</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize transmission chains by cumulative cases</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># create grouped chain trajectories</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">simulated_chains_mpox_day</span>,</span>
<span>    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">day</span>,</span>
<span>      y <span class="op">=</span> <span class="va">cases_cumsum</span>,</span>
<span>      group <span class="op">=</span> <span class="va">chain_id</span></span>
<span>    <span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>    show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># define a 100-case threshold</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Day"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Cumulative cases"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Assuming a Monkey pox outbreak with <span class="math inline">\(R\)</span> = 0.32 and <span class="math inline">\(k\)</span> = 0.58, there is no trajectory among
1000 simulations that reach more than 100 infected cases. Compared to
MERS (<span class="math inline">\(R\)</span> = 0.6 and <span class="math inline">\(k\)</span> = 0.02).</p>
</div>
</div>
</div>
</div>
<div id="accordionHint2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint2" aria-expanded="false" aria-controls="collapseHint2">
  <h4 class="accordion-header" id="headingHint2">Epidemic Risk assessment accounting for superspreading</h4>
</button>
<div id="collapseHint2" class="accordion-collapse collapse" data-bs-parent="#accordionHint2" aria-labelledby="headingHint2">
<div class="accordion-body">
<p>With <code>{superspreading}</code>, you can get numerical solutions
to processes that <code>{epichains}</code> solve using branching
processes. We invite you to read the <code>{superspreading}</code>
vignette on <a href="https://epiverse-trace.github.io/superspreading/articles/epidemic_risk.html" class="external-link">Epidemic
risk</a> and respond to:</p>
<ul>
<li>What is the probability that a newly introduced pathogen will cause
a large outbreak?</li>
<li>What is the probability that an infection will, by chance, fail to
establish following initial introduction(s)?</li>
<li>What is the probability the outbreak will be contained?</li>
</ul>
<p>Check how these estimates vary non-linearly with respect to the mean
reproduction number <span class="math inline">\(R\)</span> and
dispersion <span class="math inline">\(k\)</span> of a given
disease.</p>
<!-- draft challenge -->
<!-- Calculate the probability a new Mpox case will lead to a large outbreak in the absence of control measures. Use the appropriate package to access delay data from previous outbreaks. -->
</div>
</div>
</div>
</div>
<div id="from-a-distribution-of-secondary-cases" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="from-a-distribution-of-secondary-cases" class="callout-inner">
<h3 class="callout-title">From a distribution of secondary cases<a class="anchor" aria-label="anchor" href="#from-a-distribution-of-secondary-cases"></a>
</h3>
<div class="callout-content">
<p><a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(15)70135-0/fulltext" class="external-link">Christian
Althaus, 2015</a> reused data published by <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(14)71075-8/fulltext#gr2" class="external-link">Faye
et al., 2015 (Figure 2)</a> on the transmission tree on Ebola virus
disease in Conakry, Guinea, 2014.</p>
<p>Using the data under the <strong>hint</strong> tab, estimate the
offspring distribution from the distribution of secondary cases. Then
estimate the large outbreak potential from this data.</p>
</div>
</div>
</div>
<div id="accordionHint3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint3" aria-expanded="false" aria-controls="collapseHint3">
  <h4 class="accordion-header" id="headingHint3">Give me a hint</h4>
</button>
<div id="collapseHint3" class="accordion-collapse collapse" data-bs-parent="#accordionHint3" aria-labelledby="headingHint3">
<div class="accordion-body">
<p>Code with the transmission tree data written by <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(15)70135-0/fulltext" class="external-link">Christian
Althaus, 2015</a>:</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Number of individuals in the trees</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">152</span></span>
<span><span class="co"># Number of secondary cases for all individuals</span></span>
<span><span class="va">c1</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">14</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">8</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>,</span>
<span>        <span class="fl">4</span>, <span class="fl">9</span>, <span class="fl">9</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">17</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">2</span>,</span>
<span>        <span class="fl">5</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">9</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">c0</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="va">c1</span>, <span class="fu">rep</span><span class="op">(</span><span class="fl">0</span>, <span class="va">n</span> <span class="op">-</span> <span class="fu">length</span><span class="op">(</span><span class="va">c1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">c0</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">enframe</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load packages ---------------------------</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">fitdistrplus</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epichains</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># fit a negative binomial distribution ------------------------------------</span></span>
<span></span>
<span><span class="co"># Fitting a negative binomial distribution to the number of secondary cases</span></span>
<span><span class="va">fit.cases</span> <span class="op">&lt;-</span> <span class="fu">fitdist</span><span class="op">(</span><span class="va">c0</span>, <span class="st">"nbinom"</span><span class="op">)</span></span>
<span><span class="va">fit.cases</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Fitting of the distribution ' nbinom ' by maximum likelihood 
Parameters:
      estimate Std. Error
size 0.1814260 0.03990278
mu   0.9537995 0.19812301</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># serial interval parameters ----------------------------------------------</span></span>
<span></span>
<span><span class="va">ebola_serialinter</span> <span class="op">&lt;-</span> <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>  disease <span class="op">=</span> <span class="st">"ebola"</span>,</span>
<span>  epi_dist <span class="op">=</span> <span class="st">"serial interval"</span>,</span>
<span>  single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># simulate outbreak trajectories ------------------------------------------</span></span>
<span></span>
<span><span class="co"># Set seed for random number generator</span></span>
<span><span class="fu">set.seed</span><span class="op">(</span><span class="fl">645</span><span class="op">)</span></span>
<span><span class="co"># Number of simulation runs</span></span>
<span><span class="va">number_chains</span> <span class="op">&lt;-</span> <span class="fl">1e2</span></span>
<span><span class="co"># Number of initial cases</span></span>
<span><span class="va">initial_cases</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">sim_multiple_chains</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    .x <span class="op">=</span> <span class="fu">seq_len</span><span class="op">(</span><span class="va">number_chains</span><span class="op">)</span>,</span>
<span>    .f <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">sim</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu">simulate_chains</span><span class="op">(</span></span>
<span>        index_cases <span class="op">=</span> <span class="va">initial_cases</span>,</span>
<span>        <span class="co"># stopping</span></span>
<span>        statistic <span class="op">=</span> <span class="st">"size"</span>,</span>
<span>        <span class="co"># offspring</span></span>
<span>        offspring_dist <span class="op">=</span> <span class="va">rnbinom</span>,</span>
<span>        mu <span class="op">=</span> <span class="va">fit.cases</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"mu"</span><span class="op">]</span>,</span>
<span>        size <span class="op">=</span> <span class="va">fit.cases</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="st">"size"</span><span class="op">]</span>,</span>
<span>        <span class="co"># generation</span></span>
<span>        generation_time <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">generate</span><span class="op">(</span>x <span class="op">=</span> <span class="va">ebola_serialinter</span>, times <span class="op">=</span> <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">mutate</span><span class="op">(</span>simulation_n <span class="op">=</span> <span class="va">sim</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># combine list outputs (for each chain ID) into a single data frame</span></span>
<span>  <span class="fu">list_rbind</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize ----------------------------------------</span></span>
<span></span>
<span><span class="va">sim_chains_aggregate</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">sim_multiple_chains</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>simulation_n <span class="op">=</span> <span class="fu">as_factor</span><span class="op">(</span><span class="va">simulation_n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>day <span class="op">=</span> <span class="fu">ceiling</span><span class="op">(</span><span class="va">time</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span><span class="va">simulation_n</span>, <span class="va">day</span>, name <span class="op">=</span> <span class="st">"cases"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">simulation_n</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cases_cumsum <span class="op">=</span> <span class="fu">cumsum</span><span class="op">(</span><span class="va">cases</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">sim_chains_aggregate</span>,</span>
<span>    mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span></span>
<span>      x <span class="op">=</span> <span class="va">day</span>,</span>
<span>      y <span class="op">=</span> <span class="va">cases_cumsum</span>,</span>
<span>      group <span class="op">=</span> <span class="va">simulation_n</span></span>
<span>    <span class="op">)</span>,</span>
<span>    show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># define a 100-case threshold</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/superspreading-simulate-rendered-unnamed-chunk-26-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Remarkably, even with R0 less than 1 (R = 0.95) we can have
potentially explosive outbreaks. The observed variation in individual
infectiousness in Ebola means that although the probability of
extinction is high, new index cases also have the potential for
explosive regrowth of the epidemic.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Use <code>{epichains}</code> to simulate the large outbreak
potential of diseases with overdispersed offspring distributions.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/epiverse-trace/tutorials-middle/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/epiverse-trace/tutorials-middle/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/epiverse-trace/tutorials-middle/" class="external-link">Source</a></p>
				<p><a href="https://github.com/epiverse-trace/tutorials-middle/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:andree.valle-campos@lshtm.ac.uk">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.4)</a>, <a href="https://github.com/carpentries/pegboard/tree/a32a7836d4455f407c3cafe8ab95edc636e5e919" class="external-link">pegboard (0.7.5)</a>, and <a href="https://github.com/epiverse-trace/varnish/tree/epiversetheme" class="external-link">varnish (1.0.2)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://epiverse-trace.github.io/tutorials-middle/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "forecasts, epidemic models, interventions",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://epiverse-trace.github.io/tutorials-middle/instructor/aio.html",
  "identifier": "https://epiverse-trace.github.io/tutorials-middle/instructor/aio.html",
  "dateCreated": "2024-05-04",
  "dateModified": "2024-05-04",
  "datePublished": "2024-05-04"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

