<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Real-time analysis and forecasting for outbreak analytics with R: Estimate severity</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-XZ471458FQ"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XZ471458FQ');
    </script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg"><abbr class="badge badge-light" title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught." style="background-color: #FF4955; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link" style="color: #FFF">
            <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
            Pre-Alpha
          </a>
          <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../severity-static.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Epiverse-TRACE" src="../assets/images/epiverse-logo.svg"></div>
    <div class="lesson-title-md">
      Real-time analysis and forecasting for outbreak analytics with R
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Real-time analysis and forecasting for outbreak analytics with R
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
          </ul></li>
      </ul></div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled><input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset></form>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Real-time analysis and forecasting for outbreak analytics with R
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 88%" class="percentage">
    88%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 88%" aria-valuenow="88" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../severity-static.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="quantify-transmissibility.html">1. Quantifying transmission</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="create-forecast.html">2. Create a short-term forecast</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Estimate severity
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#data-sources-for-clinical-severity">Data sources for clinical severity</a></li>
<li><a href="#naive-cfr">Naive CFR</a></li>
<li><a href="#biases-that-affect-cfr-estimation">Biases that affect CFR estimation</a></li>
<li><a href="#delay-adjusted-cfr">Delay-adjusted CFR</a></li>
<li><a href="#an-early-stage-cfr-estimate">An early-stage CFR estimate</a></li>
<li><a href="#more-severity-measures">More severity measures</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Glossary of Terms: Epiverse-TRACE</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/create-forecast.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/create-forecast.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Create a short-term
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Estimate severity</h1>
        <p>Last updated on 2024-03-25 |
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/episodes/severity-static.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li><p>Why do we estimate the clinical severity of an epidemic?</p></li>
<li><p>How can the Case Fatality Risk (CFR) be estimated early in an
ongoing epidemic?</p></li>
<li><p>How do we estimate other severity measurements from different
data sources?</p></li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li><p>Estimate the CFR from aggregated case data using
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate a delay-adjusted CFR using <code>{epiparameter}</code>
and <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
<li><p>Estimate severity measurements like the Hospitalization Fatality
Risk (HFR) using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>.</p></li>
</ul></div>
</div>
</div>
</div>
</div>
<div id="prerequisites" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="prerequisites" class="callout-inner">
<h3 class="callout-title">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<div class="callout-content">
<p>This episode requires you to be familiar with:</p>
<p><strong>Data science</strong> : Basic programming with R.</p>
<p><strong>Epidemic theory</strong> : <a href="reference.html#delaydist">Delay distributions</a>.</p>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width"><p>Common questions at the early stage of an epidemic include:</p>
<ul><li>What is the likely public health impact of the outbreak in terms of
clinical severity?</li>
<li>What are the most severely affected groups?</li>
<li>Does the outbreak have the potential to cause a very severe
pandemic?</li>
</ul><p>We can assess the pandemic potential of an epidemic with two critical
measurements: the transmissibility and the clinical severity. (<a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a>, <a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>)</p>
<figure><img src="../fig/cfr-hhs-scenarios-psaf.png" alt="The horizontal axis is the scaled measure of clinical severity, ranging from 1 to 7, where 1 is low, 4 is moderate, and 7 is very severe. The vertical axis is the scaled measure of transmissibility, ranging from 1 to 5, where 1 is low, 3 is moderate, and 5 is highly transmissible. On the graph, HHS pandemic planning scenarios are labeled across four quadrants (A, B, C and D). From left to right, the scenarios are “seasonal range,” “moderate pandemic,” “severe pandemic” and “very severe pandemic.” As clinical severity increases along the horizontal axis, or as transmissibility increases along the vertical axis, the severity of the pandemic planning scenario also increases." class="figure mx-auto d-block"><div class="figcaption">HHS Pandemic Planning Scenarios based on the
Pandemic Severity Assessment Framework. This uses a combined measure of
clinical severity and transmissibility to characterise influenza
pandemic scenarios. HHS: United States Department of Health and Human
Services (<a href="https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html" class="external-link">CDC,
2016</a>).</div>
</figure><p>One epidemiological approach to estimating the clinical severity is
quantifying the Case Fatality Risk or CFR. CFR is the conditional
probability of death given confirmed diagnosis, calculated as the
cumulative number deaths from an infectious disease over the number of
confirmed diagnosed cases. However, calculating this directly during the
course of an epidemic tends to result in a naive or biased CFR given the
time <a href="reference.html#delaydist">delay</a> from onset to death,
varying substantially as the epidemic progresses and stabilising at the
later stages of the outbreak (<a href="https://academic.oup.com/aje/article/162/5/479/82647?login=false#620743" class="external-link">Ghani
et al., 2005</a>).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_c.png" alt="The periods are relevant: Period 1 -- 15 days where CFR is zero to indicate this is due to no reported deaths; Period from Mar 15 -- Apr 26 where CFR appears to be rising; Period Apr 30 -- May 30 where the CFR estimate stabilises." class="figure mx-auto d-block"><div class="figcaption">Observed biased confirmed case fatality risk
(CFR) estimates as a function of time (thick line) calculated as the
cumulative number of deaths over confirmed cases at time t. The estimate
at the end of an outbreak (~May 30) is the realised CFR by the end of
the epidemic. The horizontal continuous line and dotted lines show the
expected value and the 95% confidence intervals (95% CI) of the
predicted delay-adjusted CFR estimate only by using the observed data
until 27 Mar 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>The periods are relevant: Period 1 – 15 days where CFR is zero to
indicate this is due to no reported deaths; Period from Mar 15 – Apr 26
where CFR appears to be rising; Period Apr 30 – May 30 where the CFR
estimate stabilises.</p>
</div>
</div>
</div>
</div>
<p>More generally, estimating severity can be helpful even outside of a
pandemic planning scenario and in the context of routine public health.
Knowing whether an outbreak has or had a different severity from the
historical record can motivate causal investigations, which could be
intrinsic to the infectious agent (e.g. a new, more severe strain) or
due to underlying factors in the population (e.g. reduced immunity or
morbidity factors) (<a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a>).</p>
<p>In this tutorial we are going to learn how to use the
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package to calculate and adjust a CFR estimation
using <a href="reference.html#delaydist">delay distributions</a> from
<code>{epiparameter}</code> or elsewhere, based on the methods developed
by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>, also, how we can reuse <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> functions
for more severity measurements.</p>
<p>We’ll use the pipe <code>%&gt;%</code> to connect some of their
functions, so let’s also call to the <a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a>
package:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span></code></pre>
</div>
</section><section id="data-sources-for-clinical-severity"><h2 class="section-heading">Data sources for clinical severity<a class="anchor" aria-label="anchor" href="#data-sources-for-clinical-severity"></a>
</h2>
<hr class="half-width"><p>What are data sources can we use to estimate the clinical severity of
a disease outbreak? <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext" class="external-link">Verity
et al., 2020</a> summaries the spectrum of COVID-19 cases:</p>
<figure><img src="../fig/cfr-spectrum-cases-covid19.jpg" alt="Spectrum of COVID-19 cases. The CFR aims to estimate the proportion of Deaths among confirmed cases in an epidemic. (Verity et al., 2020)" class="figure mx-auto d-block"><div class="figcaption">Spectrum of COVID-19 cases. The CFR aims to
estimate the proportion of Deaths among confirmed cases in an epidemic.
(<a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30243-7/fulltext#gr1" class="external-link">Verity
et al., 2020</a>)</div>
</figure><ul><li>At the top of the pyramid, those meeting the WHO case criteria for
<strong>severe</strong> or critical cases will likely be identified in
the hospital setting, presenting with atypical viral pneumonia. These
cases will have been identified in mainland China and among those
categorised internationally as local transmission.</li>
<li>Many more cases are likely to be <strong>symptomatic</strong> (i.e.,
with fever, cough, or myalgia) but might not require hospitalisation.
These cases will have been identified through links to international
travel to high-risk areas and through contact-tracing of contacts of
confirmed cases. They might also be identified through population
surveillance of, for example, influenza-like illness.</li>
<li>The bottom part of the pyramid represents <strong>mild</strong> (and
possibly <strong>asymptomatic</strong>) cases. These cases might be
identified through contact tracing and subsequently via serological
testing.</li>
</ul></section><section id="naive-cfr"><h2 class="section-heading">Naive CFR<a class="anchor" aria-label="anchor" href="#naive-cfr"></a>
</h2>
<hr class="half-width"><p>We measure disease severity in terms of case fatality risk (CFR). The
CFR is interpreted as the conditional probability of death given
confirmed diagnosis, calculated as the cumulative number of deaths <span class="math inline">\(D_{t}\)</span> over the cumulative number of
confirmed cases <span class="math inline">\(C_{t}\)</span> at a certain
time <span class="math inline">\(t\)</span>. We can refer to the
<em>naive CFR</em> (also crude or biased CFR, <span class="math inline">\(b_{t}\)</span>):</p>
<p><span class="math display">\[ b_{t} =  \frac{D_{t}}{C_{t}}
\]</span></p>
<p>This calculation is <em>naive</em> because it tends to yield a biased
and mostly underestimated CFR due to the time-delay from onset to death,
only stabilising at the later stages of the outbreak.</p>
<!-- add here the callout on ratio or risk?  -->
<!-- https://github.com/epiverse-trace/cfr/issues/130 -->
<p>To calculate the naive CFR, the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package requires
an input data frame with three columns named:</p>
<ul><li><code>date</code></li>
<li><code>cases</code></li>
<li><code>deaths</code></li>
</ul><p>Let’s explore the <code>ebola1976</code> dataset, which comes from
the first Ebola outbreak in what was then called Zaire (now the
Democratic Republic of the Congo) in 1976, as analysed by Camacho et
al. (2014).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Ebola 1976 data provided with the package</span></span>
<span><span class="fu">data</span><span class="op">(</span><span class="st">"ebola1976"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># view top data</span></span>
<span><span class="va">ebola1976</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 73 × 3
   date       cases deaths
   &lt;date&gt;     &lt;int&gt;  &lt;int&gt;
 1 1976-08-25     1      0
 2 1976-08-26     0      0
 3 1976-08-27     0      0
 4 1976-08-28     0      0
 5 1976-08-29     0      0
 6 1976-08-30     0      0
 7 1976-08-31     0      0
 8 1976-09-01     1      0
 9 1976-09-02     1      0
10 1976-09-03     1      0
# ℹ 63 more rows</code></pre>
</div>
<div id="we-need-aggregated-incidence-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="we-need-aggregated-incidence-data" class="callout-inner">
<h3 class="callout-title">We need aggregated incidence data<a class="anchor" aria-label="anchor" href="#we-need-aggregated-incidence-data"></a>
</h3>
<div class="callout-content">
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> reads <strong>aggregated</strong> incidence data.
Similar to the <a href="https://epiforecasts.io/EpiNow2/" class="external-link">EpiNow2</a> with the difference that for
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> we need one more column named
<code>deaths</code>.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EpiNow2</span><span class="fu">::</span><span class="va">example_confirmed</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 130 × 2
   date       confirm
   &lt;date&gt;       &lt;dbl&gt;
 1 2020-02-22      14
 2 2020-02-23      62
 3 2020-02-24      53
 4 2020-02-25      97
 5 2020-02-26      93
 6 2020-02-27      78
 7 2020-02-28     250
 8 2020-02-29     238
 9 2020-03-01     240
10 2020-03-02     561
# ℹ 120 more rows</code></pre>
</div>
<p>This data input should be <strong>aggregated</strong> by day, which
means one observation <em>per day</em>, containing the <em>daily</em>
number of reported cases and deaths. Observations with zero or missing
values should also be included, similar to a time-series data.</p>
<p>Also, <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> currently works for <em>daily</em> data
only, not weekly data.</p>
<!-- suggest ways to deal with raw input weekly data -->
<!-- https://github.com/epiverse-trace/cfr/issues/117 -->
<div class="section level3">
<h3 id="aggregated-differs-from-linelist">Aggregated differs from Linelist<a class="anchor" aria-label="anchor" href="#aggregated-differs-from-linelist"></a></h3>
<p>Lastly, <em>aggregated</em> incidence data differs from
<strong>linelist</strong> data, where each observation contains
subject-level data.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_sierraleone_2014</span> <span class="op">%&gt;%</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 11,903 × 8
      id   age sex   status    date_of_onset date_of_sample district chiefdom   
   &lt;int&gt; &lt;dbl&gt; &lt;fct&gt; &lt;fct&gt;     &lt;date&gt;        &lt;date&gt;         &lt;fct&gt;    &lt;fct&gt;      
 1     1    20 F     confirmed 2014-05-18    2014-05-23     Kailahun Kissi Teng 
 2     2    42 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Teng 
 3     3    45 F     confirmed 2014-05-20    2014-05-25     Kailahun Kissi Tonge
 4     4    15 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 5     5    19 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 6     6    55 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 7     7    50 F     confirmed 2014-05-21    2014-05-26     Kailahun Kissi Teng 
 8     8     8 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
 9     9    54 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
10    10    57 F     confirmed 2014-05-22    2014-05-27     Kailahun Kissi Teng 
# ℹ 11,893 more rows</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>When we use <code>cfr_static()</code> to <code>data</code> only, we
calculate the naive CFR:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the static CFR without adjusting for delays</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1      0.955102    0.9210866     0.9773771</code></pre>
</div>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Load the <code>ebola_kikwit_1995</code> from the
<a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a> package and estimate the naive CFR.</p>
</div>
</div>
</div>
<div id="accordionHint1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint1" aria-expanded="false" aria-controls="collapseHint1">
  <h4 class="accordion-header" id="headingHint1">Give me a hint</h4>
</button>
<div id="collapseHint1" class="accordion-collapse collapse" aria-labelledby="headingHint1" data-bs-parent="#accordionHint1">
<div class="accordion-body">
<p>Inspect the format of the data input.</p>
<ul><li>Does it contain daily data?</li>
<li>Does the column names are as required by
<code>cfr_static()</code>?</li>
<li>How would you rename column names from a data frame?</li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>We can use <code>dplyr::rename()</code> to adapt the external data to
fit the data input for <code>cfr_static()</code>.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 192
Columns: 4
$ date      &lt;date&gt; 1995-01-06, 1995-01-07, 1995-01-08, 1995-01-09, 1995-01-10,…
$ onset     &lt;int&gt; 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ death     &lt;int&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ reporting &lt;lgl&gt; TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">onset</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">death</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1     0.8082192    0.7583033     0.8517565</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="biases-that-affect-cfr-estimation"><h2 class="section-heading">Biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#biases-that-affect-cfr-estimation"></a>
</h2>
<hr class="half-width"><div id="two-biases-that-affect-cfr-estimation" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="two-biases-that-affect-cfr-estimation" class="callout-inner">
<h3 class="callout-title">Two biases that affect CFR estimation<a class="anchor" aria-label="anchor" href="#two-biases-that-affect-cfr-estimation"></a>
</h3>
<div class="callout-content">
<p><a href="https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846" class="external-link">Lipsitch
et al., 2015</a> describe two potential biases that can affect the
estimation of CFR (and its possible solutions):</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Preferential ascertainment of severe cases</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>For diseases with a <em>spectrum</em> of clinical presentation, those
cases that come to the attention of public health authorities and
registered into surveillance databases will typically be people with the
most severe symptoms who seek medical care, are admitted to a hospital,
or die.</p>
<p>Therefore, the CFR will typically be higher among <em>detected
cases</em> than among the entire population of cases, given that the
latter may include individuals with mild, subclinical, and (under some
definitions of “case”) asymptomatic presentations.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Bias due to delayed reporting of death</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>During an <em>ongoing</em> epidemic, there is a delay between the
time someone dies and the time their death is reported. Therefore, at
any moment in time, the list of cases includes people who will die and
whose death has not yet occurred or has occurred but not yet been
reported. Thus, dividing the cumulative number of reported deaths by the
cumulative number of reported cases at any moment will underestimate the
true CFR.</p>
<p>The key determinants of the magnitude of the bias are the epidemic
<em>growth rate</em> and the <em>distribution of delays</em> from
case-reporting to death-reporting; the longer the delays and the faster
the growth rate, the greater the bias.</p>
<p>In this tutorial episode, we are going to focus on solutions to deal
with this specific bias using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>!</p>
</div>
</div>
</div>
</div>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div>Instructor Note</h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" aria-labelledby="headingInstructor2" data-bs-parent="#accordionInstructor2">
<div class="accordion-body">
<p>We can showcase this last bias using the <a href="https://epiverse-trace.github.io/cfr/articles/cfr.html#concept-how-reporting-delays-bias-cfr-estimates" class="external-link">concept
described in this <code>{cfr}</code> vignette</a>.</p>
<!-- create code and then a .gif? -->
</div>
</div>
</div>
</div>
<div id="biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div id="biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009" class="callout-inner">
<h3 class="callout-title">Biases during the Influenza A (H1N1) outbreak
in Mexico, 2009<a class="anchor" aria-label="anchor" href="#biases-during-the-influenza-a-h1n1-outbreak-in-mexico-2009"></a>
</h3>
<div class="callout-content">
<p>Improving an <em>early</em> epidemiological assessment of a
delay-adjusted CFR is crucial for determining virulence, shaping the
level and choices of public health intervention, and providing advice to
the general public.</p>
<p>In 2009, during the swine-flu virus, Influenza A (H1N1), Mexico had
an early biased estimation of the CFR. Initial reports from the
government of Mexico suggested a virulent infection, whereas, in other
countries, the same virus was perceived as mild (<a href="https://content.time.com/time/health/article/0,8599,1894534,00.html" class="external-link">TIME,
2009</a>).</p>
<p>In the USA and Canada, no deaths were attributed to the virus in the
first ten days following the World Health Organization’s declaration of
a public health emergency. Even under similar circumstances at the early
stage of the global pandemic, public health officials, policymakers and
the general public want to know the virulence of an emerging infectious
agent.</p>
<p><a href="https://www.science.org/doi/full/10.1126/science.1176062" class="external-link">Fraser
et al., 2009</a> reinterpreted the data assessing the biases and getting
a clinical severity lower than the 1918 influenza pandemic but
comparable with that seen in the 1957 pandemic.</p>
</div>
</div>
</div>
<div id="discussion2" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion2"></a>
</h3>
<div class="callout-content">
<p>Based on your experience:</p>
<ul><li>Share any previous outbreak in which you participated in its
response.</li>
</ul><p>Answer to these questions:</p>
<ul><li>How did you assess the clinical severity of the outbreak?</li>
<li>What were the primary sources of bias?</li>
<li>What did you do to take into account the identified bias?</li>
<li>What complementary analysis would you do to solve the bias?</li>
</ul></div>
</div>
</div>
</section><section id="delay-adjusted-cfr"><h2 class="section-heading">Delay-adjusted CFR<a class="anchor" aria-label="anchor" href="#delay-adjusted-cfr"></a>
</h2>
<hr class="half-width"><p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> developed a method that considers the <strong>time
delay</strong> from the onset of symptoms to death.</p>
<p>Real-time outbreaks may have a number of deaths that are insufficient
to determine the time distribution between onset and death. Therefore,
we can estimate the <em>distribution delay</em> from historical
outbreaks or reuse the ones accessible via R packages like
<code>{epiparameter}</code> or <code>{epireview}</code>, which collects
them from literature searches and systematic reviews.</p>
<p>Let’s use <code>{epiparameter}</code>:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get delay distribution</span></span>
<span><span class="va">onset_to_death_ebola</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"Ebola"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset_to_death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot &lt;epidist&gt; object</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, day_range <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To calculate the delay-adjusted CFR, we can use the
<code>cfr_static()</code> function with the <code>data</code> and
<code>delay_density</code> arguments.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the static CFR while adjusting for delays</span></span>
<span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.971        0.852             1</code></pre>
</div>
<p>The delay-adjusted CFR indicated that the overall disease severity
<em>at the end of the outbreak</em> or with the <em>latest data
available at the moment</em> is 0.971 with a 95% confidence interval
between 0.852 and 1, slightly higher than the naive one.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>How does {cfr} works?</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p>To adjust the CFR, <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> use the case and death incidence data to estimate the
number of cases with known outcomes:</p>
<p><span class="math display">\[
  u_t = \dfrac{\sum_{i = 0}^t
        \sum_{j = 0}^\infty c_{i - j} f_{j}}{\sum_{i = 0} c_i},
\]</span></p>
<p>where:</p>
<ul><li>
<span class="math inline">\(c_{t}\)</span> is the daily case
incidence at time <span class="math inline">\(t\)</span>,</li>
<li>
<span class="math inline">\(f_{t}\)</span> is the value of the
Probability Mass Function (PMF) of the <strong>delay
distribution</strong> between onset and death, and</li>
<li>
<span class="math inline">\(u_{t}\)</span> represents the
underestimation factor of the known outcomes.</li>
</ul><p><span class="math inline">\(u_{t}\)</span> is used to
<strong>scale</strong> the value of the cumulative number of cases in
the denominator in the calculation of the CFR. This is calculated
internally with the <a href="https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html" class="external-link"><code>estimate_outcomes()</code></a>
function.</p>
<p>The estimator for CFR can be written as:</p>
<p><span class="math display">\[p_{t} = \frac{b_{t}}{u_{t}}\]</span></p>
<p>where <span class="math inline">\(p_{t}\)</span> is the realized
proportion of confirmed cases to die from the infection (or the unbiased
CFR), and <span class="math inline">\(b_{t}\)</span>, the crude and
biased estimate of CFR (also naive CFR).</p>
<p>From this last equation, we observe that the unbiased CFR <span class="math inline">\(p_{t}\)</span> is larger than biased CFR <span class="math inline">\(b_{t}\)</span> because in <span class="math inline">\(u_{t}\)</span> the numerator is smaller than the
denominator (note that <span class="math inline">\(f_{t}\)</span> is the
probability distribution of the <em>delay distribution</em> between
onset and death). Therefore, we refer to <span class="math inline">\(b_{t}\)</span> as the biased estimator of CFR.</p>
<p>When we observe the entire course of an epidemic (from <span class="math inline">\(t \rightarrow \infty\)</span>), <span class="math inline">\(u_{t}\)</span> tends to 1, making <span class="math inline">\(b_{t}\)</span> tends to <span class="math inline">\(p_{t}\)</span> and become an unbiased estimator
(<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>).</p>
</div>
</div>
</div>
</div>
<div id="use-the-epidist-class" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="use-the-epidist-class" class="callout-inner">
<h3 class="callout-title">Use the<code>&lt;epidist&gt;</code>class<a class="anchor" aria-label="anchor" href="#use-the-epidist-class"></a>
</h3>
<div class="callout-content">
<p>When using an <code>&lt;epidist&gt;</code> class object we can use
this expression as a template:</p>
<p><code>function(x) density(&lt;EPIDIST_OBJECT&gt;, x)</code></p>
<p>For distribution functions with parameters not available in
<code>{epiparameter}</code>, we suggest you two alternatives:</p>
<ul><li><p>Create an <code>&lt;epidist&gt;</code> class object, to plug into
other R packages of the outbreak analytics pipeline. Read the <a href="https://epiverse-trace.github.io/epiparameter/reference/epidist.html" class="external-link">reference
documentation of <code>epiparameter::epidist()</code></a>, or</p></li>
<li><p>Read <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette for <a href="https://epiverse-trace.github.io/cfr/articles/delay_distributions.html" class="external-link">a
primer on working with delay distributions</a>.</p></li>
</ul></div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>When to use discrete distributions?</h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<p>For <code>cfr_static()</code> and all the <code>cfr_*()</code> family
of functions, the most appropriate choice to pass are
<strong>discrete</strong> distributions. This is because we will work
with daily case and death data.</p>
<p>We can assume that evaluating the Probability Distribution Function
(PDF) of a <em>continuous</em> distribution is equivalent to the
Probability Mass Function (PMF) of the equivalent <em>discrete</em>
distribution.</p>
<p>However, this assumption may not be appropriate for distributions
with greater picks. For example, when onset-to-death is strongly peaked
with a low variance. In such cases, the average disparity between the
PDF and PMF is expected to be more pronounced compared to distributions
with broader spreads.</p>
<p>In a scenario like the last one, you can opt to discretise the
continuous distribution using <code>epiparameter::discretise()</code> to
an <code>&lt;epidist&gt;</code> object.</p>
</div>
</div>
</div>
</div>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge2"></a>
</h3>
<div class="callout-content">
<p>Load the <code>ebola_kikwit_1995</code> from <a href="https://github.com/reconhub/outbreaks" class="external-link">outbreaks</a>
and estimate the delay-adjusted CFR using the appropriate distribution
delay. Then:</p>
<ul><li>Compare the naive and the delay-adjusted CFR solutions!</li>
</ul></div>
</div>
</div>
<div id="accordionHint2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint2" aria-expanded="false" aria-controls="collapseHint2">
  <h4 class="accordion-header" id="headingHint2">Give me a hint</h4>
</button>
<div id="collapseHint2" class="accordion-collapse collapse" aria-labelledby="headingHint2" data-bs-parent="#accordionHint2">
<div class="accordion-body">
<ul><li>Find the appropriate <code>&lt;epidist&gt;</code> object!</li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">outbreaks</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 192
Columns: 4
$ date      &lt;date&gt; 1995-01-06, 1995-01-07, 1995-01-08, 1995-01-09, 1995-01-10,…
$ onset     &lt;int&gt; 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ death     &lt;int&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ reporting &lt;lgl&gt; TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">outbreaks</span><span class="fu">::</span><span class="va">ebola_kikwit_1995</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span></span>
<span>    cases <span class="op">=</span> <span class="va">onset</span>,</span>
<span>    deaths <span class="op">=</span> <span class="va">death</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>    delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.819         0.72         0.928</code></pre>
</div>
</div>
</div>
</div>
</div>
</section><section id="an-early-stage-cfr-estimate"><h2 class="section-heading">An early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#an-early-stage-cfr-estimate"></a>
</h2>
<hr class="half-width"><p>We just found that the difference between the <em>naive</em> and
<em>delay-adjusted</em> CFR estimates using <code>cfr_static()</code>
are <em>just slightly</em> different. So, what is the main
difference?</p>
<p><a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> method aims to obtain an unbiased estimator “well
before” observing the entire course of the outbreak. For this
<a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> uses the underestimation factor <span class="math inline">\(u_{t}\)</span> to estimate the unbiased CFR <span class="math inline">\(p_{t}\)</span> using maximum-likelihood methods,
given the sampling process defined by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>.</p>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>What is the sampling process?</h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" aria-labelledby="headingSpoiler3" data-bs-parent="#accordionSpoiler3">
<div class="accordion-body">
<figure><img src="../fig/cfr-pone.0006852.g001.png" alt="The population of confirmed cases and sampling process for estimating the unbiased CFR during the course of an outbreak. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">The population of confirmed cases and sampling
process for estimating the unbiased CFR during the course of an
outbreak. (<a href="https://doi.org/10.1371/journal.pone.0006852.g001" class="external-link">Nishiura et
al., 2009</a>)</div>
</figure><p>From <em>aggregated incidence data</em>, at time <span class="math inline">\(t\)</span> we know the cumulative number of
confirmed cases and deaths, <span class="math inline">\(C_{t}\)</span>
and <span class="math inline">\(D_{t}\)</span>, and wish to estimate the
unbiased CFR <span class="math inline">\(\pi\)</span>, by way of the
factor of underestimation <span class="math inline">\(u_{t}\)</span>.</p>
<p>If we knew the factor of underestimation <span class="math inline">\(u_{t}\)</span> we could specify the size of the
population of confirmed cases no longer at risk (<span class="math inline">\(u_{t}C_{t}\)</span>, <strong>shaded</strong>),
although we do not know which surviving individuals belong to this
group. A proportion <span class="math inline">\(\pi\)</span> of those in
the group of cases still at risk (size <span class="math inline">\((1-
u_{t})C_{t}\)</span>, <strong>unshaded</strong>) is expected to die.</p>
<p>Because each case no longer at risk had an independent probability of
dying, <span class="math inline">\(\pi\)</span>, the number of deaths,
<span class="math inline">\(D_{t}\)</span>, is a sample from a binomial
distribution with sample size <span class="math inline">\(u_{t}C_{t}\)</span>, and probability of dying
<span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>.</p>
<p>This is represented by the following likelihood function to obtain
the maximum likelihood estimate of the unbiased CFR <span class="math inline">\(p_{t}\)</span> = <span class="math inline">\(\pi\)</span>:</p>
<p><span class="math display">\[
  {\sf L}(\pi | C_{t},D_{t},u_{t}) = \log{\dbinom{u_{t}C_{t}}{D_{t}}} +
D_{t} \log{\pi} +
  (u_{t}C_{t} - D_{t})\log{(1 - \pi)},
\]</span></p>
<p>This estimation is performed by the internal function
<code>?cfr:::estimate_severity()</code>.</p>
</div>
</div>
</div>
</div>
<p>The <code>cfr_static()</code> estimate is useful to get an overall
severity estimate of the outbreak (so far). Once the outbreak has
clearly ended, or has progressed such that deaths are reported more, the
estimated CFR is then <em>closest to</em> the ‘true’ unbiased CFR.</p>
<p>Additionally, a <em>delay-adjusted CFR</em> estimated with
<code>cfr_static()</code> can assess the severity of an emerging
infectious disease <strong>earlier</strong> than the biased or naive
CFR, during an epidemic.</p>
<p>This is particularly useful when an <em>epidemic curve of confirmed
cases</em> is the only data available (i.e. when individual data from
onset to death are unavailable, especially during the early stage of the
epidemic). When there are few deaths or none at all, an assumption has
to be made for the <em>delay distribution</em> from onset to death,
e.g. from literature based on previous outbreaks. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a> depict this in the figures with data from the SARS
outbreak in Hong Kong, 2003.</p>
<div id="accordionSpoiler4" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler4" aria-expanded="false" aria-controls="collapseSpoiler4">
  <h3 class="accordion-header" id="headingSpoiler4">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>CFR estimation during SARS outbreak, Hong Kong, 2003</h3>
</button>
<div id="collapseSpoiler4" class="accordion-collapse collapse" aria-labelledby="headingSpoiler4" data-bs-parent="#accordionSpoiler4">
<div class="accordion-body">
<p>Figures A and B show the cumulative numbers of cases and deaths of
SARS, and Figure C shows the observed (biased) CFR estimates as a
function of time, i.e. the cumulative number of deaths over cases at
time <span class="math inline">\(t\)</span>. Due to the delay from the
onset of symptoms to death, the biased estimate of CFR at time <span class="math inline">\(t\)</span> underestimates the realised CFR at the
end of an outbreak (i.e. 302/1755 = 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_abc.png" alt="Observed (biased) confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Observed (biased) confirmed case fatality risk
of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure><p>Nevertheless, even by only using the observed data for the period
March 19 to April 2, <code>cfr_static()</code> can yield an appropriate
prediction (Figure D), e.g. the delay-adjusted CFR at March 27 is 18.1 %
(95% CI: 10.5, 28.1). An overestimation is seen in the very early stages
of the epidemic, but the 95% confidence limits in the later stages
include the realised CFR (i.e. 17.2 %).</p>
<figure><img src="../fig/cfr-pone.0006852.g003-fig_d.png" alt="Early determination of the delay-adjusted confirmed case fatality risk of severe acute respiratory syndrome (SARS) in Hong Kong, 2003. (Nishiura et al., 2009)" class="figure mx-auto d-block"><div class="figcaption">Early determination of the delay-adjusted
confirmed case fatality risk of severe acute respiratory syndrome (SARS)
in Hong Kong, 2003. (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852" class="external-link">Nishiura
et al., 2009</a>)</div>
</figure></div>
</div>
</div>
</div>
<p>We can explore the <strong>early</strong> determination of the
<em>delay-adjusted CFR</em> using the <code>cfr_rolling()</code>
function.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout3"></a>
</h3>
<div class="callout-content">
<p><code>cfr_rolling()</code> is a utility function that automatically
calculates CFR for each day of the outbreak with the data available up
to that day, saving the user time.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily naive CFR</span></span>
<span><span class="va">rolling_cfr_naive</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span>data <span class="op">=</span> <span class="va">ebola1976</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_naive</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31     0.9510204    0.9160061     0.9744387
69 1976-11-01     0.9510204    0.9160061     0.9744387
70 1976-11-02     0.9510204    0.9160061     0.9744387
71 1976-11-03     0.9510204    0.9160061     0.9744387
72 1976-11-04     0.9510204    0.9160061     0.9744387
73 1976-11-05     0.9551020    0.9210866     0.9773771</code></pre>
</div>
<p>With <code>tail()</code>, we show that the latest CFR estimates are
equal to the <code>cfr_static()</code> outputs</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the rolling daily delay-adjusted CFR</span></span>
<span><span class="va">rolling_cfr_adjusted</span> <span class="op">&lt;-</span> <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_rolling</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">ebola1976</span>,</span>
<span>  delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">onset_to_death_ebola</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">tail</span><span class="op">(</span><span class="va">rolling_cfr_adjusted</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>         date severity_mean severity_low severity_high
68 1976-10-31         0.975        0.856             1
69 1976-11-01         0.975        0.856             1
70 1976-11-02         0.971        0.852             1
71 1976-11-03         0.971        0.852             1
72 1976-11-04         0.971        0.852             1
73 1976-11-05         0.971        0.852             1</code></pre>
</div>
<p>Now, let’s visualise both results in a time series. How would the
naive and delay-adjusted CFR estimates perform in real time?</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the latest delay-adjusted CFR</span></span>
<span><span class="va">rolling_cfr_adjusted_end</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">slice_tail</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># bind by rows both output data frames</span></span>
<span><span class="fu">bind_rows</span><span class="op">(</span></span>
<span>  <span class="va">rolling_cfr_naive</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"naive"</span><span class="op">)</span>,</span>
<span>  <span class="va">rolling_cfr_adjusted</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"adjusted"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># visualise both adjusted and unadjusted rolling estimates</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_ribbon</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span></span>
<span>      <span class="va">date</span>,</span>
<span>      ymin <span class="op">=</span> <span class="va">severity_low</span>,</span>
<span>      ymax <span class="op">=</span> <span class="va">severity_high</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">method</span></span>
<span>    <span class="op">)</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span><span class="va">date</span>, <span class="va">severity_mean</span>, colour <span class="op">=</span> <span class="va">method</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_mean</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_low</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">rolling_cfr_adjusted_end</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">severity_high</span><span class="op">)</span>,</span>
<span>    lty <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/severity-static-rendered-unnamed-chunk-13-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The horizontal line represents the delay-adjusted CFR estimated at
the outbreak’s end. The dotted line means the estimate has a 95%
confidence interval (95% CI).</p>
<div id="interpret-the-early-stage-cfr-estimate" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="interpret-the-early-stage-cfr-estimate" class="callout-inner">
<h3 class="callout-title">Interpret the early-stage CFR estimate<a class="anchor" aria-label="anchor" href="#interpret-the-early-stage-cfr-estimate"></a>
</h3>
<div class="callout-content">
<p>Based on the figure above:</p>
<ul><li>How much difference in days is between the date in which the 95% CI
of the estimated <em>delay-adjusted CFR</em> vs <em>naive CFR</em> cross
with the CFR estimated at the end of the outbreak?</li>
</ul><p>Discuss:</p>
<ul><li>What are the Public health policy implications of having a
<em>delay-adjusted CFR</em> estimate?</li>
</ul></div>
</div>
</div>
<div id="accordionHint3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint3" aria-expanded="false" aria-controls="collapseHint3">
  <h4 class="accordion-header" id="headingHint3">Give me a hint</h4>
</button>
<div id="collapseHint3" class="accordion-collapse collapse" aria-labelledby="headingHint3" data-bs-parent="#accordionHint3">
<div class="accordion-body">
<p>We can use either visual inspection or analysis of the output data
frames.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<p>There is almost one month of difference.</p>
<p>Note that the estimate has considerable uncertainty at the beginning
of the time series. After two weeks, the delay-adjusted CFR approaches
the overall CFR estimate at the outbreak’s end.</p>
<p>Is this pattern similar to other outbreaks? We can use the data sets
in this episode’s challenges. We invite you to find it out!</p>
</div>
</div>
</div>
</div>
<div id="accordionSpoiler5" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler5" aria-expanded="false" aria-controls="collapseSpoiler5">
  <h3 class="accordion-header" id="headingSpoiler5">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>rolling or time-varying?</h3>
</button>
<div id="collapseSpoiler5" class="accordion-collapse collapse" aria-labelledby="headingSpoiler5" data-bs-parent="#accordionSpoiler5">
<div class="accordion-body">
<p>The <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> package has a function called
<code>cfr_time_varying()</code> with different functionality that makes
them not comparable.</p>
<div class="section level4">
<h4 id="when-to-use-cfr_rolling">When to use <code>cfr_rolling()</code>?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_rolling"></a></h4>
<p><code>cfr_rolling()</code> shows the estimated CFR on each outbreak
day, given that future data on cases and deaths is unavailable at the
time. The final value of <code>cfr_rolling()</code> estimates is
identical to <code>cfr_static()</code> on the same data.</p>
<p>Remember, as shown above, <code>cfr_rolling()</code> is helpful to
get early-stage CFR estimates and check whether an outbreak’s CFR
estimate has stabilised. Thus, <code>cfr_rolling()</code> is not
sensitive to the length or size of the epidemic.</p>
</div>
<div class="section level4">
<h4 id="when-to-use-cfr_time_varying">When to use <code>cfr_time_varying()</code>?<a class="anchor" aria-label="anchor" href="#when-to-use-cfr_time_varying"></a></h4>
<p>On the other hand, <code>cfr_time_varying()</code> calculates the CFR
over a moving window and helps to understand changes in CFR due to
changes in the epidemic, e.g. due to a new variant or increased immunity
from vaccination.</p>
<p>However, <code>cfr_time_varying()</code> is sensitive to sampling
uncertainty. Thus, it is sensitive to the size of the outbreak. The
higher the number of cases with expected outcomes on a given day, the
more reasonable estimates of the time-varying CFR we will get.</p>
<p>For example, with 100 cases, the fatality risk estimate will, roughly
speaking, have a 95% confidence interval ±10% of the mean estimate
(binomial CI). So if we have &gt;100 cases with expected outcomes <em>on
a given day</em>, we can get reasonable estimates of the time varying
CFR. But if we only have &gt;100 cases <em>over the course of the whole
epidemic</em>, we probably need to rely on <code>cfr_rolling()</code>
that uses the cumulative data.</p>
<p>We invite you to read this <a href="https://epiverse-trace.github.io/cfr/articles/estimate_time_varying_severity.html" class="external-link">vignette
about the <code>cfr_time_varying()</code> function</a>.</p>
</div>
</div>
</div>
</div>
</div>
<div id="checklist1" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Checklist<a class="anchor" aria-label="anchor" href="#checklist1"></a>
</h3>
<div class="callout-content">
<p>With <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we estimate the CFR as the proportion of
deaths among <strong>confirmed</strong> cases.</p>
<p>By only using <strong>confirmed</strong> cases, it is clear that all
cases that do not seek medical treatment or are not notified will be
missed, as well as all asymptomatic cases. This means that the CFR
estimate is higher than the proportion of deaths among the infected.</p>
</div>
</div>
</div>
<div id="severity-heterogeneity" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="severity-heterogeneity" class="callout-inner">
<h3 class="callout-title">Severity heterogeneity<a class="anchor" aria-label="anchor" href="#severity-heterogeneity"></a>
</h3>
<div class="callout-content">
<p>The CFR may differ across populations (e.g. age, space, treatment);
quantifying these heterogeneities can help target resources
appropriately and compare different care regimens (<a href="https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371" class="external-link">Cori
et al., 2017</a>).</p>
<p>Use the <code>cfr::covid_data</code> data frame to estimate a
delay-adjusted CFR stratified by country.</p>
</div>
</div>
</div>
<div id="accordionHint4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint4" aria-expanded="false" aria-controls="collapseHint4">
  <h4 class="accordion-header" id="headingHint4">Give me a hint</h4>
</button>
<div id="collapseHint4" class="accordion-collapse collapse" aria-labelledby="headingHint4" data-bs-parent="#accordionHint4">
<div class="accordion-body">
<p>One way to do a <em>stratified analysis</em> is to apply a model to
nested data. This <a href="https://tidyr.tidyverse.org/articles/nest.html#nested-data-and-models" class="external-link"><code>{tidyr}</code>
vignette</a> shows you how to apply the <code>group_by()</code> +
<code>nest()</code> to nest data, and then <code>mutate()</code> +
<code>map()</code> to apply the model.</p>
</div>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6">Show me the solution</h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">epiparameter</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 20,786
Columns: 4
$ date    &lt;date&gt; 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2020-01-03, 2…
$ country &lt;chr&gt; "Argentina", "Brazil", "Colombia", "France", "Germany", "India…
$ cases   &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ deaths  &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_onset_death</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"onset to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    temp <span class="op">=</span></span>
<span>      <span class="fu">map</span><span class="op">(</span></span>
<span>        .x <span class="op">=</span> <span class="va">data</span>,</span>
<span>        .f <span class="op">=</span> <span class="fu">cfr</span><span class="fu">::</span><span class="va">cfr_static</span>,</span>
<span>        delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">delay_onset_death</span>, <span class="va">x</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 19 × 5
# Groups:   country [19]
   country        data                 severity_mean severity_low severity_high
   &lt;chr&gt;          &lt;list&gt;                       &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
 1 Argentina      &lt;tibble [1,094 × 3]&gt;         0.013        0.013         0.013
 2 Brazil         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 3 Colombia       &lt;tibble [1,094 × 3]&gt;         0.022        0.022         0.022
 4 France         &lt;tibble [1,094 × 3]&gt;         0.004        0.004         0.004
 5 Germany        &lt;tibble [1,094 × 3]&gt;         0.005        0.005         0.005
 6 India          &lt;tibble [1,094 × 3]&gt;         0.012        0.012         0.012
 7 Indonesia      &lt;tibble [1,094 × 3]&gt;         0.024        0.024         0.024
 8 Iran           &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
 9 Italy          &lt;tibble [1,094 × 3]&gt;         0.007        0.007         0.007
10 Mexico         &lt;tibble [1,094 × 3]&gt;         0.046        0.046         0.046
11 Peru           &lt;tibble [1,094 × 3]&gt;         0.05         0.05          0.05 
12 Poland         &lt;tibble [1,094 × 3]&gt;         0.019        0.019         0.019
13 Russia         &lt;tibble [1,094 × 3]&gt;         0.018        0.018         0.018
14 South Africa   &lt;tibble [1,094 × 3]&gt;         0.025        0.025         0.025
15 Spain          &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
16 Turkey         &lt;tibble [1,094 × 3]&gt;         0.006        0.006         0.006
17 Ukraine        &lt;tibble [1,094 × 3]&gt;         0.02         0.02          0.02 
18 United Kingdom &lt;tibble [1,094 × 3]&gt;         0.009        0.009         0.009
19 United States  &lt;tibble [1,094 × 3]&gt;         0.011        0.011         0.011</code></pre>
</div>
<p>Great! Now you can use similar code for any other stratified analysis
like age, regions or more!</p>
<p>But, how can we interpret that there is a country variability of
severity from the same diagnosed pathogen?</p>
<p>Local factors like testing capacity, the case definition, and
sampling regime can affect the report of cases and deaths, thus
affecting case ascertainment. Take a look to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>
vignette on <a href="https://epiverse-trace.github.io/cfr/articles/estimate_ascertainment.html" class="external-link">Estimating
the proportion of cases that are ascertained during an outbreak</a>!</p>
</div>
</div>
</div>
</div>
<div id="limitations" class="callout checklist">
<div class="callout-square">
<i class="callout-icon" data-feather="check-square"></i>
</div>
<div id="limitations" class="callout-inner">
<h3 class="callout-title">Limitations<a class="anchor" aria-label="anchor" href="#limitations"></a>
</h3>
<div class="callout-content">
<ul><li>The delay-adjusted CFR does not address all sources of error in data
like the underdiagnosis of infected individuals.</li>
</ul></div>
</div>
</div>
</section><section id="more-severity-measures"><h2 class="section-heading">More severity measures<a class="anchor" aria-label="anchor" href="#more-severity-measures"></a>
</h2>
<hr class="half-width"><p>Suppose we need to assess the clinical severity of the epidemic in a
context different from surveillance data, like the severity among cases
that arrive at hospitals or cases you collected from a representative
serological survey.</p>
<p>Using <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a>, we can change the inputs for the numerator
(<code>cases</code>) and denominator (<code>deaths</code>) to estimate
more severity measures like the Infection fatality risk (IFR) or the
Hospitalisation Fatality Risk (HFR). We can follow this analogy:</p>
<p>If for a <em>Case</em> fatality risk (CFR), we require:</p>
<ul><li>
<em>case</em> and death incidence data, with a</li>
<li>case-to-death delay distribution (or close approximation, such as
symptom onset-to-death).</li>
</ul><p>Then, the <em>Infection</em> fatality risk (IFR) requires:</p>
<ul><li>
<em>infection</em> and death incidence data, with an</li>
<li>exposure-to-death delay distribution (or close approximation).</li>
</ul><p>Similarly, the <em>Hospitalisation</em> Fatality Risk (HFR)
requires:</p>
<ul><li>
<em>hospitalisation</em> and death incidence data, and a</li>
<li>hospitalisation-to-death delay distribution.</li>
</ul><div id="data-sources-for-more-severity-measures" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="data-sources-for-more-severity-measures" class="callout-inner">
<h3 class="callout-title">Data sources for more severity measures<a class="anchor" aria-label="anchor" href="#data-sources-for-more-severity-measures"></a>
</h3>
<div class="callout-content">
<p><a href="https://www.nature.com/articles/s41467-020-19238-2/figures/1" class="external-link">Yang
et al., 2020</a> summarises different definitions and data sources:</p>
<figure><img src="../fig/cfr-s41467-020-19238-2-fig_a.png" alt="Severity levels of infections with SARS-CoV-2 and parameters of interest. Each level is assumed to be a subset of the level below." class="figure mx-auto d-block"><div class="figcaption">Severity levels of infections with SARS-CoV-2
and parameters of interest. Each level is assumed to be a subset of the
level below.</div>
</figure><ul><li>sCFR symptomatic case-fatality risk,</li>
<li>sCHR symptomatic case-hospitalisation risk,</li>
<li>mCFR medically attended case-fatality risk,</li>
<li>mCHR medically attended case-hospitalisation risk,</li>
<li>HFR hospitalisation-fatality risk.</li>
</ul><figure><img src="../fig/cfr-s41467-020-19238-2-fig_b.png" alt="Data source of COVID-19 cases in Wuhan: D1) 32,583 laboratory-confirmed COVID-19 cases as of March 84, D2) 17,365 clinically-diagnosed COVID-19 cases during February 9–194, D3)daily number of laboratory-confirmed cases on March 9–April 243, D4) total number of COVID-19 deaths as of April 24 obtained from the Hubei Health Commission3, D5) 325 laboratory-confirmed cases and D6) 1290 deaths were added as of April 16 through a comprehensive and systematic verification by Wuhan Authorities3, and D7) 16,781 laboratory-confirmed cases identified through universal screening10,11. Pse: RT-PCR sensitivity12. Pmed.care: proportion of seeking medical assistance among patients suffering from acute respiratory infections13." class="figure mx-auto d-block"><div class="figcaption">Schematic diagram of the baseline analyses. Red,
blue, and green arrows denote the data flow from laboratory-confirmed
cases of passive surveillance, clinically-diagnosed cases, and
laboratory-confirmed cases of active screenings.</div>
</figure></div>
</div>
</div>
<div id="how-to-rearrange-my-input-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-rearrange-my-input-data" class="callout-inner">
<h3 class="callout-title">How to rearrange my input data?<a class="anchor" aria-label="anchor" href="#how-to-rearrange-my-input-data"></a>
</h3>
<div class="callout-content">
<p>Rearranging the input data for data analysis can take most of the
time.</p>
<p>For <em>aggregated case data</em>, we encourage you to use
<a href="https://www.reconverse.org/incidence2/" class="external-link">incidence2</a>! Refer to the <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> vignette on
<a href="https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html" class="external-link">Handling
data from <code>{incidence2}</code></a>.</p>
<!-- cite howto entry one lineslist + incidence2 + cfr connection -->
</div>
</div>
</div>
<div id="hospitalisation-fatality-risk-hfr" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="hospitalisation-fatality-risk-hfr" class="callout-inner">
<h3 class="callout-title">Hospitalisation Fatality Risk (HFR)<a class="anchor" aria-label="anchor" href="#hospitalisation-fatality-risk-hfr"></a>
</h3>
<div class="callout-content">
<p>From the <code>{covidregionaldata}</code> package, load COVID cases
from the region England from country UK to estimate the delay-adjusted
HFR using the corresponding distribution delay.</p>
</div>
</div>
</div>
<div id="accordionHint5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button hint-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseHint5" aria-expanded="false" aria-controls="collapseHint5">
  <h4 class="accordion-header" id="headingHint5">Give me a hint</h4>
</button>
<div id="collapseHint5" class="accordion-collapse collapse" aria-labelledby="headingHint5" data-bs-parent="#accordionHint5">
<div class="accordion-body">
<p>Use <code>covidregionaldata::get_regional_data()</code> to get COVID
data from the region England from the country UK.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_england</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">covidregionaldata</span><span class="fu">::</span><span class="fu">get_regional_data</span><span class="op">(</span></span>
<span>    country <span class="op">=</span> <span class="st">"uk"</span>,</span>
<span>    regions <span class="op">=</span> <span class="st">"England"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Also, check out the vignette on <a href="https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html" class="external-link">Handling
data from <code>{incidence2}</code></a></p>
</div>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7">Show me the solution</h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">cfr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">covidregionaldata</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_england</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">covidregionaldata</span><span class="fu">::</span><span class="fu">get_regional_data</span><span class="op">(</span></span>
<span>    country <span class="op">=</span> <span class="st">"uk"</span>,</span>
<span>    regions <span class="op">=</span> <span class="st">"England"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_england</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 1,415
Columns: 26
$ date                             &lt;date&gt; 2020-01-30, 2020-01-31, 2020-02-01, …
$ region                           &lt;chr&gt; "England", "England", "England", "Eng…
$ region_code                      &lt;chr&gt; "E92000001", "E92000001", "E92000001"…
$ cases_new                        &lt;dbl&gt; 1, 0, 0, 1, 18, 0, 1, 0, 0, 3, 1, 1, …
$ cases_total                      &lt;dbl&gt; 1, 1, 1, 2, 20, 20, 21, 21, 21, 24, 2…
$ deaths_new                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ deaths_total                     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ recovered_new                    &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ recovered_total                  &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ hosp_new                         &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ hosp_total                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ tested_new                       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ tested_total                     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ areaType                         &lt;chr&gt; "nation", "nation", "nation", "nation…
$ cumCasesByPublishDate            &lt;dbl&gt; NA, 2, 2, 2, 2, 2, 2, 3, 3, 3, 4, 8, …
$ cumCasesBySpecimenDate           &lt;dbl&gt; 1, 1, 1, 2, 20, 20, 21, 21, 21, 24, 2…
$ newCasesByPublishDate            &lt;dbl&gt; NA, 2, 0, 0, 0, 0, 0, 1, 0, 0, 1, 4, …
$ newCasesBySpecimenDate           &lt;dbl&gt; 1, 0, 0, 1, 18, 0, 1, 0, 0, 3, 1, 1, …
$ cumDeaths28DaysByDeathDate       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ cumDeaths28DaysByPublishDate     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newDeaths28DaysByDeathDate       &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newDeaths28DaysByPublishDate     &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarFourTestsByPublishDate  &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarOneTestsByPublishDate   &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarThreeTestsByPublishDate &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ newPillarTwoTestsByPublishDate   &lt;dbl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_data_incidence</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">covid_england</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">incidence2</span><span class="fu">::</span><span class="fu">incidence</span><span class="op">(</span></span>
<span>    date_index <span class="op">=</span> <span class="st">"date"</span>,</span>
<span>    counts <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"hosp_total"</span>, <span class="st">"deaths_total"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_incidence</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># incidence:  2,830 x 3
# count vars: hosp_total, deaths_total
   date_index count_variable count
 * &lt;date&gt;     &lt;fct&gt;          &lt;dbl&gt;
 1 2020-01-30 hosp_total        NA
 2 2020-01-30 deaths_total      NA
 3 2020-01-31 hosp_total        NA
 4 2020-01-31 deaths_total      NA
 5 2020-02-01 hosp_total        NA
 6 2020-02-01 deaths_total      NA
 7 2020-02-02 hosp_total        NA
 8 2020-02-02 deaths_total      NA
 9 2020-02-03 hosp_total        NA
10 2020-02-03 deaths_total      NA
# ℹ 2,820 more rows</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_data_prepared</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">prepare_data</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">covid_data_incidence</span>,</span>
<span>    cases_variable <span class="op">=</span> <span class="st">"hosp_total"</span>,</span>
<span>    deaths_variable <span class="op">=</span> <span class="st">"deaths_total"</span>,</span>
<span>    fill_NA <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_prepared</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Rows: 1,415
Columns: 3
$ date   &lt;date&gt; 2020-01-30, 2020-01-31, 2020-02-01, 2020-02-02, 2020-02-03, 20…
$ cases  &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ deaths &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">delay_hosp_death</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">epiparameter</span><span class="fu">::</span><span class="fu">epidist_db</span><span class="op">(</span></span>
<span>    disease <span class="op">=</span> <span class="st">"covid"</span>,</span>
<span>    epi_dist <span class="op">=</span> <span class="st">"hospitalisation to death"</span>,</span>
<span>    single_epidist <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">covid_data_prepared</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">cfr</span><span class="fu">::</span><span class="fu">cfr_static</span><span class="op">(</span></span>
<span>    delay_density <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">density</span><span class="op">(</span><span class="va">delay_hosp_death</span>, <span class="va">x</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  severity_mean severity_low severity_high
1         0.215        0.215         0.215</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li><p>Use <a href="https://github.com/epiverse-trace/cfr" class="external-link">cfr</a> to estimate severity</p></li>
<li><p>Use <code>cfr_static()</code> to estimate the overall CFR with
the latest data available.</p></li>
<li><p>Use <code>cfr_rolling()</code> to show what the estimated CFR
would be on each day of the outbreak.</p></li>
<li><p>Use the <code>delay_density</code> argument to adjust the CFR by
the corresponding delay distribution.</p></li>
</ul></div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/create-forecast.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/create-forecast.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Create a short-term
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/epiverse-trace/tutorials/edit/main/episodes/severity-static.Rmd" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/epiverse-trace/tutorials/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/epiverse-trace/tutorials/" class="external-link">Source</a></p>
				<p><a href="https://github.com/epiverse-trace/tutorials/blob/main/" class="external-link">Cite</a> | <a href="mailto:andree.valle-campos@lshtm.ac.uk">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/epiverse-trace/sandpaper/tree/patch-renv-github-bug" class="external-link">sandpaper (0.16.3)</a>, <a href="https://github.com/carpentries/pegboard/tree/e9ca729c7db3ca97f016480679f9f58d115d2721" class="external-link">pegboard (0.7.4)</a>, and <a href="https://github.com/epiverse-trace/varnish/tree/epiversetheme" class="external-link">varnish (1.0.2.9000)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://epiverse-trace.github.io/tutorials/instructor/severity-static.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "forecasts, epidemic models, interventions",
  "name": "Estimate severity",
  "creativeWorkStatus": "active",
  "url": "https://epiverse-trace.github.io/tutorials/instructor/severity-static.html",
  "identifier": "https://epiverse-trace.github.io/tutorials/instructor/severity-static.html",
  "dateCreated": "2024-03-25",
  "dateModified": "2024-03-25",
  "datePublished": "2024-03-25"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

